---
title: "Preparation for factor-centric chromatin analysis"
author: "JH"
---

::: {.callout-important}
You will need to review this material before class 20.
:::

## Experimental methods

Skene PJ, Henikoff S. An efficient targeted nuclease strategy for high-resolution mapping of DNA binding sites. Elife. 2017 PMID: 28079019; PMCID: PMC5310842. [\[Link\]](https://pubmed.ncbi.nlm.nih.gov/28079019/)

Kaya-Okur HS, Wu SJ, Codomo CA, Pledger ES, Bryson TD, Henikoff JG, Ahmad K, Henikoff S. CUT&Tag for efficient epigenomic profiling of small samples and single cells. Nat Commun. 2019 PMID: 31036827; PMCID: PMC6488672. [[Link]](https://pubmed.ncbi.nlm.nih.gov/31036827/)

## Software tools

MACS is the gold-standard in peak calling. It models read coverage as a Poisson process, enabling identification of regions of higher than expected coverage (i.e., peaks) to be identified using a single parmaeter (lambda) that captures the mean and variance of read coverage. Read over the [paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2592715) to get a sense of how it works.

We'll use the [motifRG](https://www.bioconductor.org/packages//2.11/bioc/html/motifRG.html) R library, which implements a discriminative (i.e., foreground / background) approach for motif discovery and answer the question, "Which sequences drive factor association to DNA?".
