---
title: "Where do proteins bind in the genome?"
author: "{{< var instructor.block.dna >}}"
format: revealjs
---

## What to map and how to map it?

::: columns
::: {.column width="50%"}
**Targets**

-   Transcription factors
-   Histone modifications
-   Chromatin remodelers
-   RNA polymerases
-   Other factors that bind chromatin
:::

::: {.column width="50%"}
**Methods**

-   ChIP-seq
-   MNase-ChIP-seq
-   CUT&RUN
-   CUT&TAG
:::
:::

##

![](../img/block-dna/ChIPseq_2.png){fig-align="center"}

##

![](../img/block-dna/ChIP_Data.png){fig-align="center"}

##

![](../img/block-dna/cut-and-run.png){fig-align="center"}

##

![](../img/block-dna/chip-resolution-comparison.png) {fig-align="center"}

## Comparison of factor-centric methods {.smaller}

::: columns
::: {.column width="50%"}
![](../img/block-dna/chip-comparison-overview.png)
:::

::: {.column width="50%"}
|     Method     | Resolution | Sequencing cost |
|:--------------:|:----------:|:---------------:|
|    ChIP-seq    |    Low     |      High       |
| MNase-ChIP-seq |    High    |      High       |
|    CUT&RUN     |    High    |       Low       |
:::
:::

# Workflow

### FASTQ files

-   Adapter trimming

-   Aligning to the genome

### Bed files

-   Generate read density genome-wide

### Read density (wig/bedgraph)

-   Call peaks

-   Meta analysis

-   Identify motifs

-   Compare perturbations to control, compare to other datasets

## Example data: CTCF CUT&RUN in K562 cells

![](../img/block-dna/ctcf_cut_run_track.png){fig-align="center" width="800"}

## Example data: CTCF CUT&RUN in K562 cells

![](../img/block-dna/ctcf_cut_run_meta.png){fig-align="center" width="106"}

(from Skene and Henikoff, eLIFE 2017)
