<!DOCTYPE html>
<html lang="en"><head>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/tabby.min.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-8f4e3aabaa80e3e650ae20f2c24f966b.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.8.24">

  <meta name="author" content="Neel Mukherjee">
  <meta name="dcterms.date" content="2025-09-17">
  <title>MOLB 7950 – Stats Bootcamp - class 15</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    html { -webkit-text-size-adjust: 100%; }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #97947a;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #97947a;  padding-left: 4px; }
    div.sourceCode
      { color: #f8f8f2; background-color: #2b2b2b; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #f8f8f2; } /* Normal */
    code span.al { color: #dcc6e0; } /* Alert */
    code span.an { color: #d4d0ab; } /* Annotation */
    code span.at { color: #ffd700; } /* Attribute */
    code span.bn { color: #dcc6e0; } /* BaseN */
    code span.bu { color: #f5ab35; } /* BuiltIn */
    code span.cf { color: #ffa07a; } /* ControlFlow */
    code span.ch { color: #abe338; } /* Char */
    code span.cn { color: #ffa07a; } /* Constant */
    code span.co { color: #d4d0ab; } /* Comment */
    code span.cv { color: #d4d0ab; font-style: italic; } /* CommentVar */
    code span.do { color: #d4d0ab; font-style: italic; } /* Documentation */
    code span.dt { color: #dcc6e0; } /* DataType */
    code span.dv { color: #dcc6e0; } /* DecVal */
    code span.er { color: #dcc6e0; } /* Error */
    code span.ex { color: #ffd700; } /* Extension */
    code span.fl { color: #f5ab35; } /* Float */
    code span.fu { color: #ffd700; } /* Function */
    code span.im { } /* Import */
    code span.in { color: #d4d0ab; } /* Information */
    code span.kw { color: #ffa07a; } /* Keyword */
    code span.op { color: #00e0e0; } /* Operator */
    code span.ot { color: #ffa07a; } /* Other */
    code span.pp { color: #dcc6e0; } /* Preprocessor */
    code span.sc { color: #00e0e0; } /* SpecialChar */
    code span.ss { color: #abe338; } /* SpecialString */
    code span.st { color: #abe338; } /* String */
    code span.va { color: #f5ab35; } /* Variable */
    code span.vs { color: #abe338; } /* VerbatimString */
    code span.wa { color: #d4d0ab; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="../site_libs/revealjs/dist/theme/quarto-8fafb62820c35118272e29c1027a293f.css">
  <link href="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
<meta property="og:title" content="Stats Bootcamp - class 15 – MOLB 7950">
<meta property="og:description" content="Dealing with big data">
<meta property="og:site_name" content="MOLB 7950">
<meta name="twitter:title" content="Stats Bootcamp - class 15 – MOLB 7950">
<meta name="twitter:description" content="Dealing with big data">
<meta name="twitter:image" content="https://rnabioco.github.io/molb-7950/slides/images/twitter-card.png">
<meta name="twitter:creator" content="@rnabioco">
<meta name="twitter:card" content="summary_large_image">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Stats Bootcamp - class 15</h1>
  <p class="subtitle">Dealing with big data</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Neel Mukherjee 
</div>
        <p class="quarto-title-affiliation">
            RNA Bioscience Initiative | CU Anschutz
          </p>
    </div>
</div>

  <p class="date">2025-09-17</p>
</section>
<section id="learning-objectives" class="slide level2">
<h2>Learning objectives</h2>
<ul>
<li class="fragment">Types of error and multiple test corrections</li>
<li class="fragment">Exploratory data analysis</li>
<li class="fragment">Clustering and overlaps</li>
</ul>
</section>
<section id="types-i-and-ii-error" class="slide level2">
<h2>Types I and II error</h2>
<p>False positives and False negatives</p>

<img data-src="https://www.scribbr.com/wp-content/uploads/2021/01/type-i-and-ii-error-2.png" class="r-stretch"><p><span class="math inline">\(\alpha\)</span> - significance level OR evidentiary standard</p>
<p><span class="math inline">\(\beta\)</span> - type II error rate, 1 - <span class="math inline">\(\beta\)</span> is power</p>
</section>
<section id="different-visualization" class="slide level2">
<h2>Different visualization</h2>
<p>Power vs Significance</p>

<img data-src="https://www.scribbr.com/wp-content/uploads/2021/01/type-i-and-type-ii-error.png" class="r-stretch"></section>
<section id="genomics---lots-of-data---lots-of-hypothesis-tests" class="slide level2">
<h2>Genomics -&gt; Lots of Data -&gt; Lots of Hypothesis Tests</h2>
<p>In a typical RNA-seq experiment, we test ~10K different hypotheses. For example, you have 10K genes and for each gene you test whether the mean expression changed in condition A vs condition B. Using a standard p-value cut-off of 0.05, we’d expect <strong>500 genes</strong> to be deemed “significant” by chance. Thus, we are very concerned about <strong>False Positives or Type I Errors</strong>.</p>
</section>
<section id="multiple-test-corrections" class="slide level2 smaller">
<h2>Multiple test corrections</h2>
<ol type="1">
<li class="fragment"><p>Control overall α (also known as family-wise error rate or <a href="https://en.wikipedia.org/wiki/Family-wise_error_rate">FWER</a>), which will affect the α* for each test. That is, we are controlling the overall probability of making <em>at least one</em> false discovery. Bonferroni and Sidak corrections all control FWER.</p></li>
<li class="fragment"><p>Control <a href="https://en.wikipedia.org/wiki/False_discovery_rate">false discovery rate</a> (FDR). These procedures allow for type 1 errors (false positives) but control the proportion of these false positives in relation to true positives. This is done by adjusting the decision made for the p-value associated with each individual test to decide rejection or not. Because this will result in a higher type 1 error rate, it has higher <a href="https://en.wikipedia.org/wiki/Power_(statistics)">power</a>. This affords a higher probability of <em>true discoveries.</em> The step procedures control for FDR.</p></li>
</ol>
</section>
<section id="bonferroni-correction" class="slide level2 smaller">
<h2>Bonferroni Correction</h2>
<p>The most conservative of corrections, the Bonferroni correction is also perhaps the most straightforward in its approach. Simply divide α by the number of tests (<em>m</em>).</p>
<blockquote>
<p>α = α/m</p>
</blockquote>
<p>However, with many tests, α will become very small. This reduces power, which means that we are very unlikely to make any true discoveries.</p>
<h3 id="sidak-correction">Sidak Correction</h3>
<blockquote>
<p>α = 1-(1-α)^(1/<em>m</em>)</p>
</blockquote>
</section>
<section id="holms-step-down-procedure" class="slide level2 smaller">
<h2>Holm’s Step-Down Procedure</h2>
<p>The Holm-Bonferroni method is also fairly simple to calculate, but it is more powerful than the single-step Bonferroni.</p>
<p><span class="math inline">\(HB = \displaystyle \frac {target \alpha}{n - rank + 1}\)</span></p>
<p>H1: 0.005<br>
H2: 0.01<br>
H3: 0.03<br>
H4: = 0.04<br>
</p>
<p>Step 1: Order the p-values from smallest to greatest (already done)</p>
<p>Step 2: Calc HB for the first rank HB = .05 / 4 – 1 + 1 = .05 / 4 = .0125 H1: 0.005 &lt; .0125, so we reject the null</p>
<p>Step 4: Repeat the HB formula for the second rank and keep going until we find <span class="math inline">\(H{_N}\)</span> &gt; <span class="math inline">\(HB{_N}\)</span>. All subsequent hypotheses are non-significant (i.e.&nbsp;not rejected).</p>
</section>
<section id="hochbergs-step-up-procedure" class="slide level2 smaller">
<h2>Hochberg’s Step-Up Procedure</h2>
<p>More powerful than Holm’s step-down procedure, Hochberg’s step-up procedure also seeks to control the FDR and follows a similar process, only p-values are ranked from largest to smallest.</p>
<p>For each ranked p-value, it is compared to the α calculated for its respective rank (same formula as Holm’s procedure). Testing continues until you reach the first non-rejected hypothesis. You would then fail to reject all following hypotheses.</p>
</section>
<section id="example" class="slide level2 smaller">
<h2>Example</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a></a>rna <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">here</span>(<span class="st">"data/bootcamp/data_rna_protein.csv.gz"</span>)) <span class="sc">|&gt;</span> <span class="fu">select</span>(iDUX4_pval)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="example-output" class="slide level2 smaller output-location-slide"><h2>Example</h2><div class="cell output-location-slide">
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 21282 Columns: 17
── Column specification ────────────────────────────────────
Delimiter: ","
chr  (1): geneid
dbl (16): iDUX4_logFC, iDUX4_logCPM, iDUX4_LR, iDUX4_pva...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a></a>rna<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> rna<span class="sc">$</span>iDUX4_pval, <span class="at">method =</span> <span class="st">"fdr"</span>, <span class="at">n =</span> <span class="fu">nrow</span>(rna))</span>
<span id="cb3-2"><a></a>rna<span class="sc">$</span>BH <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> rna<span class="sc">$</span>iDUX4_pval, <span class="at">method =</span> <span class="st">"BH"</span>, <span class="at">n =</span> <span class="fu">nrow</span>(rna))</span>
<span id="cb3-3"><a></a>rna<span class="sc">$</span>bon <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> rna<span class="sc">$</span>iDUX4_pval, <span class="at">method =</span> <span class="st">"bonferroni"</span>, <span class="at">n =</span> <span class="fu">nrow</span>(rna))</span>
<span id="cb3-4"><a></a>rna_long <span class="ot">&lt;-</span> rna <span class="sc">|&gt;</span> <span class="fu">pivot_longer</span>(<span class="at">cols =</span> iDUX4_pval<span class="sc">:</span>bon, <span class="at">names_to =</span> <span class="st">"type"</span>)</span>
<span id="cb3-5"><a></a></span>
<span id="cb3-6"><a></a><span class="fu">ggplot</span>(<span class="at">data =</span> rna_long, <span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">fill =</span> type)) <span class="sc">+</span></span>
<span id="cb3-7"><a></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb3-8"><a></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>type) <span class="sc">+</span></span>
<span id="cb3-9"><a></a>  <span class="fu">theme_cowplot</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 35800 rows containing non-finite outside the scale
range (`stat_bin()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-15_files/figure-revealjs/unnamed-chunk-2-1.png" width="960"></p>
</figure>
</div>
</div>
</div></section><section id="exploratory-data-analysis-eda" class="slide level2 smaller">
<h2>Exploratory data analysis (EDA)</h2>
<p>Our goal here is to get an top-down big picture of the similarity/differences between variables in a dataset. For example, let’s say you do RNA-seq in triplicate on 4 treatment/developmental times.</p>
<h3 id="pca">PCA</h3>
<p>We will perform PCA on all of the samples and visualize the relationship between samples.</p>
<h3 id="correlation-matrix">Correlation matrix</h3>
<p>We will perform hierarchical clustering on a matrix representing the pairwise correlation between all these samples.</p>
</section>
<section id="explore-data" class="slide level2 smaller">
<h2>Explore data</h2>
<p>Is it normal-ish?</p>
<div class="cell columns column-output-location">
<div class="column">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a></a><span class="co"># get dux targets</span></span>
<span id="cb5-2"><a></a>dux_targets <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"target_genes.csv.gz"</span>))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div><div class="column fragment">
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 74 Columns: 1
── Column specification ────────────────────────────────────
Delimiter: ","
chr (1): hgnc_symbol

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a></a>d <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"data_genelevel.tsv.gz"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb7-2"><a></a>  <span class="fu">mutate</span>(<span class="at">target =</span> <span class="fu">case_when</span>(</span>
<span id="cb7-3"><a></a>    gene_symbol <span class="sc">%in%</span> dux_targets<span class="sc">$</span>hgnc_symbol <span class="sc">~</span> <span class="st">"target"</span>,</span>
<span id="cb7-4"><a></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"not_target"</span></span>
<span id="cb7-5"><a></a>  )) <span class="sc">|&gt;</span></span>
<span id="cb7-6"><a></a>  <span class="fu">filter</span>(gene_symbol <span class="sc">!=</span> <span class="st">"ISCA1"</span>) <span class="sc">|&gt;</span></span>
<span id="cb7-7"><a></a>  <span class="fu">drop_na</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 10568 Columns: 13
── Column specification ────────────────────────────────────
Delimiter: "\t"
chr  (1): gene_symbol
dbl (12): hour00_rep1, hour00_rep2, hour00_rep3, hour04_...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a></a>d <span class="sc">|&gt;</span></span>
<span id="cb9-2"><a></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> hour00_rep1<span class="sc">:</span>hour14_rep3) <span class="sc">|&gt;</span></span>
<span id="cb9-3"><a></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">color =</span> name)) <span class="sc">+</span></span>
<span id="cb9-4"><a></a>  <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb9-5"><a></a>  <span class="fu">theme_cowplot</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-15_files/figure-revealjs/unnamed-chunk-3-1.png" width="960"></p>
</figure>
</div>
</div>
</div></div>
<div class="fragment">
<p>Definitely not normal</p>
</div>
</section>
<section id="data-transformations" class="slide level2 smaller">
<h2>Data transformations</h2>
<p>We often transform data to make it closer to being normally-distributed. This allows us to use more powerful statistical tests on the same data. One approach is to log-transform the data.</p>
<div class="cell columns column-output-location">
<div class="column">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a></a>d <span class="sc">|&gt;</span></span>
<span id="cb10-2"><a></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> hour00_rep1<span class="sc">:</span>hour14_rep3) <span class="sc">|&gt;</span></span>
<span id="cb10-3"><a></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log10</span>(value), <span class="at">color =</span> name)) <span class="sc">+</span></span>
<span id="cb10-4"><a></a>  <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb10-5"><a></a>  <span class="fu">theme_cowplot</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div><div class="column fragment">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1239 rows containing non-finite outside the scale
range (`stat_density()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-15_files/figure-revealjs/unnamed-chunk-4-1.png" width="960"></p>
</figure>
</div>
</div>
</div></div>
<div class="fragment">
<p>What is this?</p>
<blockquote>
<p>Warning message: Removed 1251 rows containing non-finite values (stat_density()).</p>
</blockquote>
</div>
</section>
<section id="pseudocounts" class="slide level2 smaller">
<h2>Pseudocounts</h2>
<p><span class="math inline">\(log_{x}(0)\)</span> is a common problem. One solution is to add a pseudocount. Since this is read count data, the smallest unit is 1 and so we will add 1 to all the observations before perforing the log transformation. <span class="math inline">\(1\)</span> represents the pseudocount in this case.</p>
<div class="cell columns column-output-location">
<div class="column">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a></a>d <span class="sc">|&gt;</span></span>
<span id="cb12-2"><a></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> hour00_rep1<span class="sc">:</span>hour14_rep3) <span class="sc">|&gt;</span></span>
<span id="cb12-3"><a></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log10</span>(value <span class="sc">+</span> <span class="dv">1</span>), <span class="at">color =</span> name)) <span class="sc">+</span></span>
<span id="cb12-4"><a></a>  <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb12-5"><a></a>  <span class="fu">theme_cowplot</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div><div class="column fragment">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-15_files/figure-revealjs/unnamed-chunk-5-1.png" width="960"></p>
</figure>
</div>
</div>
</div></div>
</section>
<section id="correlation-analysis" class="slide level2">
<h2>correlation analysis</h2>
<p>prepare the data for analysis</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a></a><span class="co"># pull counts</span></span>
<span id="cb13-2"><a></a>x <span class="ot">&lt;-</span> d <span class="sc">|&gt;</span></span>
<span id="cb13-3"><a></a>  <span class="fu">select_if</span>(is.numeric) <span class="sc">|&gt;</span> <span class="co"># keep only the numeric columns</span></span>
<span id="cb13-4"><a></a>  <span class="fu">mutate_all</span>(<span class="fu">funs</span>(<span class="fu">log2</span>(<span class="dv">1</span> <span class="sc">+</span> .))) <span class="sc">|&gt;</span> <span class="co"># log2 transform</span></span>
<span id="cb13-5"><a></a>  <span class="fu">as.matrix</span>() <span class="co"># matrix</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="correlation-analysis-output" class="slide level2 output-location-slide"><h2>correlation analysis</h2><div class="cell output-location-slide">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `funs()` was deprecated in dplyr 0.8.0.
ℹ Please use a list of either functions or lambdas:

# Simple named list: list(mean = mean, median = median)

# Auto named with `tibble::lst()`: tibble::lst(mean,
  median)

# Using lambdas list(~ mean(., trim = .2), ~ median(.,
  na.rm = TRUE))</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a></a><span class="fu">rownames</span>(x) <span class="ot">&lt;-</span> d<span class="sc">$</span>gene_symbol</span>
<span id="cb15-2"><a></a></span>
<span id="cb15-3"><a></a></span>
<span id="cb15-4"><a></a>x <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(x)))</span>
<span id="cb15-5"><a></a></span>
<span id="cb15-6"><a></a></span>
<span id="cb15-7"><a></a><span class="co"># pairwise pearson correlation</span></span>
<span id="cb15-8"><a></a>p <span class="ot">&lt;-</span> <span class="fu">cor</span>(x, <span class="at">method =</span> <span class="st">"pearson"</span>)</span>
<span id="cb15-9"><a></a></span>
<span id="cb15-10"><a></a><span class="fu">pheatmap</span>(</span>
<span id="cb15-11"><a></a>  <span class="at">mat =</span> p,</span>
<span id="cb15-12"><a></a>  <span class="at">clustering_distance_rows =</span> <span class="st">"euclidean"</span>,</span>
<span id="cb15-13"><a></a>  <span class="at">clustering_distance_cols =</span> <span class="st">"euclidean"</span>,</span>
<span id="cb15-14"><a></a>  <span class="at">clustering_method =</span> <span class="st">"ward.D2"</span></span>
<span id="cb15-15"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-15_files/figure-revealjs/unnamed-chunk-6-1.png" width="960"></p>
</figure>
</div>
</div>
</div></section><section id="pca-1" class="slide level2">
<h2>PCA</h2>
<p>PCA is a common dimensionality reduction method that is used to visualize the similarity and differences in your data.</p>
<p><strong>Let’s watch this fantastic 5 minute video explaining PCA</strong></p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/HMOI_lkzW08" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen="">
</iframe>
<p>For more detailed explanations go <a href="https://www.nature.com/articles/nbt0308-303">here</a> and <a href="https://liorpachter.wordpress.com/2014/05/26/what-is-principal-component-analysis/">here</a>.</p>
</section>
<section id="pca-2" class="slide level2">
<h2>PCA</h2>
<div class="cell columns column-output-location">
<div class="column">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a></a><span class="co"># pairwise pearson correlation</span></span>
<span id="cb16-2"><a></a>pc <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(x)</span>
<span id="cb16-3"><a></a></span>
<span id="cb16-4"><a></a><span class="fu">summary</span>(pc) <span class="co"># summarize the PCs by variance</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div><div class="column fragment">
<div class="cell-output cell-output-stdout">
<pre><code>Importance of components:
                          PC1    PC2     PC3     PC4
Standard deviation     2.3743 1.2799 0.79753 0.64825
Proportion of Variance 0.5739 0.1668 0.06475 0.04278
Cumulative Proportion  0.5739 0.7407 0.80544 0.84822
                           PC5     PC6     PC7    PC8
Standard deviation     0.53023 0.47044 0.46316 0.4541
Proportion of Variance 0.02862 0.02253 0.02184 0.0210
Cumulative Proportion  0.87684 0.89937 0.92121 0.9422
                           PC9    PC10   PC11      PC12
Standard deviation     0.44316 0.43982 0.4217 7.512e-15
Proportion of Variance 0.01999 0.01969 0.0181 0.000e+00
Cumulative Proportion  0.96221 0.98190 1.0000 1.000e+00</code></pre>
</div>
</div></div>
</section>
<section id="pca---prepare-visualization" class="slide level2">
<h2>PCA - prepare visualization</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a></a><span class="co"># create a dataframe with the importance/explanation of variation for each PC</span></span>
<span id="cb18-2"><a></a>pca_data_info <span class="ot">&lt;-</span> <span class="fu">summary</span>(pc)<span class="sc">$</span>importance <span class="sc">|&gt;</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb18-3"><a></a></span>
<span id="cb18-4"><a></a>pca_data_info <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="at">x =</span> pca_data_info, <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb18-5"><a></a></span>
<span id="cb18-6"><a></a><span class="co"># we make a dataframe out of the rotations and will use this to plot</span></span>
<span id="cb18-7"><a></a>pca_plot_data <span class="ot">&lt;-</span> pc<span class="sc">$</span>rotation <span class="sc">|&gt;</span></span>
<span id="cb18-8"><a></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb18-9"><a></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"ID"</span>) <span class="sc">|&gt;</span></span>
<span id="cb18-10"><a></a>  <span class="fu">separate</span>(<span class="at">col =</span> ID, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"time"</span>, <span class="st">"rep"</span>), <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb18-11"><a></a></span>
<span id="cb18-12"><a></a><span class="co"># recode "rep"</span></span>
<span id="cb18-13"><a></a>pca_plot_data<span class="sc">$</span>rep <span class="ot">&lt;-</span> <span class="fu">recode</span>(pca_plot_data<span class="sc">$</span>rep,</span>
<span id="cb18-14"><a></a>  <span class="at">rep1 =</span> <span class="st">"A"</span>,</span>
<span id="cb18-15"><a></a>  <span class="at">rep2 =</span> <span class="st">"B"</span>,</span>
<span id="cb18-16"><a></a>  <span class="at">rep3 =</span> <span class="st">"C"</span></span>
<span id="cb18-17"><a></a>)</span>
<span id="cb18-18"><a></a></span>
<span id="cb18-19"><a></a><span class="co"># gsub hour</span></span>
<span id="cb18-20"><a></a>pca_plot_data<span class="sc">$</span>time <span class="ot">&lt;-</span> <span class="fu">gsub</span>(</span>
<span id="cb18-21"><a></a>  <span class="at">pattern =</span> <span class="st">"hour"</span>,</span>
<span id="cb18-22"><a></a>  <span class="at">replacement =</span> <span class="st">""</span>,</span>
<span id="cb18-23"><a></a>  <span class="at">x =</span> pca_plot_data<span class="sc">$</span>time</span>
<span id="cb18-24"><a></a>)</span>
<span id="cb18-25"><a></a></span>
<span id="cb18-26"><a></a><span class="fu">ggplot</span>(<span class="at">data =</span> pca_plot_data, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> time)) <span class="sc">+</span></span>
<span id="cb18-27"><a></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb18-28"><a></a>  <span class="fu">xlab</span>(<span class="fu">paste</span>(<span class="st">"PC1, %"</span>, <span class="dv">100</span> <span class="sc">*</span> pca_data_info[<span class="st">"Proportion of Variance"</span>, <span class="st">"PC1"</span>])) <span class="sc">+</span></span>
<span id="cb18-29"><a></a>  <span class="fu">ylab</span>(<span class="fu">paste</span>(<span class="st">"PC2, %"</span>, <span class="dv">100</span> <span class="sc">*</span> pca_data_info[<span class="st">"Proportion of Variance"</span>, <span class="st">"PC2"</span>])) <span class="sc">+</span></span>
<span id="cb18-30"><a></a>  <span class="fu">ggtitle</span>(<span class="st">"PCA for DUX4 timecourse"</span>) <span class="sc">+</span></span>
<span id="cb18-31"><a></a>  <span class="fu">theme_cowplot</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="pca---prepare-visualization-output" class="slide level2 output-location-slide"><h2>PCA - prepare visualization</h2><div class="cell output-location-slide">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-15_files/figure-revealjs/unnamed-chunk-8-1.png" width="960"></p>
</figure>
</div>
</div>
</div></section><section id="famous-pca-example" class="slide level2 smaller">
<h2>Famous PCA example</h2>
<p>Using gene expression as your measurement, do you think the mouse liver is more similar to a mouse heart or a human liver?</p>
<p>The Mouse ENCODE Consortium reported that comparative gene expression data from human and mouse tend to cluster more by species rather than by tissue.</p>

<img data-src="../img/bad_pca.jpg" class="r-stretch"><p><a href="https://www.nature.com/articles/nature13992">A comparative encyclopedia of DNA elements in the mouse genome</a></p>
<p><a href="https://www.pnas.org/doi/full/10.1073/pnas.1413624111">Comparison of the transcriptional landscapes between human and mouse tissues</a></p>
</section>
<section id="some-found-this-hard-to-believe" class="slide level2 smaller">
<h2>Some found this hard to believe</h2>
<p><a href="https://f1000research.com/articles/4-121#ref-1">Yoav Gilad’s lab recapitulated the initial result:</a></p>

<img data-src="../img/redo_encode.jpg" class="r-stretch"><p>This observation was surprising, as it contradicted much of the comparative gene regulatory data collected previously, as well as the common notion that major developmental pathways are highly conserved across a wide range of species, in particular across mammals.</p>
</section>
<section id="careful-with-batch-effects" class="slide level2 smaller">
<h2>Careful with batch effects</h2>
<p>But noticed something funny about which samples were sequenced on the same lanes.</p>

<img data-src="../img/batch.jpg" class="r-stretch"></section>
<section id="accounting-for-batch-effects" class="slide level2 smaller">
<h2>Accounting for batch effects</h2>

<img data-src="../img/fixencode.jpg" class="r-stretch"><p>Here we show that the Mouse ENCODE gene expression data were collected using a flawed study design, which confounded sequencing batch (namely, the assignment of samples to sequencing flowcells and lanes) with species. When we account for the batch effect, the corrected comparative gene expression data from human and mouse tend to cluster by tissue, not by species.</p>
</section>
<section id="k-means-clustering-to-look-for-patterns" class="slide level2 smaller">
<h2>K-means clustering to look for patterns</h2>
<p>Goal: to partition <code>n</code> observations into <code>k</code> clusters in which each observation belongs to the cluster with the nearest mean, serving as a prototype of the cluster. –Wiki</p>
<p><strong>K-means</strong> <iframe width="300" height="150" src="https://www.youtube.com/embed/4b5d3muPQmA" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe></p>
</section>
<section id="k-means-data-preparation" class="slide level2 smaller">
<h2>K-Means data preparation</h2>
<ol type="1">
<li class="fragment">Rows are observations (individuals) and columns are variables</li>
<li class="fragment">Any missing value in the data must be removed or estimated.</li>
<li class="fragment">The data must be standardized (i.e., scaled) to make variables comparable.</li>
</ol>
</section>
<section id="scaling-or-z-score" class="slide level2 smaller">
<h2>Scaling or z-score</h2>
<p><img data-src="http://www.z-table.com/uploads/2/1/7/9/21795380/1426878678.png">{%20}</p>
<p><span class="math inline">\(x\)</span> = observation<br>
<span class="math inline">\(\mu\)</span> = population mean<br>
<span class="math inline">\(\sigma\)</span> = population sd</p>
<p>We will be using this function on each row. This will allow comparison of relative changes across a row, for all rows.</p>
</section>
<section id="k-means-clustering" class="slide level2">
<h2>K-Means clustering</h2>
<ol type="1">
<li class="fragment">Computing k-means clustering in R (pheatmap)</li>
<li class="fragment">Determine appropriate cluster number</li>
<li class="fragment">Add new column with cluster number to initial data</li>
</ol>
</section>
<section id="how-do-we-figure-out-the-optimal-clusters" class="slide level2">
<h2>How do we figure out the optimal # clusters?</h2>
<p>There are many methods, but we will stick with the “elbow” method.</p>
<p>K-means is minimizing the total within cluster sum of squares (wss).</p>
<p>We pick the cluster where that drop in total reaches diminishing returns -&gt; the elbow.</p>
</section>
<section id="lets-cluster-once-to-see" class="slide level2 smaller">
<h2>Let’s cluster once to see</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a></a><span class="fu">set.seed</span>(<span class="dv">33</span>)</span>
<span id="cb19-2"><a></a>tmp <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(</span>
<span id="cb19-3"><a></a>  <span class="at">mat =</span> x,</span>
<span id="cb19-4"><a></a>  <span class="at">clustering_distance_rows =</span> <span class="st">"euclidean"</span>,</span>
<span id="cb19-5"><a></a>  <span class="at">clustering_method =</span> <span class="st">"ward.D2"</span>,</span>
<span id="cb19-6"><a></a>  <span class="at">kmeans_k =</span> <span class="dv">8</span>,</span>
<span id="cb19-7"><a></a>  <span class="at">cluster_cols =</span> F,</span>
<span id="cb19-8"><a></a>  <span class="at">scale =</span> <span class="st">"none"</span></span>
<span id="cb19-9"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="lets-cluster-once-to-see-output" class="slide level2 smaller output-location-slide"><h2>Let’s cluster once to see</h2><div class="cell output-location-slide">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Quick-TRANSfer stage steps exceeded maximum (=
528300)</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-15_files/figure-revealjs/unnamed-chunk-9-1.png" width="960"></p>
</figure>
</div>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a></a>tmp<span class="sc">$</span>kmeans<span class="sc">$</span>tot.withinss</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 32071.65</code></pre>
</div>
</div></section><section id="functions-in-r" class="slide level2 smaller">
<h2>Functions in R</h2>

<img data-src="https://www.learnbyexample.org/wp-content/uploads/r/r-function-syntax.png" class="r-stretch"></section>
<section id="create-function-to-calculate-wss" class="slide level2 smaller">
<h2>Create function to calculate wss</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a></a>wss <span class="ot">&lt;-</span> <span class="cf">function</span>(knum) {</span>
<span id="cb23-2"><a></a>  ph <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(</span>
<span id="cb23-3"><a></a>    <span class="at">mat =</span> x,</span>
<span id="cb23-4"><a></a>    <span class="at">kmeans_k =</span> knum,</span>
<span id="cb23-5"><a></a>    <span class="at">scale =</span> <span class="st">"none"</span>,</span>
<span id="cb23-6"><a></a>    <span class="at">cluster_cols =</span> F,</span>
<span id="cb23-7"><a></a>    <span class="at">clustering_distance_rows =</span> <span class="st">"euclidean"</span>,</span>
<span id="cb23-8"><a></a>    <span class="at">clustering_method =</span> <span class="st">"ward.D2"</span>,</span>
<span id="cb23-9"><a></a>    <span class="at">silent =</span> T</span>
<span id="cb23-10"><a></a>  )</span>
<span id="cb23-11"><a></a>  <span class="fu">return</span>(ph<span class="sc">$</span>kmeans<span class="sc">$</span>tot.withinss)</span>
<span id="cb23-12"><a></a>}</span>
<span id="cb23-13"><a></a></span>
<span id="cb23-14"><a></a><span class="fu">wss</span>(<span class="dv">6</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 34318.06</code></pre>
</div>
</div>
</section>
<section id="find-the-elbow" class="slide level2 smaller">
<h2>find the elbow</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a></a><span class="fu">tibble</span>(</span>
<span id="cb25-2"><a></a>  <span class="at">wss =</span> <span class="fu">map_vec</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">15</span>, wss),</span>
<span id="cb25-3"><a></a>  <span class="at">k =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">15</span></span>
<span id="cb25-4"><a></a>) <span class="sc">|&gt;</span></span>
<span id="cb25-5"><a></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> k, <span class="at">y =</span> wss)) <span class="sc">+</span></span>
<span id="cb25-6"><a></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb25-7"><a></a>  <span class="fu">theme_cowplot</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell fragment">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-15_files/figure-revealjs/unnamed-chunk-11-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="final-clustering" class="slide level2 smaller">
<h2>Final clustering</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a></a><span class="fu">set.seed</span>(<span class="dv">33</span>)</span>
<span id="cb26-2"><a></a>c <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(</span>
<span id="cb26-3"><a></a>  <span class="at">mat =</span> x,</span>
<span id="cb26-4"><a></a>  <span class="at">clustering_distance_rows =</span> <span class="st">"euclidean"</span>,</span>
<span id="cb26-5"><a></a>  <span class="at">clustering_method =</span> <span class="st">"ward.D2"</span>,</span>
<span id="cb26-6"><a></a>  <span class="at">kmeans_k =</span> <span class="dv">7</span>,</span>
<span id="cb26-7"><a></a>  <span class="at">cluster_cols =</span> F,</span>
<span id="cb26-8"><a></a>  <span class="at">scale =</span> <span class="st">"none"</span></span>
<span id="cb26-9"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell fragment">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-15_files/figure-revealjs/unnamed-chunk-12-1.png" width="960"></p>
</figure>
</div>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a></a>cg <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb27-2"><a></a>  <span class="at">Cluster =</span> c<span class="sc">$</span>kmeans<span class="sc">$</span>cluster,</span>
<span id="cb27-3"><a></a>  <span class="at">gene_symbol =</span> <span class="fu">names</span>(c<span class="sc">$</span>kmeans<span class="sc">$</span>cluster)</span>
<span id="cb27-4"><a></a>)</span>
<span id="cb27-5"><a></a></span>
<span id="cb27-6"><a></a>cd <span class="ot">&lt;-</span> <span class="fu">left_join</span>(d, cg, <span class="at">by =</span> <span class="st">"gene_symbol"</span>) <span class="sc">|&gt;</span> <span class="fu">as_tibble</span>()</span>
<span id="cb27-7"><a></a></span>
<span id="cb27-8"><a></a><span class="fu">write_tsv</span>(cd, <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"dux4_clustering_results.csv.gz"</span>))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="which-clusters-contains-dux4-targets" class="slide level2 smaller">
<h2>Which cluster(s) contains DUX4 targets?</h2>
<p><a href="https://www.youtube.com/watch?v=udyAvvaMjfM">Fisher’s Exact Test and the Hypergeometric Distribution</a></p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a></a><span class="co"># list of genes by dux4 targeting</span></span>
<span id="cb28-2"><a></a>duxList <span class="ot">&lt;-</span> <span class="fu">split</span>(cd<span class="sc">$</span>gene_symbol, cd<span class="sc">$</span>target)</span>
<span id="cb28-3"><a></a></span>
<span id="cb28-4"><a></a><span class="co"># list of genes by clustering</span></span>
<span id="cb28-5"><a></a>clustList <span class="ot">&lt;-</span> <span class="fu">split</span>(cd<span class="sc">$</span>gene_symbol, <span class="fu">as.factor</span>(cd<span class="sc">$</span>Cluster))</span>
<span id="cb28-6"><a></a></span>
<span id="cb28-7"><a></a><span class="co"># calculate all overlaps between lists</span></span>
<span id="cb28-8"><a></a>gom.duxclust <span class="ot">&lt;-</span> <span class="fu">newGOM</span>(duxList,</span>
<span id="cb28-9"><a></a>  clustList,</span>
<span id="cb28-10"><a></a>  <span class="at">genome.size =</span> <span class="fu">nrow</span>(d)</span>
<span id="cb28-11"><a></a>)</span>
<span id="cb28-12"><a></a></span>
<span id="cb28-13"><a></a></span>
<span id="cb28-14"><a></a><span class="fu">getMatrix</span>(gom.duxclust, <span class="st">"pval"</span>) <span class="sc">|&gt;</span></span>
<span id="cb28-15"><a></a>  <span class="fu">t</span>() <span class="sc">|&gt;</span></span>
<span id="cb28-16"><a></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb28-17"><a></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"clust"</span>) <span class="sc">|&gt;</span></span>
<span id="cb28-18"><a></a>  <span class="fu">as.tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb28-19"><a></a>  <span class="fu">arrange</span>(target)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="which-clusters-contains-dux4-targets-output" class="slide level2 smaller output-location-slide"><h2>Which cluster(s) contains DUX4 targets?</h2><div class="cell output-location-slide">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `as.tibble()` was deprecated in tibble 2.0.0.
ℹ Please use `as_tibble()` instead.
ℹ The signature and semantics have changed, see
  `?as_tibble`.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 7 × 3
  clust not_target    target
  &lt;chr&gt;      &lt;dbl&gt;     &lt;dbl&gt;
1 5     1.000      2.59 e-14
2 6     1.000      1.57 e- 5
3 2     0.182      9.25 e- 1
4 4     0.0261     9.96 e- 1
5 1     0.000114   1.000e+ 0
6 7     0.00000128 1.000e+ 0
7 3     0.0107     1    e+ 0</code></pre>
</div>
</div></section><section id="lets-calculate-the-empirical-p-value-of-the-cluster-most-enriched-for-dux4-targets-by-sampling" class="slide level2 smaller">
<h2>Let’s calculate the empirical p-value of the cluster most enriched for DUX4 targets by sampling</h2>
<p>In order to do this, you will need to:</p>
<ol type="1">
<li class="fragment">Identify which cluster is the most enriched for DUX4 targets.
<ul>
<li class="fragment">Determine how many genes are in the cluster. You will need to know this to figure out how many genes to sample from the whole data set.</li>
<li class="fragment">Determine how many of the genes in the cluster are DUX4 targets. This is the metric that you are interested in comparing between the null distribution and your observation.</li>
</ul></li>
<li class="fragment">Generate 1000 random sample of the proper size from all genes and find out how many of them are DUX4 targets.</li>
<li class="fragment">Visualize the distribution of DUX4 targets in these 1000 random (your null distribution) and overlay the number of DUX4 targets you observed in the cluster that was most enriched for DUX4 targets.</li>
</ol>


</section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">
<p><a href="https://rnabioco.github.io/molb-7950">Course 🔗</a></p>
</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': true,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"8\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdn.jsdelivr.net/npm/mathjax@2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
          const outerScaffold = trigger.parentElement.cloneNode(true);
          const codeEl = outerScaffold.querySelector('code');
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp("https:\/\/rnabioco\.github\.io\/molb-7950\/");
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>