<!DOCTYPE html>
<html lang="en"><head>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/tabby.min.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-8f4e3aabaa80e3e650ae20f2c24f966b.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.8.24">

  <meta name="author" content="Neel Mukherjee">
  <meta name="dcterms.date" content="2025-09-29">
  <title>MOLB 7950 – RNAseq DE</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    html { -webkit-text-size-adjust: 100%; }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #97947a;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #97947a;  padding-left: 4px; }
    div.sourceCode
      { color: #f8f8f2; background-color: #2b2b2b; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #f8f8f2; } /* Normal */
    code span.al { color: #dcc6e0; } /* Alert */
    code span.an { color: #d4d0ab; } /* Annotation */
    code span.at { color: #ffd700; } /* Attribute */
    code span.bn { color: #dcc6e0; } /* BaseN */
    code span.bu { color: #f5ab35; } /* BuiltIn */
    code span.cf { color: #ffa07a; } /* ControlFlow */
    code span.ch { color: #abe338; } /* Char */
    code span.cn { color: #ffa07a; } /* Constant */
    code span.co { color: #d4d0ab; } /* Comment */
    code span.cv { color: #d4d0ab; font-style: italic; } /* CommentVar */
    code span.do { color: #d4d0ab; font-style: italic; } /* Documentation */
    code span.dt { color: #dcc6e0; } /* DataType */
    code span.dv { color: #dcc6e0; } /* DecVal */
    code span.er { color: #dcc6e0; } /* Error */
    code span.ex { color: #ffd700; } /* Extension */
    code span.fl { color: #f5ab35; } /* Float */
    code span.fu { color: #ffd700; } /* Function */
    code span.im { } /* Import */
    code span.in { color: #d4d0ab; } /* Information */
    code span.kw { color: #ffa07a; } /* Keyword */
    code span.op { color: #00e0e0; } /* Operator */
    code span.ot { color: #ffa07a; } /* Other */
    code span.pp { color: #dcc6e0; } /* Preprocessor */
    code span.sc { color: #00e0e0; } /* SpecialChar */
    code span.ss { color: #abe338; } /* SpecialString */
    code span.st { color: #abe338; } /* String */
    code span.va { color: #f5ab35; } /* Variable */
    code span.vs { color: #abe338; } /* VerbatimString */
    code span.wa { color: #d4d0ab; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="../site_libs/revealjs/dist/theme/quarto-8fafb62820c35118272e29c1027a293f.css">
  <link href="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
<meta property="og:title" content="RNAseq DE – MOLB 7950">
<meta property="og:site_name" content="MOLB 7950">
<meta name="twitter:title" content="RNAseq DE – MOLB 7950">
<meta name="twitter:image" content="https://rnabioco.github.io/molb-7950/slides/images/twitter-card.png">
<meta name="twitter:creator" content="@rnabioco">
<meta name="twitter:card" content="summary_large_image">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">RNAseq DE</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Neel Mukherjee 
</div>
        <p class="quarto-title-affiliation">
            RNA Bioscience Initiative | CU Anschutz
          </p>
    </div>
</div>

  <p class="date">2025-09-29</p>
</section>
<section id="overview" class="slide level2 smaller">
<h2>Overview</h2>
<p>Last time we took RNAseq data from an <em>in vitro</em> differentiation timecourse from mouse ESCs to glutaminergic neurons <a href="10.12688/f1000research.2-35.v1">(Hubbard et al, F1000 Research (2013))</a>. We took in transcript-level quantifications produced by <code>salmon</code> and collapsed them to gene-level quantifications using <code>tximport</code>. We then inspected the quality of the data by relating distances between samples using two methods: hierarchical clustering and principal components analysis. We found that the data was of high quality, as evidenced by the fact that replicates from a given timepoint were highly similar to other replicates for the same timepoint, and the distances between samples made sense with what we know about how the experiment was conducted.</p>
<p>Today, we are going to pretend that this isn’t a timecourse. For the sake of simplicity, we are going to imagine that we have only two conditions: DIV0 and DIV7. We will use <code>DESeq2</code> to identify genes that are differentially expressed between these two timepoints. We will then plot changes in expression for both individual genes and groups of genes that we already know going in might be interesting to look at. Finally, we will look at some features of transcripts and genes that are differentially expressed between these two timepoints.</p>
</section>
<section id="prepare-t2g" class="slide level2 smaller">
<h2>Prepare <code>t2g</code></h2>
<p>The first thing we need to do is read in the data again and move from transcript-level expression values to gene-level expression values with <code>tximport</code>. Let’s use <code>biomaRt</code> to get a table that relates gene and transcript IDs.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a></a>mart <span class="ot">&lt;-</span> biomaRt<span class="sc">::</span><span class="fu">useMart</span>(</span>
<span id="cb1-2"><a></a>  <span class="st">"ENSEMBL_MART_ENSEMBL"</span>,</span>
<span id="cb1-3"><a></a>  <span class="at">dataset =</span> <span class="st">"mmusculus_gene_ensembl"</span></span>
<span id="cb1-4"><a></a>)</span>
<span id="cb1-5"><a></a></span>
<span id="cb1-6"><a></a>t2g <span class="ot">&lt;-</span> biomaRt<span class="sc">::</span><span class="fu">getBM</span>(</span>
<span id="cb1-7"><a></a>  <span class="at">attributes =</span> <span class="fu">c</span>(</span>
<span id="cb1-8"><a></a>    <span class="st">"ensembl_transcript_id"</span>,</span>
<span id="cb1-9"><a></a>    <span class="st">"ensembl_gene_id"</span>,</span>
<span id="cb1-10"><a></a>    <span class="st">"external_gene_name"</span></span>
<span id="cb1-11"><a></a>  ),</span>
<span id="cb1-12"><a></a>  <span class="at">mart =</span> mart</span>
<span id="cb1-13"><a></a>) <span class="sc">|&gt;</span></span>
<span id="cb1-14"><a></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb1-15"><a></a></span>
<span id="cb1-16"><a></a><span class="co"># maps systematic to common gene names</span></span>
<span id="cb1-17"><a></a>gene_name_map <span class="ot">&lt;-</span> t2g <span class="sc">|&gt;</span></span>
<span id="cb1-18"><a></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>ensembl_transcript_id) <span class="sc">|&gt;</span></span>
<span id="cb1-19"><a></a>  <span class="fu">unique</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="prepare-metdata-and-import" class="slide level2 smaller">
<h2>Prepare <code>metdata</code> and import</h2>
<p>Now we can read in the transcript-level data and collapse to gene-level data with <code>tximport</code></p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a></a>metadata <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb2-2"><a></a>  <span class="at">salmon_dirs =</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(</span>
<span id="cb2-3"><a></a>    <span class="fu">here</span>(<span class="st">"data/block-rna/differentiation_salmonouts"</span>),</span>
<span id="cb2-4"><a></a>    <span class="at">recurse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-5"><a></a>    <span class="at">glob =</span> <span class="st">"*quant.sf"</span></span>
<span id="cb2-6"><a></a>  )</span>
<span id="cb2-7"><a></a>) <span class="sc">|&gt;</span></span>
<span id="cb2-8"><a></a>  <span class="fu">mutate</span>(</span>
<span id="cb2-9"><a></a>    <span class="at">sample_id =</span> fs<span class="sc">::</span><span class="fu">path_file</span>(fs<span class="sc">::</span><span class="fu">path_dir</span>(salmon_dirs))</span>
<span id="cb2-10"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2-11"><a></a>  <span class="fu">filter</span>(<span class="fu">str_starts</span>(sample_id, <span class="st">"DIV"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb2-12"><a></a>  <span class="fu">separate_wider_delim</span>(</span>
<span id="cb2-13"><a></a>    <span class="at">col =</span> sample_id,</span>
<span id="cb2-14"><a></a>    <span class="at">delim =</span> <span class="st">"."</span>,</span>
<span id="cb2-15"><a></a>    <span class="at">names =</span> <span class="fu">c</span>(<span class="st">"timepoint"</span>, <span class="st">"rep"</span>),</span>
<span id="cb2-16"><a></a>    <span class="at">too_few =</span> <span class="st">"align_start"</span>,</span>
<span id="cb2-17"><a></a>    <span class="at">cols_remove =</span> <span class="cn">FALSE</span></span>
<span id="cb2-18"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2-19"><a></a>  <span class="fu">mutate</span>(<span class="at">rep =</span> <span class="fu">str_remove</span>(rep, <span class="st">"Rep"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb2-20"><a></a>  <span class="fu">column_to_rownames</span>(<span class="st">"sample_id"</span>)</span>
<span id="cb2-21"><a></a></span>
<span id="cb2-22"><a></a><span class="co"># Add the sample_id column back from rownames</span></span>
<span id="cb2-23"><a></a>metadata<span class="sc">$</span>sample_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(metadata)</span>
<span id="cb2-24"><a></a></span>
<span id="cb2-25"><a></a>metadata <span class="ot">&lt;-</span> metadata <span class="sc">|&gt;</span></span>
<span id="cb2-26"><a></a>  <span class="fu">filter</span>(</span>
<span id="cb2-27"><a></a>    timepoint <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"DIV0"</span>, <span class="st">"DIV7"</span>)</span>
<span id="cb2-28"><a></a>  )</span>
<span id="cb2-29"><a></a></span>
<span id="cb2-30"><a></a>salmdir <span class="ot">&lt;-</span> metadata<span class="sc">$</span>salmon_dirs</span>
<span id="cb2-31"><a></a><span class="fu">names</span>(salmdir) <span class="ot">&lt;-</span> metadata<span class="sc">$</span>sample_id</span>
<span id="cb2-32"><a></a></span>
<span id="cb2-33"><a></a>txi <span class="ot">&lt;-</span> <span class="fu">tximport</span>(</span>
<span id="cb2-34"><a></a>  <span class="at">files =</span> salmdir,</span>
<span id="cb2-35"><a></a>  <span class="at">type =</span> <span class="st">"salmon"</span>,</span>
<span id="cb2-36"><a></a>  <span class="at">tx2gene =</span> t2g,</span>
<span id="cb2-37"><a></a>  <span class="at">dropInfReps =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-38"><a></a>  <span class="at">countsFromAbundance =</span> <span class="st">"lengthScaledTPM"</span></span>
<span id="cb2-39"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>reading in files with read_tsv</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>1 2 3 4 5 6 7 
transcripts missing from tx2gene: 4395
summarizing abundance
summarizing counts
summarizing length</code></pre>
</div>
</div>
</section>
<section id="filter-lowly-expressed-genes" class="slide level2">
<h2>Filter lowly expressed genes</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a></a><span class="co"># examine distribution of TPMs</span></span>
<span id="cb5-2"><a></a><span class="fu">hist</span>(<span class="fu">log2</span>(<span class="dv">1</span> <span class="sc">+</span> <span class="fu">rowSums</span>(txi<span class="sc">$</span>abundance)), <span class="at">breaks =</span> <span class="dv">40</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>

<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a></a><span class="co"># decide a cutoff</span></span>
<span id="cb6-2"><a></a>keepG <span class="ot">&lt;-</span> txi<span class="sc">$</span>abundance[<span class="fu">log2</span>(<span class="dv">1</span> <span class="sc">+</span> <span class="fu">rowSums</span>(txi<span class="sc">$</span>abundance)) <span class="sc">&gt;</span> <span class="fl">4.5</span>, ] <span class="sc">|&gt;</span></span>
<span id="cb6-3"><a></a>  <span class="fu">rownames</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<img data-src="slides-24_files/figure-revealjs/remove-low-genes-1.png" width="960" class="r-stretch"></section>
<section id="create-deseq-object" class="slide level2 smaller">
<h2>Create DESeq object</h2>
<p>There are essentially two steps to using <code>DESeq2</code>. The first involves creating a <code>DESeqDataSet</code> from your data. Luckily, if you have a <code>tximport</code> object, which we do in the form of <code>txi</code>, then this becomes easy.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a></a>ddsTxi <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(</span>
<span id="cb7-2"><a></a>  txi,</span>
<span id="cb7-3"><a></a>  <span class="at">colData =</span> metadata,</span>
<span id="cb7-4"><a></a>  <span class="at">design =</span> <span class="sc">~</span>timepoint</span>
<span id="cb7-5"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in DESeqDataSet(se, design = design, ignoreRank):
some variables in design formula are characters, converting
to factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using just counts from tximport</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a></a><span class="co"># keep genes with sufficient expession</span></span>
<span id="cb10-2"><a></a>ddsTxi <span class="ot">&lt;-</span> ddsTxi[keepG, ]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="design-formula" class="slide level2 smaller">
<h2>Design formula</h2>
<p>You can see that <code>DESeqDataSetFromTximport</code> wants three things. The first is our <code>tximport</code> object. The second is the dataframe we made that relates samples and conditions (or in this case timepoints). The last is something called a <strong>design formula</strong>. A design formula contains all of the variables that will go into <code>DESeq2</code>’s model. The formula starts with a tilde and then has variables separated by a plus sign think <code>lm()</code>. It is common practice, and in fact basically required with <code>DESeq2</code>, to put the variable of interest last. In our case, that’s trivial because we only have one: timepoint. So our design formula is very simple:</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a></a>design <span class="ot">=</span> <span class="er">~</span> timepoint</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<p>Your design formula should ideally include <strong>all of the sources of variation in your data</strong>. For example, let’s say that here we thought there was a batch effect with the replicates. Maybe all of the Rep1 samples were prepped and sequenced on a different day than the Rep2 samples and so on. We could potentially account for this in <code>DESeq2</code>’s model with the following forumula:</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a></a>design <span class="ot">=</span> <span class="er">~</span> rep <span class="sc">+</span> timepoint</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<p>Here, timepoint is still the variable of interest, but we are controlling for differences that arise due to differences in replicates.</p>
</section>
<section id="run-deseq2" class="slide level2 smaller">
<h2>Run <code>DESeq2</code></h2>
<p>We can see here that <code>DESeq2</code> is taking the counts produced by <code>tximport</code> for gene quantifications. There are 52346 genes (rows) here and 7 samples (columns). Now using this ddsTxi object, we can run <code>DESeq2</code>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a></a><span class="co"># create DESeq object</span></span>
<span id="cb13-2"><a></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(ddsTxi)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
</div>
<p>There are many useful things in this <code>dds</code> object. Take a look at the <a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">vignette</a> for a full explanation. Including info on many more tests and analyses that can be done with <code>DESeq2</code>.</p>
<p>The results can be accessed using the <code>results()</code> function. We will use the <code>contrast</code> argument here. <code>DESeq2</code> reports changes in RNA abundance between two samples as a <code>log2FoldChange</code>. But, it’s often not clear what the numerator and denominator of that fold change ratio…it could be either DIV7/DIV0 or DIV0/DIV7.</p>
<p>The lexographically first condition will be the numerator. I find it easier to explicitly specify what the numerator and denominator of this ratio are using the <code>contrast</code> argument. The <code>contrast</code> argument can be used to implement more complicated design formula. Remember our design formula that accounted for potential differences due to Replicate batch effects:</p>
<pre><code>~ replicate + timepoint</code></pre>
<p><code>DESeq2</code> will account for differences between replicates here to find differences between timepoints.</p>
</section>
<section id="contrasts-to-get-results" class="slide level2 smaller">
<h2>Contrasts to get results</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a></a><span class="co"># For contrast, we give three strings: the factor we are interested in, the numerator, and the denominator</span></span>
<span id="cb21-2"><a></a><span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"timepoint"</span>, <span class="st">"DIV7"</span>, <span class="st">"DIV0"</span>))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): timepoint DIV7 vs DIV0 
Wald test p-value: timepoint DIV7 vs DIV0 
DataFrame with 13191 rows and 6 columns
                     baseMean log2FoldChange     lfcSE
                    &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;
ENSMUSG00000000001   5616.419      -0.950902 0.0533784
ENSMUSG00000000028    723.202      -2.551463 0.1076496
ENSMUSG00000000031   3652.172       4.121041 0.2921400
ENSMUSG00000000037    244.242      -1.272542 0.2049733
ENSMUSG00000000056   2391.087       0.789258 0.1470825
...                       ...            ...       ...
ENSMUSG00000144223  510.92457      -0.595178 0.1209084
ENSMUSG00000144232   78.69414       0.383407 0.2425846
ENSMUSG00000144287   41.62399      -0.630864 0.3691944
ENSMUSG00002076020    2.70261       1.186678 1.1743062
ENSMUSG00002076083 1281.79528       0.893079 0.0600122
                        stat       pvalue         padj
                   &lt;numeric&gt;    &lt;numeric&gt;    &lt;numeric&gt;
ENSMUSG00000000001 -17.81435  5.46892e-71  2.59106e-70
ENSMUSG00000000028 -23.70156 3.47448e-124 2.91715e-123
ENSMUSG00000000031  14.10639  3.46880e-45  1.15568e-44
ENSMUSG00000000037  -6.20833  5.35506e-10  8.80823e-10
ENSMUSG00000000056   5.36609  8.04610e-08  1.23520e-07
...                      ...          ...          ...
ENSMUSG00000144223  -4.92256  8.54203e-07  1.26880e-06
ENSMUSG00000144232   1.58051  1.13990e-01  1.29827e-01
ENSMUSG00000144287  -1.70876  8.74958e-02  1.00669e-01
ENSMUSG00002076020   1.01054  3.12239e-01  3.39245e-01
ENSMUSG00002076083  14.88162  4.33859e-50  1.54540e-49</code></pre>
</div>
</div>
<p>The columns we are most interested in are <strong>log2FoldChange</strong> and <strong>padj</strong>.</p>
<p>log2FoldChange is self-explanatory. padj is the Benjamini-Hochberg corrected pvalue for a test asking if the expression of this gene is different between the two conditions.</p>
</section>
<section id="cleanup-results" class="slide level2 smaller">
<h2>Cleanup results</h2>
<p>Let’s do a little work on this data frame to make it slightly cleaner and more informative.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a></a>diff <span class="ot">&lt;-</span></span>
<span id="cb23-2"><a></a>  <span class="fu">results</span>(</span>
<span id="cb23-3"><a></a>    dds,</span>
<span id="cb23-4"><a></a>    <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"timepoint"</span>, <span class="st">"DIV7"</span>, <span class="st">"DIV0"</span>)</span>
<span id="cb23-5"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb23-6"><a></a>  <span class="co"># Change this into a dataframe</span></span>
<span id="cb23-7"><a></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb23-8"><a></a>  <span class="co"># Move ensembl gene IDs into their own column</span></span>
<span id="cb23-9"><a></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"ensembl_gene_id"</span>) <span class="sc">|&gt;</span></span>
<span id="cb23-10"><a></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb23-11"><a></a>  <span class="co"># drop unused columns</span></span>
<span id="cb23-12"><a></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(baseMean, lfcSE, stat, pvalue)) <span class="sc">|&gt;</span></span>
<span id="cb23-13"><a></a>  <span class="co"># Merge this with a table relating ensembl_gene_id with gene short names</span></span>
<span id="cb23-14"><a></a>  <span class="fu">inner_join</span>(gene_name_map) <span class="sc">|&gt;</span></span>
<span id="cb23-15"><a></a>  <span class="co"># Rename external_gene_name column</span></span>
<span id="cb23-16"><a></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> external_gene_name)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(ensembl_gene_id)`</code></pre>
</div>
</div>
</section>
<section id="how-many-are-significant" class="slide level2 smaller">
<h2>How many are significant</h2>
<p>OK now we have a table of gene expression results. How many genes are significantly up/down regulated between these two timepoints? We will use 0.01 as an FDR (p.adj) cutoff.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a></a><span class="co"># number of upregulated genes</span></span>
<span id="cb25-2"><a></a><span class="fu">nrow</span>(<span class="fu">filter</span>(diff, padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 5294</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a></a><span class="co"># number of downregulated genes</span></span>
<span id="cb27-2"><a></a><span class="fu">nrow</span>(<span class="fu">filter</span>(diff, padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span>))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 5325</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-of-differential-expression-results" class="slide level2 smaller">
<h2>Volcano plot of differential expression results</h2>
<p>Let’s make a volcano plot of these results.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a></a><span class="co"># meets the FDR cutoff</span></span>
<span id="cb29-2"><a></a>diff_sig <span class="ot">&lt;-</span></span>
<span id="cb29-3"><a></a>  <span class="fu">mutate</span>(</span>
<span id="cb29-4"><a></a>    diff,</span>
<span id="cb29-5"><a></a>    <span class="at">sig =</span> <span class="fu">case_when</span>(</span>
<span id="cb29-6"><a></a>      padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"yes"</span>,</span>
<span id="cb29-7"><a></a>      <span class="at">.default =</span> <span class="st">"no"</span></span>
<span id="cb29-8"><a></a>    )</span>
<span id="cb29-9"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb29-10"><a></a>  <span class="co"># if a gene did not meet expression cutoffs that DESeq2 automatically does, it gets a pvalue of NA</span></span>
<span id="cb29-11"><a></a>  <span class="fu">drop_na</span>()</span>
<span id="cb29-12"><a></a></span>
<span id="cb29-13"><a></a><span class="fu">ggplot</span>(</span>
<span id="cb29-14"><a></a>  diff_sig,</span>
<span id="cb29-15"><a></a>  <span class="fu">aes</span>(</span>
<span id="cb29-16"><a></a>    <span class="at">x =</span> log2FoldChange,</span>
<span id="cb29-17"><a></a>    <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj),</span>
<span id="cb29-18"><a></a>    <span class="at">color =</span> sig</span>
<span id="cb29-19"><a></a>  )</span>
<span id="cb29-20"><a></a>) <span class="sc">+</span></span>
<span id="cb29-21"><a></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb29-22"><a></a>  <span class="fu">labs</span>(</span>
<span id="cb29-23"><a></a>    <span class="at">x =</span> <span class="st">"DIV7 expression / DIV0 expression, log2"</span>,</span>
<span id="cb29-24"><a></a>    <span class="at">y =</span> <span class="st">"-log10(FDR)"</span></span>
<span id="cb29-25"><a></a>  ) <span class="sc">+</span></span>
<span id="cb29-26"><a></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb29-27"><a></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"red"</span>),</span>
<span id="cb29-28"><a></a>    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"NS"</span>, <span class="st">"FDR &lt; 0.01"</span>),</span>
<span id="cb29-29"><a></a>    <span class="at">name =</span> <span class="st">""</span></span>
<span id="cb29-30"><a></a>  ) <span class="sc">+</span></span>
<span id="cb29-31"><a></a>  <span class="fu">theme_cowplot</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="volcano-plot-of-differential-expression-results-output" class="slide level2 smaller output-location-slide"><h2>Volcano plot of differential expression results</h2><div class="cell output-location-slide">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-24_files/figure-revealjs/volcano-1-1.png" alt="Description of the plot - PLEASE FILL IN" width="960"></p>
</figure>
</div>
</div>
</div></section><section id="change-the-lfc-threshold" class="slide level2 smaller">
<h2>Change the LFC threshold</h2>
<p>In addition to an FDR cutoff, let’s also apply a log2FoldChange cutoff. This will of course be more conservative, but will probably give you a more confident set of genes.</p>
<div class="cell columns column-output-location">
<div class="column">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a></a><span class="co"># Is the expression of the gene at least 3-fold different?</span></span>
<span id="cb30-2"><a></a>diff_lfc <span class="ot">&lt;-</span></span>
<span id="cb30-3"><a></a>  <span class="fu">results</span>(</span>
<span id="cb30-4"><a></a>    dds,</span>
<span id="cb30-5"><a></a>    <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"timepoint"</span>, <span class="st">"DIV7"</span>, <span class="st">"DIV0"</span>),</span>
<span id="cb30-6"><a></a>    <span class="at">lfcThreshold =</span> <span class="fu">log</span>(<span class="dv">3</span>, <span class="dv">2</span>)</span>
<span id="cb30-7"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb30-8"><a></a>  <span class="co"># Change this into a dataframe</span></span>
<span id="cb30-9"><a></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb30-10"><a></a>  <span class="co"># Move ensembl gene IDs into their own column</span></span>
<span id="cb30-11"><a></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"ensembl_gene_id"</span>) <span class="sc">|&gt;</span></span>
<span id="cb30-12"><a></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb30-13"><a></a>  <span class="co"># drop unused columns</span></span>
<span id="cb30-14"><a></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(baseMean, lfcSE, stat, pvalue)) <span class="sc">|&gt;</span></span>
<span id="cb30-15"><a></a>  <span class="co"># Merge this with a table relating ensembl_gene_id with gene short names</span></span>
<span id="cb30-16"><a></a>  <span class="fu">inner_join</span>(gene_name_map) <span class="sc">|&gt;</span></span>
<span id="cb30-17"><a></a>  <span class="co"># Rename external_gene_name column</span></span>
<span id="cb30-18"><a></a>  <span class="fu">rename</span>(<span class="at">gene =</span> external_gene_name)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div><div class="column fragment">
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(ensembl_gene_id)`</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a></a><span class="co"># number of upregulated genes</span></span>
<span id="cb32-2"><a></a><span class="fu">nrow</span>(</span>
<span id="cb32-3"><a></a>  <span class="fu">filter</span>(</span>
<span id="cb32-4"><a></a>    diff_lfc,</span>
<span id="cb32-5"><a></a>    padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb32-6"><a></a>  )</span>
<span id="cb32-7"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1507</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a></a><span class="co"># number of downregulated genes</span></span>
<span id="cb34-2"><a></a><span class="fu">nrow</span>(</span>
<span id="cb34-3"><a></a>  <span class="fu">filter</span>(</span>
<span id="cb34-4"><a></a>    diff_lfc,</span>
<span id="cb34-5"><a></a>    padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span></span>
<span id="cb34-6"><a></a>  )</span>
<span id="cb34-7"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 990</code></pre>
</div>
</div></div>
</section>
<section id="change-the-lfc-threshold-1" class="slide level2 smaller">
<h2>Change the LFC threshold</h2>
<div class="cell columns column-output-location">
<div class="column">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a></a>diff_lfc_sig <span class="ot">&lt;-</span></span>
<span id="cb36-2"><a></a>  <span class="fu">mutate</span>(</span>
<span id="cb36-3"><a></a>    diff_lfc,</span>
<span id="cb36-4"><a></a>    <span class="at">sig =</span> <span class="fu">case_when</span>(</span>
<span id="cb36-5"><a></a>      padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"yes"</span>,</span>
<span id="cb36-6"><a></a>      <span class="at">.default =</span> <span class="st">"no"</span></span>
<span id="cb36-7"><a></a>    )</span>
<span id="cb36-8"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb36-9"><a></a>  <span class="fu">drop_na</span>()</span>
<span id="cb36-10"><a></a></span>
<span id="cb36-11"><a></a></span>
<span id="cb36-12"><a></a><span class="co"># look at some specific genes</span></span>
<span id="cb36-13"><a></a></span>
<span id="cb36-14"><a></a>diff_lfc_sig <span class="sc">|&gt;</span></span>
<span id="cb36-15"><a></a>  <span class="fu">filter</span>(</span>
<span id="cb36-16"><a></a>    gene <span class="sc">%in%</span></span>
<span id="cb36-17"><a></a>      <span class="fu">c</span>(<span class="st">"Bdnf"</span>, <span class="st">"Dlg4"</span>, <span class="st">"Klf4"</span>, <span class="st">"Sox2"</span>)</span>
<span id="cb36-18"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb36-19"><a></a>  <span class="fu">gt</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div><div class="column fragment">
<div class="cell-output-display">
<div id="ealwbssqpp" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#ealwbssqpp table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#ealwbssqpp thead, #ealwbssqpp tbody, #ealwbssqpp tfoot, #ealwbssqpp tr, #ealwbssqpp td, #ealwbssqpp th {
  border-style: none;
}

#ealwbssqpp p {
  margin: 0;
  padding: 0;
}

#ealwbssqpp .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#ealwbssqpp .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#ealwbssqpp .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#ealwbssqpp .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#ealwbssqpp .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ealwbssqpp .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ealwbssqpp .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ealwbssqpp .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#ealwbssqpp .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#ealwbssqpp .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#ealwbssqpp .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#ealwbssqpp .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#ealwbssqpp .gt_spanner_row {
  border-bottom-style: hidden;
}

#ealwbssqpp .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#ealwbssqpp .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#ealwbssqpp .gt_from_md > :first-child {
  margin-top: 0;
}

#ealwbssqpp .gt_from_md > :last-child {
  margin-bottom: 0;
}

#ealwbssqpp .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#ealwbssqpp .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#ealwbssqpp .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#ealwbssqpp .gt_row_group_first td {
  border-top-width: 2px;
}

#ealwbssqpp .gt_row_group_first th {
  border-top-width: 2px;
}

#ealwbssqpp .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ealwbssqpp .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#ealwbssqpp .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#ealwbssqpp .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ealwbssqpp .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ealwbssqpp .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#ealwbssqpp .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#ealwbssqpp .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#ealwbssqpp .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ealwbssqpp .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ealwbssqpp .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ealwbssqpp .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ealwbssqpp .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ealwbssqpp .gt_left {
  text-align: left;
}

#ealwbssqpp .gt_center {
  text-align: center;
}

#ealwbssqpp .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#ealwbssqpp .gt_font_normal {
  font-weight: normal;
}

#ealwbssqpp .gt_font_bold {
  font-weight: bold;
}

#ealwbssqpp .gt_font_italic {
  font-style: italic;
}

#ealwbssqpp .gt_super {
  font-size: 65%;
}

#ealwbssqpp .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#ealwbssqpp .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#ealwbssqpp .gt_indent_1 {
  text-indent: 5px;
}

#ealwbssqpp .gt_indent_2 {
  text-indent: 10px;
}

#ealwbssqpp .gt_indent_3 {
  text-indent: 15px;
}

#ealwbssqpp .gt_indent_4 {
  text-indent: 20px;
}

#ealwbssqpp .gt_indent_5 {
  text-indent: 25px;
}

#ealwbssqpp .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#ealwbssqpp div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="ensembl_gene_id" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ensembl_gene_id</th>
<th id="log2FoldChange" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">log2FoldChange</th>
<th id="padj" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">padj</th>
<th id="gene" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">gene</th>
<th id="sig" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">sig</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000003032</td>
<td class="gt_row gt_right" headers="log2FoldChange">-2.344141</td>
<td class="gt_row gt_right" headers="padj">4.559786e-09</td>
<td class="gt_row gt_left" headers="gene">Klf4</td>
<td class="gt_row gt_left" headers="sig">yes</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000020886</td>
<td class="gt_row gt_right" headers="log2FoldChange">3.339265</td>
<td class="gt_row gt_right" headers="padj">2.017800e-45</td>
<td class="gt_row gt_left" headers="gene">Dlg4</td>
<td class="gt_row gt_left" headers="sig">yes</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000048482</td>
<td class="gt_row gt_right" headers="log2FoldChange">1.710410</td>
<td class="gt_row gt_right" headers="padj">4.871606e-01</td>
<td class="gt_row gt_left" headers="gene">Bdnf</td>
<td class="gt_row gt_left" headers="sig">no</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000074637</td>
<td class="gt_row gt_right" headers="log2FoldChange">-2.169572</td>
<td class="gt_row gt_right" headers="padj">4.986817e-13</td>
<td class="gt_row gt_left" headers="gene">Sox2</td>
<td class="gt_row gt_left" headers="sig">yes</td>
</tr>
</tbody>
</table>

</div>
</div>
</div></div>
</section>
<section id="filtered-volcano" class="slide level2 smaller">
<h2>Filtered Volcano</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb37"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a></a><span class="fu">ggplot</span>(</span>
<span id="cb37-2"><a></a>  diff_lfc_sig,</span>
<span id="cb37-3"><a></a>  <span class="fu">aes</span>(</span>
<span id="cb37-4"><a></a>    <span class="at">x =</span> log2FoldChange,</span>
<span id="cb37-5"><a></a>    <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj),</span>
<span id="cb37-6"><a></a>    <span class="at">color =</span> sig</span>
<span id="cb37-7"><a></a>  )</span>
<span id="cb37-8"><a></a>) <span class="sc">+</span></span>
<span id="cb37-9"><a></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb37-10"><a></a>  <span class="fu">labs</span>(</span>
<span id="cb37-11"><a></a>    <span class="at">x =</span> <span class="st">"DIV7 expression / DIV0 expression, log2"</span>,</span>
<span id="cb37-12"><a></a>    <span class="at">y =</span> <span class="st">"-log10(FDR)"</span></span>
<span id="cb37-13"><a></a>  ) <span class="sc">+</span></span>
<span id="cb37-14"><a></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb37-15"><a></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"red"</span>),</span>
<span id="cb37-16"><a></a>    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"NS"</span>, <span class="st">"FDR &lt; 0.01"</span>),</span>
<span id="cb37-17"><a></a>    <span class="at">name =</span> <span class="st">""</span></span>
<span id="cb37-18"><a></a>  ) <span class="sc">+</span></span>
<span id="cb37-19"><a></a>  <span class="fu">theme_cowplot</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="filtered-volcano-output" class="slide level2 smaller output-location-slide"><h2>Filtered Volcano</h2><div class="cell output-location-slide">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-24_files/figure-revealjs/plot-2-1.png" alt="Description of the plot - PLEASE FILL IN" width="960"></p>
</figure>
</div>
</div>
</div></section><section id="plotting-the-expression-of-single-genes" class="slide level2 smaller">
<h2>Plotting the expression of single genes</h2>
<p>Sometimes we will have particular marker genes that we might want to highlight to give confidence that the experiment worked as expected. We can plot the expression of these genes in each replicate. Let’s plot the expression of two pluripotency genes (which we expect to decrease) and two neuronal genes (which we expect to increase).</p>
<p>So what is the value that we would plot? We could use the ‘normalized counts’ value provided by <code>DESeq2</code>. However, remember there is not length calculation so it is difficult to compare accross genes.</p>
<p>A more interpretable value to plot might be TPM, since TPM is length-normalized. Let’s say a gene was expressed at 500 TPM. Right off the bat, I know generally what kind of expression that reflects (pretty high).</p>
</section>
<section id="get-tpms" class="slide level2 smaller">
<h2>Get TPMs</h2>
<p>Let’s plot the expression of Klf4, Sox2, Bdnf, and Dlg4 in our samples.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb38"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a></a>tpms <span class="ot">&lt;-</span> txi<span class="sc">$</span>abundance <span class="sc">|&gt;</span></span>
<span id="cb38-2"><a></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb38-3"><a></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"ensembl_gene_id"</span>) <span class="sc">|&gt;</span></span>
<span id="cb38-4"><a></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb38-5"><a></a>  <span class="fu">inner_join</span>(gene_name_map) <span class="sc">|&gt;</span></span>
<span id="cb38-6"><a></a>  <span class="fu">rename</span>(<span class="at">gene =</span> external_gene_name) <span class="sc">|&gt;</span></span>
<span id="cb38-7"><a></a>  <span class="co"># Filter for genes we are interested in</span></span>
<span id="cb38-8"><a></a>  <span class="fu">filter</span>(gene <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Klf4"</span>, <span class="st">"Sox2"</span>, <span class="st">"Bdnf"</span>, <span class="st">"Dlg4"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb38-9"><a></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(ensembl_gene_id, gene)) <span class="sc">|&gt;</span></span>
<span id="cb38-10"><a></a>  <span class="fu">separate_wider_delim</span>(</span>
<span id="cb38-11"><a></a>    <span class="at">col =</span> name,</span>
<span id="cb38-12"><a></a>    <span class="at">delim =</span> <span class="st">"."</span>,</span>
<span id="cb38-13"><a></a>    <span class="at">names =</span> <span class="fu">c</span>(<span class="st">"condition"</span>, <span class="st">"rep"</span>)</span>
<span id="cb38-14"><a></a>  )</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="get-tpms-output" class="slide level2 smaller output-location-slide"><h2>Get TPMs</h2><div class="cell output-location-slide">
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(ensembl_gene_id)`</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb40"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a></a><span class="fu">gt</span>(tpms)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="yyqhrrlrtg" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#yyqhrrlrtg table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#yyqhrrlrtg thead, #yyqhrrlrtg tbody, #yyqhrrlrtg tfoot, #yyqhrrlrtg tr, #yyqhrrlrtg td, #yyqhrrlrtg th {
  border-style: none;
}

#yyqhrrlrtg p {
  margin: 0;
  padding: 0;
}

#yyqhrrlrtg .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#yyqhrrlrtg .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#yyqhrrlrtg .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#yyqhrrlrtg .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#yyqhrrlrtg .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#yyqhrrlrtg .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#yyqhrrlrtg .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#yyqhrrlrtg .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#yyqhrrlrtg .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#yyqhrrlrtg .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#yyqhrrlrtg .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#yyqhrrlrtg .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#yyqhrrlrtg .gt_spanner_row {
  border-bottom-style: hidden;
}

#yyqhrrlrtg .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#yyqhrrlrtg .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#yyqhrrlrtg .gt_from_md > :first-child {
  margin-top: 0;
}

#yyqhrrlrtg .gt_from_md > :last-child {
  margin-bottom: 0;
}

#yyqhrrlrtg .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#yyqhrrlrtg .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#yyqhrrlrtg .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#yyqhrrlrtg .gt_row_group_first td {
  border-top-width: 2px;
}

#yyqhrrlrtg .gt_row_group_first th {
  border-top-width: 2px;
}

#yyqhrrlrtg .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#yyqhrrlrtg .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#yyqhrrlrtg .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#yyqhrrlrtg .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#yyqhrrlrtg .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#yyqhrrlrtg .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#yyqhrrlrtg .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#yyqhrrlrtg .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#yyqhrrlrtg .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#yyqhrrlrtg .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#yyqhrrlrtg .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#yyqhrrlrtg .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#yyqhrrlrtg .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#yyqhrrlrtg .gt_left {
  text-align: left;
}

#yyqhrrlrtg .gt_center {
  text-align: center;
}

#yyqhrrlrtg .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#yyqhrrlrtg .gt_font_normal {
  font-weight: normal;
}

#yyqhrrlrtg .gt_font_bold {
  font-weight: bold;
}

#yyqhrrlrtg .gt_font_italic {
  font-style: italic;
}

#yyqhrrlrtg .gt_super {
  font-size: 65%;
}

#yyqhrrlrtg .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#yyqhrrlrtg .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#yyqhrrlrtg .gt_indent_1 {
  text-indent: 5px;
}

#yyqhrrlrtg .gt_indent_2 {
  text-indent: 10px;
}

#yyqhrrlrtg .gt_indent_3 {
  text-indent: 15px;
}

#yyqhrrlrtg .gt_indent_4 {
  text-indent: 20px;
}

#yyqhrrlrtg .gt_indent_5 {
  text-indent: 25px;
}

#yyqhrrlrtg .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#yyqhrrlrtg div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="ensembl_gene_id" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ensembl_gene_id</th>
<th id="gene" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">gene</th>
<th id="condition" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">condition</th>
<th id="rep" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">rep</th>
<th id="value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">value</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000003032</td>
<td class="gt_row gt_left" headers="gene">Klf4</td>
<td class="gt_row gt_left" headers="condition">DIV0</td>
<td class="gt_row gt_left" headers="rep">Rep1</td>
<td class="gt_row gt_right" headers="value">40.022629</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000003032</td>
<td class="gt_row gt_left" headers="gene">Klf4</td>
<td class="gt_row gt_left" headers="condition">DIV0</td>
<td class="gt_row gt_left" headers="rep">Rep2</td>
<td class="gt_row gt_right" headers="value">46.241797</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000003032</td>
<td class="gt_row gt_left" headers="gene">Klf4</td>
<td class="gt_row gt_left" headers="condition">DIV0</td>
<td class="gt_row gt_left" headers="rep">Rep3</td>
<td class="gt_row gt_right" headers="value">57.374566</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000003032</td>
<td class="gt_row gt_left" headers="gene">Klf4</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep1</td>
<td class="gt_row gt_right" headers="value">9.063035</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000003032</td>
<td class="gt_row gt_left" headers="gene">Klf4</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep2</td>
<td class="gt_row gt_right" headers="value">9.208364</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000003032</td>
<td class="gt_row gt_left" headers="gene">Klf4</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep3</td>
<td class="gt_row gt_right" headers="value">9.715119</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000003032</td>
<td class="gt_row gt_left" headers="gene">Klf4</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep4</td>
<td class="gt_row gt_right" headers="value">8.817037</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000020886</td>
<td class="gt_row gt_left" headers="gene">Dlg4</td>
<td class="gt_row gt_left" headers="condition">DIV0</td>
<td class="gt_row gt_left" headers="rep">Rep1</td>
<td class="gt_row gt_right" headers="value">18.752921</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000020886</td>
<td class="gt_row gt_left" headers="gene">Dlg4</td>
<td class="gt_row gt_left" headers="condition">DIV0</td>
<td class="gt_row gt_left" headers="rep">Rep2</td>
<td class="gt_row gt_right" headers="value">15.749749</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000020886</td>
<td class="gt_row gt_left" headers="gene">Dlg4</td>
<td class="gt_row gt_left" headers="condition">DIV0</td>
<td class="gt_row gt_left" headers="rep">Rep3</td>
<td class="gt_row gt_right" headers="value">12.212636</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000020886</td>
<td class="gt_row gt_left" headers="gene">Dlg4</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep1</td>
<td class="gt_row gt_right" headers="value">151.558403</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000020886</td>
<td class="gt_row gt_left" headers="gene">Dlg4</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep2</td>
<td class="gt_row gt_right" headers="value">156.043098</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000020886</td>
<td class="gt_row gt_left" headers="gene">Dlg4</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep3</td>
<td class="gt_row gt_right" headers="value">153.269475</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000020886</td>
<td class="gt_row gt_left" headers="gene">Dlg4</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep4</td>
<td class="gt_row gt_right" headers="value">153.538436</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000048482</td>
<td class="gt_row gt_left" headers="gene">Bdnf</td>
<td class="gt_row gt_left" headers="condition">DIV0</td>
<td class="gt_row gt_left" headers="rep">Rep1</td>
<td class="gt_row gt_right" headers="value">2.456452</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000048482</td>
<td class="gt_row gt_left" headers="gene">Bdnf</td>
<td class="gt_row gt_left" headers="condition">DIV0</td>
<td class="gt_row gt_left" headers="rep">Rep2</td>
<td class="gt_row gt_right" headers="value">2.401157</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000048482</td>
<td class="gt_row gt_left" headers="gene">Bdnf</td>
<td class="gt_row gt_left" headers="condition">DIV0</td>
<td class="gt_row gt_left" headers="rep">Rep3</td>
<td class="gt_row gt_right" headers="value">2.351342</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000048482</td>
<td class="gt_row gt_left" headers="gene">Bdnf</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep1</td>
<td class="gt_row gt_right" headers="value">7.799981</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000048482</td>
<td class="gt_row gt_left" headers="gene">Bdnf</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep2</td>
<td class="gt_row gt_right" headers="value">7.762255</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000048482</td>
<td class="gt_row gt_left" headers="gene">Bdnf</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep3</td>
<td class="gt_row gt_right" headers="value">7.638515</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000048482</td>
<td class="gt_row gt_left" headers="gene">Bdnf</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep4</td>
<td class="gt_row gt_right" headers="value">7.433529</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000074637</td>
<td class="gt_row gt_left" headers="gene">Sox2</td>
<td class="gt_row gt_left" headers="condition">DIV0</td>
<td class="gt_row gt_left" headers="rep">Rep1</td>
<td class="gt_row gt_right" headers="value">131.032132</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000074637</td>
<td class="gt_row gt_left" headers="gene">Sox2</td>
<td class="gt_row gt_left" headers="condition">DIV0</td>
<td class="gt_row gt_left" headers="rep">Rep2</td>
<td class="gt_row gt_right" headers="value">120.433905</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000074637</td>
<td class="gt_row gt_left" headers="gene">Sox2</td>
<td class="gt_row gt_left" headers="condition">DIV0</td>
<td class="gt_row gt_left" headers="rep">Rep3</td>
<td class="gt_row gt_right" headers="value">110.664468</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000074637</td>
<td class="gt_row gt_left" headers="gene">Sox2</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep1</td>
<td class="gt_row gt_right" headers="value">24.484942</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000074637</td>
<td class="gt_row gt_left" headers="gene">Sox2</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep2</td>
<td class="gt_row gt_right" headers="value">26.080394</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000074637</td>
<td class="gt_row gt_left" headers="gene">Sox2</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep3</td>
<td class="gt_row gt_right" headers="value">27.106054</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000074637</td>
<td class="gt_row gt_left" headers="gene">Sox2</td>
<td class="gt_row gt_left" headers="condition">DIV7</td>
<td class="gt_row gt_left" headers="rep">Rep4</td>
<td class="gt_row gt_right" headers="value">26.995461</td>
</tr>
</tbody>
</table>

</div>
</div>
</div></section><section id="now-plot" class="slide level2 smaller">
<h2>Now plot</h2>
<div class="cell columns column-output-location">
<div class="column">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb41"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a></a><span class="fu">ggplot</span>(</span>
<span id="cb41-2"><a></a>  tpms,</span>
<span id="cb41-3"><a></a>  <span class="fu">aes</span>(</span>
<span id="cb41-4"><a></a>    <span class="at">x =</span> condition,</span>
<span id="cb41-5"><a></a>    <span class="at">y =</span> value,</span>
<span id="cb41-6"><a></a>    <span class="at">color =</span> condition</span>
<span id="cb41-7"><a></a>  )</span>
<span id="cb41-8"><a></a>) <span class="sc">+</span></span>
<span id="cb41-9"><a></a>  <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="dv">2</span>, <span class="at">width =</span> .<span class="dv">25</span>) <span class="sc">+</span></span>
<span id="cb41-10"><a></a>  <span class="fu">labs</span>(</span>
<span id="cb41-11"><a></a>    <span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb41-12"><a></a>    <span class="at">y =</span> <span class="st">"TPM"</span></span>
<span id="cb41-13"><a></a>  ) <span class="sc">+</span></span>
<span id="cb41-14"><a></a>  <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb41-15"><a></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb41-16"><a></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene, <span class="at">scales =</span> <span class="st">"free_y"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div><div class="column fragment">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-24_files/figure-revealjs/plot-tpms-1.png" alt="Description of the plot - PLEASE FILL IN" width="960"></p>
</figure>
</div>
</div>
</div></div>
</section>
<section id="how-about-pathways" class="slide level2 smaller">
<h2>How about pathways?</h2>
<p>Say that instead of plotting individual genes we wanted to ask whether a whole class of genes are going up or down. We can do that by retrieving all genes that belong to a particular gene ontology term.</p>
<p>There are three classes of genes we will look at here:</p>
<ul>
<li class="fragment">Maintenance of pluripotency (GO:0019827)</li>
<li class="fragment">Positive regulation of the cell cycle (GO:0045787)</li>
<li class="fragment">Neuronal differentitaion (GO:0030182)</li>
</ul>
</section>
<section id="retrieve-pathway-information" class="slide level2 smaller">
<h2>Retrieve pathway information</h2>
<p>We can use <code>biomaRt</code> to get all genes that belong to each of these categories. Think of it like doing a gene ontology enrichment analysis in reverse.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb42"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a></a>pluripotencygenes <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb42-2"><a></a>  <span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>),</span>
<span id="cb42-3"><a></a>  <span class="at">filters =</span> <span class="fu">c</span>(<span class="st">"go_parent_term"</span>),</span>
<span id="cb42-4"><a></a>  <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"GO:0019827"</span>),</span>
<span id="cb42-5"><a></a>  <span class="at">mart =</span> mart</span>
<span id="cb42-6"><a></a>)</span>
<span id="cb42-7"><a></a></span>
<span id="cb42-8"><a></a>cellcyclegenes <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb42-9"><a></a>  <span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>),</span>
<span id="cb42-10"><a></a>  <span class="at">filters =</span> <span class="fu">c</span>(<span class="st">"go_parent_term"</span>),</span>
<span id="cb42-11"><a></a>  <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"GO:0045787"</span>),</span>
<span id="cb42-12"><a></a>  <span class="at">mart =</span> mart</span>
<span id="cb42-13"><a></a>)</span>
<span id="cb42-14"><a></a></span>
<span id="cb42-15"><a></a>neurongenes <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb42-16"><a></a>  <span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>),</span>
<span id="cb42-17"><a></a>  <span class="at">filters =</span> <span class="fu">c</span>(<span class="st">"go_parent_term"</span>),</span>
<span id="cb42-18"><a></a>  <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"GO:0030182"</span>),</span>
<span id="cb42-19"><a></a>  <span class="at">mart =</span> mart</span>
<span id="cb42-20"><a></a>)</span>
<span id="cb42-21"><a></a></span>
<span id="cb42-22"><a></a></span>
<span id="cb42-23"><a></a><span class="co"># pathway &lt;- bind_rows(pluripotencygenes,</span></span>
<span id="cb42-24"><a></a><span class="co">#           cellcyclegenes,</span></span>
<span id="cb42-25"><a></a><span class="co">#           neurongenes</span></span>
<span id="cb42-26"><a></a><span class="co">#           )</span></span>
<span id="cb42-27"><a></a><span class="co">#</span></span>
<span id="cb42-28"><a></a><span class="co"># pathway$path &lt;- c(</span></span>
<span id="cb42-29"><a></a><span class="co">#   rep("pluri",nrow(pluripotencygenes)),</span></span>
<span id="cb42-30"><a></a><span class="co">#   rep("cellcycle",nrow(cellcyclegenes)),</span></span>
<span id="cb42-31"><a></a><span class="co">#   rep("neuron",nrow(neurongenes))</span></span>
<span id="cb42-32"><a></a><span class="co">#                   )</span></span>
<span id="cb42-33"><a></a><span class="co">#</span></span>
<span id="cb42-34"><a></a><span class="co"># write_csv(x = pathway, file = here("data","block-rna","pathwaygenes.csv.gz"))</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="add-pathway-information-to-results" class="slide level2 smaller">
<h2>Add pathway information to results</h2>
<p>You can see that these items are one-column dataframes that have the column name ‘ensembl_gene_id’. We can now go through our results dataframe and add an annotation column that marks whether the gene is in any of these categories.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb43"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a></a>diff_paths <span class="ot">&lt;-</span></span>
<span id="cb43-2"><a></a>  diff_lfc <span class="sc">|&gt;</span></span>
<span id="cb43-3"><a></a>  <span class="fu">mutate</span>(</span>
<span id="cb43-4"><a></a>    <span class="at">annot =</span> <span class="fu">case_when</span>(</span>
<span id="cb43-5"><a></a>      ensembl_gene_id <span class="sc">%in%</span> pluripotencygenes<span class="sc">$</span>ensembl_gene_id <span class="sc">~</span> <span class="st">"pluripotency"</span>,</span>
<span id="cb43-6"><a></a>      ensembl_gene_id <span class="sc">%in%</span> cellcyclegenes<span class="sc">$</span>ensembl_gene_id <span class="sc">~</span> <span class="st">"cellcycle"</span>,</span>
<span id="cb43-7"><a></a>      ensembl_gene_id <span class="sc">%in%</span> neurongenes<span class="sc">$</span>ensembl_gene_id <span class="sc">~</span> <span class="st">"neurondiff"</span>,</span>
<span id="cb43-8"><a></a>      <span class="at">.default =</span> <span class="st">"none"</span></span>
<span id="cb43-9"><a></a>    ),</span>
<span id="cb43-10"><a></a>    <span class="co"># Reorder these for plotting purposes</span></span>
<span id="cb43-11"><a></a>    <span class="at">annot =</span> <span class="fu">factor</span>(</span>
<span id="cb43-12"><a></a>      annot,</span>
<span id="cb43-13"><a></a>      <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"none"</span>, <span class="st">"cellcycle"</span>, <span class="st">"pluripotency"</span>, <span class="st">"neurondiff"</span>)</span>
<span id="cb43-14"><a></a>    )</span>
<span id="cb43-15"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb43-16"><a></a>  <span class="fu">drop_na</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="are-there-significant-differences" class="slide level2 smaller">
<h2>Are there significant differences?</h2>
<p>OK we’ve got our table, now we are going to ask if the log2FoldChange values for the genes in each of these classes are different that what we would expect. So what is the expected value? Well, we have a distribution of log2 fold changes for all the genes that are <strong>not</strong> in any of these categories. So we will ask if the distribution of log2 fold changes for each gene category is different than that null distribution.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb44"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a></a>pvals <span class="ot">&lt;-</span> rstatix<span class="sc">::</span><span class="fu">wilcox_test</span>(</span>
<span id="cb44-2"><a></a>  <span class="at">data =</span> diff_paths,</span>
<span id="cb44-3"><a></a>  log2FoldChange <span class="sc">~</span> annot,</span>
<span id="cb44-4"><a></a>  <span class="at">ref.group =</span> <span class="st">"none"</span></span>
<span id="cb44-5"><a></a>)</span>
<span id="cb44-6"><a></a></span>
<span id="cb44-7"><a></a>p.pluripotency <span class="ot">&lt;-</span> pvals <span class="sc">|&gt;</span></span>
<span id="cb44-8"><a></a>  <span class="fu">filter</span>(group2 <span class="sc">==</span> <span class="st">"pluripotency"</span>) <span class="sc">|&gt;</span></span>
<span id="cb44-9"><a></a>  <span class="fu">pull</span>(p.adj)</span>
<span id="cb44-10"><a></a></span>
<span id="cb44-11"><a></a>p.cellcycle <span class="ot">&lt;-</span> pvals <span class="sc">|&gt;</span></span>
<span id="cb44-12"><a></a>  <span class="fu">filter</span>(group2 <span class="sc">==</span> <span class="st">"cellcycle"</span>) <span class="sc">|&gt;</span></span>
<span id="cb44-13"><a></a>  <span class="fu">pull</span>(p.adj)</span>
<span id="cb44-14"><a></a></span>
<span id="cb44-15"><a></a>p.neurondiff <span class="ot">&lt;-</span> pvals <span class="sc">|&gt;</span></span>
<span id="cb44-16"><a></a>  <span class="fu">filter</span>(group2 <span class="sc">==</span> <span class="st">"neurondiff"</span>) <span class="sc">|&gt;</span></span>
<span id="cb44-17"><a></a>  <span class="fu">pull</span>(p.adj)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="plot-pathway-differences" class="slide level2 smaller">
<h2>plot pathway differences</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb45"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a></a><span class="fu">ggplot</span>(</span>
<span id="cb45-2"><a></a>  diff_paths,</span>
<span id="cb45-3"><a></a>  <span class="fu">aes</span>(</span>
<span id="cb45-4"><a></a>    <span class="at">x =</span> annot,</span>
<span id="cb45-5"><a></a>    <span class="at">y =</span> log2FoldChange,</span>
<span id="cb45-6"><a></a>    <span class="at">fill =</span> annot</span>
<span id="cb45-7"><a></a>  )</span>
<span id="cb45-8"><a></a>) <span class="sc">+</span></span>
<span id="cb45-9"><a></a>  <span class="fu">labs</span>(</span>
<span id="cb45-10"><a></a>    <span class="at">x =</span> <span class="st">"Gene class"</span>,</span>
<span id="cb45-11"><a></a>    <span class="at">y =</span> <span class="st">"DIV7/DIV0, log2"</span></span>
<span id="cb45-12"><a></a>  ) <span class="sc">+</span></span>
<span id="cb45-13"><a></a>  <span class="fu">geom_hline</span>(</span>
<span id="cb45-14"><a></a>    <span class="at">yintercept =</span> <span class="dv">0</span>,</span>
<span id="cb45-15"><a></a>    <span class="at">color =</span> <span class="st">"gray"</span>,</span>
<span id="cb45-16"><a></a>    <span class="at">linetype =</span> <span class="st">"dashed"</span></span>
<span id="cb45-17"><a></a>  ) <span class="sc">+</span></span>
<span id="cb45-18"><a></a>  <span class="fu">geom_boxplot</span>(</span>
<span id="cb45-19"><a></a>    <span class="at">notch =</span> <span class="cn">TRUE</span>,</span>
<span id="cb45-20"><a></a>    <span class="at">outlier.shape =</span> <span class="cn">NA</span></span>
<span id="cb45-21"><a></a>  ) <span class="sc">+</span></span>
<span id="cb45-22"><a></a>  <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb45-23"><a></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"gray"</span>, <span class="st">"red"</span>, <span class="st">"blue"</span>, <span class="st">"purple"</span>), <span class="at">guide =</span> F) <span class="sc">+</span></span>
<span id="cb45-24"><a></a>  <span class="fu">scale_x_discrete</span>(</span>
<span id="cb45-25"><a></a>    <span class="at">labels =</span> <span class="fu">c</span>(</span>
<span id="cb45-26"><a></a>      <span class="st">"none"</span>,</span>
<span id="cb45-27"><a></a>      <span class="st">"Cell cycle"</span>,</span>
<span id="cb45-28"><a></a>      <span class="st">"Pluripotency"</span>,</span>
<span id="cb45-29"><a></a>      <span class="st">"Neuron</span><span class="sc">\n</span><span class="st">differentiation"</span></span>
<span id="cb45-30"><a></a>    )</span>
<span id="cb45-31"><a></a>  ) <span class="sc">+</span></span>
<span id="cb45-32"><a></a>  <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">7</span>) <span class="sc">+</span></span>
<span id="cb45-33"><a></a>  <span class="co"># hacky significance bars</span></span>
<span id="cb45-34"><a></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>, <span class="at">y =</span> <span class="dv">4</span>, <span class="at">yend =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb45-35"><a></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">3</span>, <span class="at">y =</span> <span class="dv">5</span>, <span class="at">yend =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb45-36"><a></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">4</span>, <span class="at">y =</span> <span class="dv">6</span>, <span class="at">yend =</span> <span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb45-37"><a></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>, <span class="at">y =</span> <span class="fl">4.4</span>, <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"p = "</span>, p.cellcycle)) <span class="sc">+</span></span>
<span id="cb45-38"><a></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">2</span>, <span class="at">y =</span> <span class="fl">5.4</span>, <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"p = "</span>, p.pluripotency)) <span class="sc">+</span></span>
<span id="cb45-39"><a></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">2.5</span>, <span class="at">y =</span> <span class="fl">6.4</span>, <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"p = "</span>, p.neurondiff))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="plot-pathway-differences-output" class="slide level2 smaller output-location-slide"><h2>plot pathway differences</h2><div class="cell output-location-slide">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-24_files/figure-revealjs/plot-diff-results-1.png" alt="Description of the plot - PLEASE FILL IN" width="960"></p>
</figure>
</div>
</div>
</div></section><section id="what-if-we-want-to-look-at-pathways-in-an-unbiased-way" class="slide level2 smaller">
<h2>What if we want to look at pathways in an unbiased way?</h2>
<p>We will use Gene Set Enrichment Analysis (GSEA) to determine if pre-defined gene sets (pathways, GO terms, experimentally defined genes) are coordinately up-regulated or down-regulated between the two conditions you are comparing. To run gsea you need 2 things. 1. You list of expressed genes ranked by fold change. 2. Pre-defined gene sets. See <a href="https://www.gsea-msigdb.org/gsea/msigdb/index.jsp">MSigDb</a></p>

<img data-src="../img/block-rna/gsea_overview.png" class="r-stretch"><p>PMID: 12808457, 16199517</p>
</section>
<section id="gsea-examples" class="slide level2 smaller">
<h2>GSEA examples</h2>
<p>Top = upregulated</p>

<img data-src="../img/block-rna/gsea_examples.png" style="width:3in" class="r-stretch"><p>Bottom = downregulated</p>
</section>
<section id="prep-gsea" class="slide level2 smaller">
<h2>Prep GSEA</h2>
<ol type="1">
<li class="fragment"><p>We need to make a list of all genes and their LFC.</p></li>
<li class="fragment"><p>We need to find interesting gene sets.</p></li>
<li class="fragment"><p>Run GSEA</p></li>
</ol>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb46"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a></a><span class="co"># retrieve hallmark gene set from msigdb</span></span>
<span id="cb46-2"><a></a>mouse_hallmark <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">"Mus musculus"</span>) <span class="sc">|&gt;</span></span>
<span id="cb46-3"><a></a>  <span class="fu">filter</span>(gs_collection <span class="sc">==</span> <span class="st">"H"</span>) <span class="sc">|&gt;</span></span>
<span id="cb46-4"><a></a>  <span class="fu">select</span>(gs_name, gene_symbol)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="prep-gsea-output" class="slide level2 smaller output-location-slide"><h2>Prep GSEA</h2><div class="cell output-location-slide">
<div class="cell-output cell-output-stderr">
<pre><code>Using human MSigDB with ortholog mapping to mouse. Use `db_species = "MM"` for mouse-native gene sets.
This message is displayed once per session.</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb48"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a></a><span class="co"># create a list of gene LFCs</span></span>
<span id="cb48-2"><a></a>rankedgenes <span class="ot">&lt;-</span> diff_lfc <span class="sc">|&gt;</span> <span class="fu">pull</span>(log2FoldChange)</span>
<span id="cb48-3"><a></a></span>
<span id="cb48-4"><a></a><span class="co"># add symbols as names of the list</span></span>
<span id="cb48-5"><a></a><span class="fu">names</span>(rankedgenes) <span class="ot">&lt;-</span> diff<span class="sc">$</span>gene</span>
<span id="cb48-6"><a></a></span>
<span id="cb48-7"><a></a><span class="co"># sort by LFC</span></span>
<span id="cb48-8"><a></a>rankedgenes <span class="ot">&lt;-</span> <span class="fu">sort</span>(rankedgenes, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb48-9"><a></a></span>
<span id="cb48-10"><a></a><span class="co"># deduplicate</span></span>
<span id="cb48-11"><a></a>rankedgenes <span class="ot">&lt;-</span> rankedgenes[<span class="sc">!</span><span class="fu">duplicated</span>(<span class="fu">names</span>(rankedgenes))]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div></section><section id="run-gsea" class="slide level2 smaller">
<h2>Run GSEA</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb49"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a></a><span class="co"># run gsea</span></span>
<span id="cb49-2"><a></a>div7vs0 <span class="ot">&lt;-</span> <span class="fu">GSEA</span>(</span>
<span id="cb49-3"><a></a>  <span class="at">geneList =</span> rankedgenes,</span>
<span id="cb49-4"><a></a>  <span class="at">eps =</span> <span class="dv">0</span>,</span>
<span id="cb49-5"><a></a>  <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>,</span>
<span id="cb49-6"><a></a>  <span class="at">pvalueCutoff =</span> .<span class="dv">05</span>,</span>
<span id="cb49-7"><a></a>  <span class="at">minGSSize =</span> <span class="dv">20</span>,</span>
<span id="cb49-8"><a></a>  <span class="at">maxGSSize =</span> <span class="dv">1000</span>,</span>
<span id="cb49-9"><a></a>  <span class="at">TERM2GENE =</span> mouse_hallmark</span>
<span id="cb49-10"><a></a>)</span>
<span id="cb49-11"><a></a></span>
<span id="cb49-12"><a></a>div7vs0<span class="sc">@</span>result <span class="sc">|&gt;</span></span>
<span id="cb49-13"><a></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(ID, NES, p.adjust) <span class="sc">|&gt;</span></span>
<span id="cb49-14"><a></a>  <span class="fu">gt</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="run-gsea-output" class="slide level2 smaller output-location-slide"><h2>Run GSEA</h2><div class="cell output-location-slide">

</div></section><section id="plot-gsea" class="slide level2 smaller">
<h2>Plot GSEA</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb50"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a></a><span class="co"># plot "HALLMARK_G2M_CHECKPOINT"</span></span>
<span id="cb50-2"><a></a><span class="fu">gseaplot</span>(<span class="at">x =</span> div7vs0, <span class="at">geneSetID =</span> <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>


</section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">
<p><a href="https://rnabioco.github.io/molb-7950">Course 🔗</a></p>
</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': true,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"8\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdn.jsdelivr.net/npm/mathjax@2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
          const outerScaffold = trigger.parentElement.cloneNode(true);
          const codeEl = outerScaffold.querySelector('code');
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp("https:\/\/rnabioco\.github\.io\/molb-7950\/");
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>