<!DOCTYPE html>
<html lang="en"><head>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/tabby.min.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-8f4e3aabaa80e3e650ae20f2c24f966b.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.8.24">

  <meta name="author" content="Neel Mukherjee">
  <meta name="dcterms.date" content="2025-09-29">
  <title>MOLB 7950 – RNAseq QC</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    html { -webkit-text-size-adjust: 100%; }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #97947a;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #97947a;  padding-left: 4px; }
    div.sourceCode
      { color: #f8f8f2; background-color: #2b2b2b; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #f8f8f2; } /* Normal */
    code span.al { color: #dcc6e0; } /* Alert */
    code span.an { color: #d4d0ab; } /* Annotation */
    code span.at { color: #ffd700; } /* Attribute */
    code span.bn { color: #dcc6e0; } /* BaseN */
    code span.bu { color: #f5ab35; } /* BuiltIn */
    code span.cf { color: #ffa07a; } /* ControlFlow */
    code span.ch { color: #abe338; } /* Char */
    code span.cn { color: #ffa07a; } /* Constant */
    code span.co { color: #d4d0ab; } /* Comment */
    code span.cv { color: #d4d0ab; font-style: italic; } /* CommentVar */
    code span.do { color: #d4d0ab; font-style: italic; } /* Documentation */
    code span.dt { color: #dcc6e0; } /* DataType */
    code span.dv { color: #dcc6e0; } /* DecVal */
    code span.er { color: #dcc6e0; } /* Error */
    code span.ex { color: #ffd700; } /* Extension */
    code span.fl { color: #f5ab35; } /* Float */
    code span.fu { color: #ffd700; } /* Function */
    code span.im { } /* Import */
    code span.in { color: #d4d0ab; } /* Information */
    code span.kw { color: #ffa07a; } /* Keyword */
    code span.op { color: #00e0e0; } /* Operator */
    code span.ot { color: #ffa07a; } /* Other */
    code span.pp { color: #dcc6e0; } /* Preprocessor */
    code span.sc { color: #00e0e0; } /* SpecialChar */
    code span.ss { color: #abe338; } /* SpecialString */
    code span.st { color: #abe338; } /* String */
    code span.va { color: #f5ab35; } /* Variable */
    code span.vs { color: #abe338; } /* VerbatimString */
    code span.wa { color: #d4d0ab; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="../site_libs/revealjs/dist/theme/quarto-8fafb62820c35118272e29c1027a293f.css">
  <link href="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
<meta property="og:title" content="RNAseq QC – MOLB 7950">
<meta property="og:site_name" content="MOLB 7950">
<meta name="twitter:title" content="RNAseq QC – MOLB 7950">
<meta name="twitter:image" content="https://rnabioco.github.io/molb-7950/slides/images/twitter-card.png">
<meta name="twitter:creator" content="@rnabioco">
<meta name="twitter:card" content="summary_large_image">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">RNAseq QC</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Neel Mukherjee 
</div>
        <p class="quarto-title-affiliation">
            RNA Bioscience Initiative | CU Anschutz
          </p>
    </div>
</div>

  <p class="date">2025-09-29</p>
</section>
<section id="learning-objectives" class="slide level2 smaller">
<h2>Learning Objectives</h2>
<p>By the end of the class, you should be able to:</p>
<ul>
<li class="fragment"><p>Understand how <code>salmon</code> can be used to quantify gene expression given an RNAseq dataset</p></li>
<li class="fragment"><p>Collapse transcript-level quantifications to gene level quantifications using <code>tximport</code></p></li>
<li class="fragment"><p>Deduce the relationships of samples to each other using hierarchical clustering and PCA</p></li>
</ul>
</section>
<section id="rigor-reproducibility" class="slide level2 smaller">
<h2>Rigor &amp; Reproducibility</h2>
<p>As with all computational <strong>experiments</strong> (yes, they are experiments, don’t let your pipette-toting friends tell you otherwise), keeping track of what you did is key. In the old days, I kept a written notebook of commands that I ran. Sounds silly, but there were many times that I went back to that notebook to see exactly what the parameters were for a given run using a piece of software.</p>
<p>Today, there are better options. You are using one of the better ones right now. Notebooks, including RMarkdown (mainly for R) and Jupyter (mainly for Python), are a great way to keep track of what you did as well as give justification or explanation for your analyses using plain ’ol English.</p>
<p>Trust me, one day you will be glad you used them. The Methods section of your paper is never fun to write without them.</p>
</section>
<section id="overview" class="slide level2 smaller">
<h2>Overview</h2>
<p>In this class, we will examine RNAseq data collected over a time-course of differentiation from mouse embryonic stem cells to cortical glutamatergic neurons <a href="10.12688/f1000research.2-35.v1">(Hubbard et al, F1000 Research (2013))</a>. In this publication, the authors differentiated mESCs to neurons using a series of <em>in vitro</em> culture steps over a period of 37 days. During this timecourse, samples were extracted at selected intervals for transcriptome analysis. Importantly, for each timepoint, either 3 or 4 samples were taken for RNA extraction, library preparation and sequencing. This allows us to efficiently use the statistical frameworks provided by the <a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">DESeq2</a> package to identify genes whose RNA expression changes across the timecourse.</p>

<img data-src="../img/block-rna/RNAseqpaper.png" class="r-stretch"></section>
<section id="experiment-design" class="slide level2 smaller">
<h2>Experiment design</h2>
<p>Cells were grown in generic differentiation-promoting media (LIF-) for 8 days until aggreates were dissociated and replated in neuronal differentiation media. This day of replating was designated as <em>in vitro</em> day 0 (DIV0). The timepoints taken before this replating therefore happened at “negative” times (DIV-8 and DIV-4). Because naming files with dashes or minus signs can cause problems, these samples are referred to as DIVminus8 and DIVminus4. Following the replating, samples were taken at days 1, 7, 16, 21, and 28 (DIV1, DIV7, DIV16, DIV21, and DIV28).</p>
</section>
<section id="qc-overview" class="slide level2 smaller">
<h2>QC overview</h2>
<p>Today we will focus on some quality control steps that are good ideas to do for every RNAseq dataset you encounter, whether produced by yourself or someone else.</p>

<img data-src="../img/block-rna/flowchart.png" class="r-stretch"></section>
<section id="goal" class="slide level2 smaller">
<h2>Goal</h2>
<p>Today we will focus on some <strong>Quality Control</strong> steps that are good ideas to do for every RNA-seq data set you encounter, whether produced by yourself or someone else.</p>

<img data-src="../img/block-rna/flowchart_RNASeqQC.png" class="r-stretch"></section>
<section id="quantifying-transcript-expression-with-salmon" class="slide level2 smaller">
<h2>Quantifying transcript expression with <code>salmon</code></h2>
<p>We recently learned about the RNAseq quantification tool <a href="https://combine-lab.github.io/salmon/">salmon</a>. We won’t rehash the details here about how salmon works. For our purposes, we just need to know that salmon reads in a fastq file of sequencing reads and a fasta file of transcript sequences to be quantified. Let’s take a look at this fasta file of transcripts:</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb1"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a></a><span class="fu">head</span> <span class="at">-n</span> 20 data/block-rna/gencodecomprehensive.vM17.allcdna_subset.fa</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>

<img data-src="../img/block-rna/fasta_cdna.png" class="r-stretch"><p>Looks like we have <a href="www.ensembl.org">ensembl</a> transcript IDs, which is a good idea. I can tell because they start with ‘ENS’. Using ensembl IDs as transcript names will allow us to later collate transcript expression levels into gene expression levels using a database that relates transcripts and genes. More on that later.</p>
</section>
<section id="making-a-transcriptome-index" class="slide level2 smaller">
<h2>Making a transcriptome index</h2>
<p>The first step in quantifying these transcripts is to make an index from them. This is done as follows:</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb2"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a></a><span class="ex">salmon</span> index <span class="at">-t</span> <span class="op">&lt;</span>transcripts.fa<span class="op">&gt;</span> -i <span class="op">&lt;</span>transcripts.idx<span class="op">&gt;</span> --type quasi <span class="at">-k</span> <span class="op">&lt;</span>k<span class="op">&gt;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<p>Here, <code>transcripts.fa</code> is a path to our fasta file, transcripts.idx is the name of the index that will be created, and k is the length of the kmers that will be used in the hash table related kmers and transcripts. k is the length of the minimum accepted match for a kmer in a read and a kmer in a transcript. Longer kmers (higher values of k) will therefore be more stringent, and lowering k may improve mapping sensitivity at the cost of some specificity. You may also see here how read lengths can influence what value for k you should choose.</p>
<p>Consider an experiment where we had 25 nt reads (this was true wayyyyyy back in the old, dark days of high-throughput sequencing). What’s going to happen if I quantify these reads using an index where the kmer size was set to 29? Well, nothing will align. The index has represented the transcriptome in 29 nt chunks. However, no read will match to these 29mers because there are no 29mers in these reads! <em>As a general rule of thumb, for reads 75 nt and longer (which is the bulk of the data produced nowadays), a good value for k that maximizes both specificity and sensitivity is 31</em>. However, datasets that you may retrieve from the internet, particularly older ones, may have shorter read lengths, so keep this is mind when defining k.</p>
</section>
<section id="quantifying-reads-against-this-index" class="slide level2 smaller">
<h2>Quantifying reads against this index</h2>
<p>Once we have our index, we can quantify transcripts in the index using reads from our fastq files.</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb3"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a></a><span class="ex">salmon</span> quant <span class="at">--libType</span> A <span class="at">-p</span> 8 <span class="at">--seqBias</span> <span class="at">--gcBias</span> <span class="at">--validateMappings</span> <span class="at">-1</span> <span class="op">&lt;</span>forwardreads.fastq<span class="op">&gt;</span> -2 <span class="op">&lt;</span>reversereads.fastq<span class="op">&gt;</span> -o <span class="op">&lt;</span>outputname<span class="op">&gt;</span> --index <span class="op">&lt;</span>transcripts.idx<span class="op">&gt;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<p>In this command, our forward and reverse read fastq files are supplied to -1 and -2, respectively. If the experiment produced single end reads, -2 is omitted. &lt;transcripts.idx&gt; is the path to the index produced in the previous step. I’m not going to go through the rest of the flags used here, but their meanings as well as other options can be found <a href="https://salmon.readthedocs.io/en/latest/salmon.html#using-salmon">here</a>.</p>
</section>
<section id="salmon-outputs" class="slide level2 smaller">
<h2>Salmon outputs</h2>
<p>Let’s take a look at what salmon spits out. The first file we will look at is a log that is found at <strong>/logs/salmon_quant.log</strong>. This file contains info about the quantification, but there’s one line of this file in particular that we are interested in. It lets us know how many of the reads in the fastq file that salmon found a home for in the transcriptome fasta.</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb4"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a></a><span class="fu">less</span> data/block-rna/differentiation_salmonouts/DIV0.Rep1/logs/salmon_quant.log</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<p>There are a lot of lines in this file, but really only one that we are interested in. We want the one that tells us the “Mapping rate.” How could we easily and efficiently look at the mapping rates of all our samples? <strong>Grep!</strong></p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb5"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a></a><span class="co">#Get the mapping rates for all samples</span></span>
<span id="cb5-2"><a></a><span class="co">#In each log file, the line that we are interested in contains the string 'Mapping ' (notice the space)</span></span>
<span id="cb5-3"><a></a><span class="fu">grep</span> <span class="st">'Mapping '</span> data/block-rna/differentiation_salmonouts/<span class="pp">*</span>/logs/salmon_quant.log</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</section>
<section id="moving-from-transcript-quantifications-to-gene-quantifications" class="slide level2 smaller">
<h2>Moving from transcript quantifications to gene quantifications</h2>
<p>As we discussed, salmon quantifies <em>transcripts</em>, not <em>genes</em>. However, genes are made up of transcripts, so we can calculate gene expression values from transcript expression values if we knew which transcripts belonged to which genes.</p>

<img data-src="../img/block-rna/txtogene.png" class="r-stretch"></section>
<section id="relating-genes-and-transcripts" class="slide level2 smaller">
<h2>Relating genes and transcripts</h2>
<p>We can get this relationships between <em>transcripts</em> and <em>genes</em> through <code>biomaRt</code>.</p>
<p><code>biomaRt</code> has many tables that relate genes, transcripts, and other useful data include gene biotypes and gene ontology categories, even across species. Let’s use it here to get a table of genes and transcripts for the mouse genome.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a></a><span class="co"># First we need to define a 'mart' to use.</span></span>
<span id="cb6-2"><a></a><span class="co"># There are a handful of them that</span></span>
<span id="cb6-3"><a></a><span class="co"># you can see here:</span></span>
<span id="cb6-4"><a></a><span class="fu">listMarts</span>(</span>
<span id="cb6-5"><a></a>  <span class="at">mart =</span> <span class="cn">NULL</span>,</span>
<span id="cb6-6"><a></a>  <span class="at">host =</span> <span class="st">"www.ensembl.org"</span></span>
<span id="cb6-7"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Ensembl will soon enforce the use of https.
Ensure the 'host' argument includes "https://"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>               biomart                version
1 ENSEMBL_MART_ENSEMBL      Ensembl Genes 115
2   ENSEMBL_MART_MOUSE      Mouse strains 115
3     ENSEMBL_MART_SNP  Ensembl Variation 115
4 ENSEMBL_MART_FUNCGEN Ensembl Regulation 115</code></pre>
</div>
</div>
<p>I encourage you to see what is in each mart, but for now we are only going to use ENSEMBL_MART_ENSEMBL.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a></a>mart <span class="ot">&lt;-</span> biomaRt<span class="sc">::</span><span class="fu">useMart</span>(</span>
<span id="cb9-2"><a></a>  <span class="st">"ENSEMBL_MART_ENSEMBL"</span>,</span>
<span id="cb9-3"><a></a>  <span class="at">host =</span> <span class="st">"https://www.ensembl.org"</span></span>
<span id="cb9-4"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="using-biomart" class="slide level2 smaller">
<h2>Using biomaRt</h2>
<p>Alright, we’ve chosen our mart. What data sets are available in this mart?</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a></a>datasets <span class="ot">&lt;-</span> <span class="fu">listDatasets</span>(mart)</span>
<span id="cb10-2"><a></a><span class="fu">gt</span>(datasets)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="uemfalbdqp" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#uemfalbdqp table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#uemfalbdqp thead, #uemfalbdqp tbody, #uemfalbdqp tfoot, #uemfalbdqp tr, #uemfalbdqp td, #uemfalbdqp th {
  border-style: none;
}

#uemfalbdqp p {
  margin: 0;
  padding: 0;
}

#uemfalbdqp .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#uemfalbdqp .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#uemfalbdqp .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#uemfalbdqp .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#uemfalbdqp .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#uemfalbdqp .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#uemfalbdqp .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#uemfalbdqp .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#uemfalbdqp .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#uemfalbdqp .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#uemfalbdqp .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#uemfalbdqp .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#uemfalbdqp .gt_spanner_row {
  border-bottom-style: hidden;
}

#uemfalbdqp .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#uemfalbdqp .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#uemfalbdqp .gt_from_md > :first-child {
  margin-top: 0;
}

#uemfalbdqp .gt_from_md > :last-child {
  margin-bottom: 0;
}

#uemfalbdqp .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#uemfalbdqp .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#uemfalbdqp .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#uemfalbdqp .gt_row_group_first td {
  border-top-width: 2px;
}

#uemfalbdqp .gt_row_group_first th {
  border-top-width: 2px;
}

#uemfalbdqp .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#uemfalbdqp .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#uemfalbdqp .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#uemfalbdqp .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#uemfalbdqp .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#uemfalbdqp .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#uemfalbdqp .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#uemfalbdqp .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#uemfalbdqp .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#uemfalbdqp .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#uemfalbdqp .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#uemfalbdqp .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#uemfalbdqp .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#uemfalbdqp .gt_left {
  text-align: left;
}

#uemfalbdqp .gt_center {
  text-align: center;
}

#uemfalbdqp .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#uemfalbdqp .gt_font_normal {
  font-weight: normal;
}

#uemfalbdqp .gt_font_bold {
  font-weight: bold;
}

#uemfalbdqp .gt_font_italic {
  font-style: italic;
}

#uemfalbdqp .gt_super {
  font-size: 65%;
}

#uemfalbdqp .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#uemfalbdqp .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#uemfalbdqp .gt_indent_1 {
  text-indent: 5px;
}

#uemfalbdqp .gt_indent_2 {
  text-indent: 10px;
}

#uemfalbdqp .gt_indent_3 {
  text-indent: 15px;
}

#uemfalbdqp .gt_indent_4 {
  text-indent: 20px;
}

#uemfalbdqp .gt_indent_5 {
  text-indent: 25px;
}

#uemfalbdqp .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#uemfalbdqp div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="dataset" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">dataset</th>
<th id="description" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">description</th>
<th id="version" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">version</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">abrachyrhynchus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pink-footed goose genes (ASM259213v1)</td>
<td class="gt_row gt_center" headers="version">ASM259213v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">acalliptera_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Eastern happy genes (fAstCal1.3)</td>
<td class="gt_row gt_center" headers="version">fAstCal1.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">acarolinensis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Green anole genes (AnoCar2.0v2)</td>
<td class="gt_row gt_center" headers="version">AnoCar2.0v2</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">acchrysaetos_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Golden eagle genes (bAquChr1.2)</td>
<td class="gt_row gt_center" headers="version">bAquChr1.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">acitrinellus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Midas cichlid genes (Midas_v5)</td>
<td class="gt_row gt_center" headers="version">Midas_v5</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">amelanoleuca_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Giant panda genes (ASM200744v2)</td>
<td class="gt_row gt_center" headers="version">ASM200744v2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">amexicanus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Mexican tetra genes (Astyanax_mexicanus-2.0)</td>
<td class="gt_row gt_center" headers="version">Astyanax_mexicanus-2.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">anancymaae_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Ma's night monkey genes (Anan_2.0)</td>
<td class="gt_row gt_center" headers="version">Anan_2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">aocellaris_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Clown anemonefish genes (ASM2253959v1)</td>
<td class="gt_row gt_center" headers="version">ASM2253959v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">apercula_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Orange clownfish genes (Nemo_v1)</td>
<td class="gt_row gt_center" headers="version">Nemo_v1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">aplatyrhynchos_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Mallard genes (ASM874695v1)</td>
<td class="gt_row gt_center" headers="version">ASM874695v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">apolyacanthus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Spiny chromis genes (ASM210954v1)</td>
<td class="gt_row gt_center" headers="version">ASM210954v1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">applatyrhynchos_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Duck genes (CAU_duck1.0)</td>
<td class="gt_row gt_center" headers="version">CAU_duck1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">atestudineus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Climbing perch genes (fAnaTes1.3)</td>
<td class="gt_row gt_center" headers="version">fAnaTes1.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">bbbison_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">American bison genes (Bison_UMD1.0)</td>
<td class="gt_row gt_center" headers="version">Bison_UMD1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">bgrunniens_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Domestic yak genes (LU_Bosgru_v3.0)</td>
<td class="gt_row gt_center" headers="version">LU_Bosgru_v3.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">bihybrid_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Hybrid - Bos Indicus genes (UOA_Brahman_1)</td>
<td class="gt_row gt_center" headers="version">UOA_Brahman_1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">bmusculus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Blue whale genes (mBalMus1.v2)</td>
<td class="gt_row gt_center" headers="version">mBalMus1.v2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">bmutus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Wild yak genes (BosGru_v2.0)</td>
<td class="gt_row gt_center" headers="version">BosGru_v2.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">bsplendens_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Siamese fighting fish genes (fBetSpl5.2)</td>
<td class="gt_row gt_center" headers="version">fBetSpl5.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">btaurus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Cattle genes (ARS-UCD2.0)</td>
<td class="gt_row gt_center" headers="version">ARS-UCD2.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">cabingdonii_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Abingdon island giant tortoise genes (ASM359739v1)</td>
<td class="gt_row gt_center" headers="version">ASM359739v1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">catys_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Sooty mangabey genes (Caty_1.0)</td>
<td class="gt_row gt_center" headers="version">Caty_1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">cauratus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Goldfish genes (ASM336829v1)</td>
<td class="gt_row gt_center" headers="version">ASM336829v1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">cccarpio_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Common carp genes (Cypcar_WagV4.0)</td>
<td class="gt_row gt_center" headers="version">Cypcar_WagV4.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">cdromedarius_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Arabian camel genes (CamDro2)</td>
<td class="gt_row gt_center" headers="version">CamDro2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">celegans_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Caenorhabditis elegans (Nematode, N2) genes (WBcel235)</td>
<td class="gt_row gt_center" headers="version">WBcel235</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">cgobio_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Channel bull blenny genes (fCotGob3.1)</td>
<td class="gt_row gt_center" headers="version">fCotGob3.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">cgpicr_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Chinese hamster PICR genes (CriGri-PICRH-1.0)</td>
<td class="gt_row gt_center" headers="version">CriGri-PICRH-1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">charengus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Atlantic herring genes (Ch_v2.0.2v2)</td>
<td class="gt_row gt_center" headers="version">Ch_v2.0.2v2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">chircus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Goat genes (ARS1)</td>
<td class="gt_row gt_center" headers="version">ARS1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">choffmanni_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Sloth genes (choHof1)</td>
<td class="gt_row gt_center" headers="version">choHof1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">chyarkandensis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Yarkand deer genes (CEY_v1)</td>
<td class="gt_row gt_center" headers="version">CEY_v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">cimitator_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Panamanian white-faced capuchin genes (Cebus_imitator-1.0)</td>
<td class="gt_row gt_center" headers="version">Cebus_imitator-1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">cintestinalis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">C.intestinalis genes (KH)</td>
<td class="gt_row gt_center" headers="version">KH</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">cjacchus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">White-tufted-ear marmoset genes (mCalJac1.pat.X)</td>
<td class="gt_row gt_center" headers="version">mCalJac1.pat.X</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">cjaponica_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Japanese quail genes (Coturnix_japonica_2.0)</td>
<td class="gt_row gt_center" headers="version">Coturnix_japonica_2.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">clanigera_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Long-tailed chinchilla genes (ChiLan1.0)</td>
<td class="gt_row gt_center" headers="version">ChiLan1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">cldingo_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Dingo genes (ASM325472v1)</td>
<td class="gt_row gt_center" headers="version">ASM325472v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">clfamiliaris_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Dog genes (ROS_Cfam_1.0)</td>
<td class="gt_row gt_center" headers="version">ROS_Cfam_1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">clumpus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Lumpfish genes (fCycLum1.pri)</td>
<td class="gt_row gt_center" headers="version">fCycLum1.pri</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">cmilii_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Elephant shark genes (Callorhinchus_milii-6.1.3)</td>
<td class="gt_row gt_center" headers="version">Callorhinchus_milii-6.1.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">cpbellii_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Painted turtle genes (Chrysemys_picta_bellii-3.0.3)</td>
<td class="gt_row gt_center" headers="version">Chrysemys_picta_bellii-3.0.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">cporcellus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Guinea Pig genes (Cavpor3.0)</td>
<td class="gt_row gt_center" headers="version">Cavpor3.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">cporosus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Australian saltwater crocodile genes (CroPor_comp1)</td>
<td class="gt_row gt_center" headers="version">CroPor_comp1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">csabaeus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Vervet-AGM genes (ChlSab1.1)</td>
<td class="gt_row gt_center" headers="version">ChlSab1.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">csavignyi_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">C.savignyi genes (CSAV 2.0)</td>
<td class="gt_row gt_center" headers="version">CSAV 2.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">csemilaevis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Tongue sole genes (Cse_v1.0)</td>
<td class="gt_row gt_center" headers="version">Cse_v1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">csyrichta_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Tarsier genes (Tarsius_syrichta-2.0.1)</td>
<td class="gt_row gt_center" headers="version">Tarsius_syrichta-2.0.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">cvariegatus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Sheepshead minnow genes (C_variegatus-1.0)</td>
<td class="gt_row gt_center" headers="version">C_variegatus-1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">cwagneri_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Chacoan peccary genes (CatWag_v2_BIUU_UCD)</td>
<td class="gt_row gt_center" headers="version">CatWag_v2_BIUU_UCD</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">dclupeoides_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Denticle herring genes (fDenClu1.2)</td>
<td class="gt_row gt_center" headers="version">fDenClu1.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">dlabrax_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">European seabass genes (dlabrax2021)</td>
<td class="gt_row gt_center" headers="version">dlabrax2021</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">dleucas_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Beluga whale genes (ASM228892v3)</td>
<td class="gt_row gt_center" headers="version">ASM228892v3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">dmelanogaster_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Drosophila melanogaster - (Fruit fly) genes (BDGP6.54)</td>
<td class="gt_row gt_center" headers="version">BDGP6.54</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">dnovemcinctus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Armadillo genes (Dasnov3.0)</td>
<td class="gt_row gt_center" headers="version">Dasnov3.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">dordii_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Kangaroo rat genes (Dord_2.0)</td>
<td class="gt_row gt_center" headers="version">Dord_2.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">drerio_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Zebrafish genes (GRCz11)</td>
<td class="gt_row gt_center" headers="version">GRCz11</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">easinus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Donkey genes (ASM1607732v2)</td>
<td class="gt_row gt_center" headers="version">ASM1607732v2</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">eburgeri_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Hagfish genes (Eburgeri_3.2)</td>
<td class="gt_row gt_center" headers="version">Eburgeri_3.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">ecaballus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Horse genes (EquCab3.0)</td>
<td class="gt_row gt_center" headers="version">EquCab3.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">ecalabaricus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Reedfish genes (fErpCal1.1)</td>
<td class="gt_row gt_center" headers="version">fErpCal1.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">eelectricus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Electric eel genes (fEleEle1.pri)</td>
<td class="gt_row gt_center" headers="version">fEleEle1.pri</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">eeuropaeus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Hedgehog genes (eriEur1)</td>
<td class="gt_row gt_center" headers="version">eriEur1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">elucius_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Northern pike genes (fEsoLuc1.pri)</td>
<td class="gt_row gt_center" headers="version">fEsoLuc1.pri</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">etelfairi_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Lesser hedgehog tenrec genes (TENREC)</td>
<td class="gt_row gt_center" headers="version">TENREC</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">falbicollis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Collared flycatcher genes (FicAlb1.5)</td>
<td class="gt_row gt_center" headers="version">FicAlb1.5</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">fcatus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Domestic cat genes (F.catus_Fca126_mat1.0)</td>
<td class="gt_row gt_center" headers="version">F.catus_Fca126_mat1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">fheteroclitus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Mummichog genes (Fundulus_heteroclitus-3.0.2)</td>
<td class="gt_row gt_center" headers="version">Fundulus_heteroclitus-3.0.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">gaculeatus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Stickleback genes (GAculeatus_UGA_version5)</td>
<td class="gt_row gt_center" headers="version">GAculeatus_UGA_version5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">gevgoodei_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Goodes thornscrub tortoise genes (rGopEvg1_v1.p)</td>
<td class="gt_row gt_center" headers="version">rGopEvg1_v1.p</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">gfortis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Medium ground-finch genes (GeoFor_1.0)</td>
<td class="gt_row gt_center" headers="version">GeoFor_1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">ggallus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Chicken genes (bGalGal1.mat.broiler.GRCg7b)</td>
<td class="gt_row gt_center" headers="version">bGalGal1.mat.broiler.GRCg7b</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">ggorilla_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Gorilla genes (gorGor4)</td>
<td class="gt_row gt_center" headers="version">gorGor4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">gmorhua_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Atlantic cod genes (gadMor3.0)</td>
<td class="gt_row gt_center" headers="version">gadMor3.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">hburtoni_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Burton's mouthbrooder genes (AstBur1.0)</td>
<td class="gt_row gt_center" headers="version">AstBur1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">hcomes_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Tiger tail seahorse genes (H_comes_QL1_v1)</td>
<td class="gt_row gt_center" headers="version">H_comes_QL1_v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">hgfemale_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Naked mole-rat female genes (Naked_mole-rat_maternal)</td>
<td class="gt_row gt_center" headers="version">Naked_mole-rat_maternal</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">hhucho_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Huchen genes (ASM331708v1)</td>
<td class="gt_row gt_center" headers="version">ASM331708v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">hsapiens_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Human genes (GRCh38.p14)</td>
<td class="gt_row gt_center" headers="version">GRCh38.p14</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">ipunctatus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Channel catfish genes (ASM400665v3)</td>
<td class="gt_row gt_center" headers="version">ASM400665v3</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">itridecemlineatus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Squirrel genes (SpeTri2.0)</td>
<td class="gt_row gt_center" headers="version">SpeTri2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">jjaculus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Lesser Egyptian jerboa genes (JacJac1.0)</td>
<td class="gt_row gt_center" headers="version">JacJac1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">kmarmoratus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Mangrove rivulus genes (ASM164957v1)</td>
<td class="gt_row gt_center" headers="version">ASM164957v1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">lafricana_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Elephant genes (Loxafr3.0)</td>
<td class="gt_row gt_center" headers="version">Loxafr3.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">lbergylta_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Ballan wrasse genes (BallGen_V1)</td>
<td class="gt_row gt_center" headers="version">BallGen_V1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">lcalcarifer_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Barramundi perch genes (ASB_HGAPassembly_v1)</td>
<td class="gt_row gt_center" headers="version">ASB_HGAPassembly_v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">lchalumnae_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Coelacanth genes (LatCha1)</td>
<td class="gt_row gt_center" headers="version">LatCha1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">lcrocea_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Large yellow croaker genes (L_crocea_2.0)</td>
<td class="gt_row gt_center" headers="version">L_crocea_2.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">llaticaudata_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Blue-ringed sea krait genes (latLat_1.0)</td>
<td class="gt_row gt_center" headers="version">latLat_1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">lleishanense_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Leishan spiny toad genes (ASM966780v1)</td>
<td class="gt_row gt_center" headers="version">ASM966780v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">loculatus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Spotted gar genes (LepOcu1)</td>
<td class="gt_row gt_center" headers="version">LepOcu1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">marmatus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Zig-zag eel genes (fMasArm1.2)</td>
<td class="gt_row gt_center" headers="version">fMasArm1.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">mauratus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Golden Hamster genes (MesAur1.0)</td>
<td class="gt_row gt_center" headers="version">MesAur1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">mcaroli_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Ryukyu mouse genes (CAROLI_EIJ_v1.1)</td>
<td class="gt_row gt_center" headers="version">CAROLI_EIJ_v1.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">mdomestica_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Opossum genes (ASM229v1)</td>
<td class="gt_row gt_center" headers="version">ASM229v1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">mfascicularis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Crab-eating macaque genes (Macaca_fascicularis_6.0)</td>
<td class="gt_row gt_center" headers="version">Macaca_fascicularis_6.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">mgallopavo_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Turkey genes (Turkey_5.1)</td>
<td class="gt_row gt_center" headers="version">Turkey_5.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">mleucophaeus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Drill genes (Mleu.le_1.0)</td>
<td class="gt_row gt_center" headers="version">Mleu.le_1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">mlucifugus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Microbat genes (Myoluc2.0)</td>
<td class="gt_row gt_center" headers="version">Myoluc2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">mmmarmota_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Alpine marmot genes (marMar2.1)</td>
<td class="gt_row gt_center" headers="version">marMar2.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">mmonoceros_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Narwhal genes (NGI_Narwhal_1)</td>
<td class="gt_row gt_center" headers="version">NGI_Narwhal_1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">mmoschiferus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Siberian musk deer genes (MosMos_v2_BIUU_UCD)</td>
<td class="gt_row gt_center" headers="version">MosMos_v2_BIUU_UCD</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">mmulatta_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Macaque genes (Mmul_10)</td>
<td class="gt_row gt_center" headers="version">Mmul_10</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">mmurdjan_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pinecone soldierfish genes (fMyrMur1.1)</td>
<td class="gt_row gt_center" headers="version">fMyrMur1.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">mmurinus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Mouse Lemur genes (Mmur_3.0)</td>
<td class="gt_row gt_center" headers="version">Mmur_3.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">mmusculus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Mouse genes (GRCm39)</td>
<td class="gt_row gt_center" headers="version">GRCm39</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">mnemestrina_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig-tailed macaque genes (Mnem_1.0)</td>
<td class="gt_row gt_center" headers="version">Mnem_1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">mochrogaster_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Prairie vole genes (MicOch1.0)</td>
<td class="gt_row gt_center" headers="version">MicOch1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">mpahari_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Shrew mouse genes (PAHARI_EIJ_v1.1)</td>
<td class="gt_row gt_center" headers="version">PAHARI_EIJ_v1.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">mpfuro_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Ferret genes (MusPutFur1.0)</td>
<td class="gt_row gt_center" headers="version">MusPutFur1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">mspicilegus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Steppe mouse genes (MUSP714)</td>
<td class="gt_row gt_center" headers="version">MUSP714</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">mspretus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Western wild mouse genes (SPRET_EiJ_v3)</td>
<td class="gt_row gt_center" headers="version">SPRET_EiJ_v3</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">mzebra_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Zebra mbuna genes (M_zebra_UMD2a)</td>
<td class="gt_row gt_center" headers="version">M_zebra_UMD2a</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">nbrichardi_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Lyretail cichlid genes (NeoBri1.0)</td>
<td class="gt_row gt_center" headers="version">NeoBri1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">neugenii_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Wallaby genes (Meug_1.0)</td>
<td class="gt_row gt_center" headers="version">Meug_1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">nfurzeri_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Turquoise killifish genes (Nfu_20140520)</td>
<td class="gt_row gt_center" headers="version">Nfu_20140520</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">ngalili_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Upper Galilee mountains blind mole rat genes (S.galili_v1.0)</td>
<td class="gt_row gt_center" headers="version">S.galili_v1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">nleucogenys_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Gibbon genes (Nleu_3.0)</td>
<td class="gt_row gt_center" headers="version">Nleu_3.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">nnaja_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Indian cobra genes (Nana_v5)</td>
<td class="gt_row gt_center" headers="version">Nana_v5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">nscutatus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Mainland tiger snake genes (TS10Xv2-PRI)</td>
<td class="gt_row gt_center" headers="version">TS10Xv2-PRI</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">nvison_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">American mink genes (NNQGG.v01)</td>
<td class="gt_row gt_center" headers="version">NNQGG.v01</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">oanatinus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Platypus genes (mOrnAna1.p.v1)</td>
<td class="gt_row gt_center" headers="version">mOrnAna1.p.v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">oaries_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Sheep genes (ARS-UI_Ramb_v3.0)</td>
<td class="gt_row gt_center" headers="version">ARS-UI_Ramb_v3.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">ocuniculus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Rabbit genes (OryCun2.0)</td>
<td class="gt_row gt_center" headers="version">OryCun2.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">odegus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Degu genes (OctDeg1.0)</td>
<td class="gt_row gt_center" headers="version">OctDeg1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">ogarnettii_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Bushbaby genes (OtoGar3)</td>
<td class="gt_row gt_center" headers="version">OtoGar3</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">ojavanicus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Javanese ricefish genes (OJAV_1.1)</td>
<td class="gt_row gt_center" headers="version">OJAV_1.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">okisutch_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Coho salmon genes (Okis_V2)</td>
<td class="gt_row gt_center" headers="version">Okis_V2</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">olatipes_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Japanese medaka HdrR genes (ASM223467v1)</td>
<td class="gt_row gt_center" headers="version">ASM223467v1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">omelastigma_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Indian medaka genes (Om_v0.7.RACA)</td>
<td class="gt_row gt_center" headers="version">Om_v0.7.RACA</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">omykiss_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Rainbow trout genes (USDA_OmykA_1.1)</td>
<td class="gt_row gt_center" headers="version">USDA_OmykA_1.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">oniloticus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Nile tilapia genes (O_niloticus_UMD_NMBU)</td>
<td class="gt_row gt_center" headers="version">O_niloticus_UMD_NMBU</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">oprinceps_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pika genes (OchPri2.0-Ens)</td>
<td class="gt_row gt_center" headers="version">OchPri2.0-Ens</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">osinensis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Chinese medaka genes (ASM858656v1)</td>
<td class="gt_row gt_center" headers="version">ASM858656v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">otshawytscha_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Chinook salmon genes (Otsh_v2.0)</td>
<td class="gt_row gt_center" headers="version">Otsh_v2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">pabelii_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Sumatran orangutan genes (Susie_PABv2)</td>
<td class="gt_row gt_center" headers="version">Susie_PABv2</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">panubis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Olive baboon genes (Panubis1.0)</td>
<td class="gt_row gt_center" headers="version">Panubis1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">pcapensis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Hyrax genes (proCap1)</td>
<td class="gt_row gt_center" headers="version">proCap1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">pcatodon_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Sperm whale genes (ASM283717v2)</td>
<td class="gt_row gt_center" headers="version">ASM283717v2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">pcinereus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Koala genes (phaCin_unsw_v4.1)</td>
<td class="gt_row gt_center" headers="version">phaCin_unsw_v4.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">pcoquereli_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Coquerel's sifaka genes (Pcoq_1.0)</td>
<td class="gt_row gt_center" headers="version">Pcoq_1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">pformosa_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Amazon molly genes (Poecilia_formosa-5.1.2)</td>
<td class="gt_row gt_center" headers="version">Poecilia_formosa-5.1.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">pkingsleyae_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Paramormyrops kingsleyae genes (PKINGS_0.1)</td>
<td class="gt_row gt_center" headers="version">PKINGS_0.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">platipinna_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Sailfin molly genes (P_latipinna-1.0)</td>
<td class="gt_row gt_center" headers="version">P_latipinna-1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">pleo_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Lion genes (PanLeo1.0)</td>
<td class="gt_row gt_center" headers="version">PanLeo1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">pmajor_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Great Tit genes (Parus_major1.1)</td>
<td class="gt_row gt_center" headers="version">Parus_major1.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">pmarinus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Lamprey genes (Pmarinus_7.0)</td>
<td class="gt_row gt_center" headers="version">Pmarinus_7.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">pmbairdii_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Northern American deer mouse genes (HU_Pman_2.1)</td>
<td class="gt_row gt_center" headers="version">HU_Pman_2.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">pmuralis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Common wall lizard genes (PodMur_1.0)</td>
<td class="gt_row gt_center" headers="version">PodMur_1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">pnattereri_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Red-bellied piranha genes (fPygNat1.pri)</td>
<td class="gt_row gt_center" headers="version">fPygNat1.pri</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">pnyererei_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Makobe Island cichlid genes (PunNye1.0)</td>
<td class="gt_row gt_center" headers="version">PunNye1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">ppaniscus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Bonobo genes (panpan1.1)</td>
<td class="gt_row gt_center" headers="version">panpan1.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">ppardus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Leopard genes (PanPar1.0)</td>
<td class="gt_row gt_center" headers="version">PanPar1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">preticulata_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Guppy genes (Guppy_female_1.0_MT)</td>
<td class="gt_row gt_center" headers="version">Guppy_female_1.0_MT</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">psimus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Greater bamboo lemur genes (Prosim_1.0)</td>
<td class="gt_row gt_center" headers="version">Prosim_1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">psinensis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Chinese softshell turtle genes (PelSin_1.0)</td>
<td class="gt_row gt_center" headers="version">PelSin_1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">psinus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Vaquita genes (mPhoSin1.pri)</td>
<td class="gt_row gt_center" headers="version">mPhoSin1.pri</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">ptaltaica_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Tiger genes (PanTig1.0)</td>
<td class="gt_row gt_center" headers="version">PanTig1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">ptextilis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Eastern brown snake genes (EBS10Xv2-PRI)</td>
<td class="gt_row gt_center" headers="version">EBS10Xv2-PRI</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">ptroglodytes_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Chimpanzee genes (Pan_tro_3.0)</td>
<td class="gt_row gt_center" headers="version">Pan_tro_3.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">pvampyrus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Megabat genes (pteVam1)</td>
<td class="gt_row gt_center" headers="version">pteVam1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">rbieti_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Black snub-nosed monkey genes (ASM169854v1)</td>
<td class="gt_row gt_center" headers="version">ASM169854v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">rferrumequinum_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Greater horseshoe bat genes (mRhiFer1_v1.p)</td>
<td class="gt_row gt_center" headers="version">mRhiFer1_v1.p</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">rnorvegicus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Norway rat - BN/NHsdMcwi genes (GRCr8)</td>
<td class="gt_row gt_center" headers="version">GRCr8</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">rroxellana_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Golden snub-nosed monkey genes (Rrox_v1)</td>
<td class="gt_row gt_center" headers="version">Rrox_v1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">saraneus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Shrew genes (sorAra1)</td>
<td class="gt_row gt_center" headers="version">sorAra1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">saurata_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Gilthead seabream genes (fSpaAur1.1)</td>
<td class="gt_row gt_center" headers="version">fSpaAur1.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">sbboliviensis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Bolivian squirrel monkey genes (SaiBol1.0)</td>
<td class="gt_row gt_center" headers="version">SaiBol1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">scanaria_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Common canary genes (SCA1)</td>
<td class="gt_row gt_center" headers="version">SCA1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">scaustralis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">African ostrich genes (ASM69896v1)</td>
<td class="gt_row gt_center" headers="version">ASM69896v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">scerevisiae_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Saccharomyces cerevisiae genes (R64-1-1)</td>
<td class="gt_row gt_center" headers="version">R64-1-1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">sdumerili_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Greater amberjack genes (Sdu_1.0)</td>
<td class="gt_row gt_center" headers="version">Sdu_1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">sformosus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Asian bonytongue genes (fSclFor1.1)</td>
<td class="gt_row gt_center" headers="version">fSclFor1.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">shabroptila_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Kakapo genes (bStrHab1_v1.p)</td>
<td class="gt_row gt_center" headers="version">bStrHab1_v1.p</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">sharrisii_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Tasmanian devil genes (mSarHar1.11)</td>
<td class="gt_row gt_center" headers="version">mSarHar1.11</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">sldorsalis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Yellowtail amberjack genes (Sedor1)</td>
<td class="gt_row gt_center" headers="version">Sedor1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">slucioperca_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pike-perch genes (SLUC_FBN_1)</td>
<td class="gt_row gt_center" headers="version">SLUC_FBN_1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">smaximus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Turbot genes (ASM1334776v1)</td>
<td class="gt_row gt_center" headers="version">ASM1334776v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">smerianae_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Argentine black and white tegu genes (HLtupMer3)</td>
<td class="gt_row gt_center" headers="version">HLtupMer3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">spartitus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Bicolor damselfish genes (Stegastes_partitus-1.0.2)</td>
<td class="gt_row gt_center" headers="version">Stegastes_partitus-1.0.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">spunctatus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Tuatara genes (ASM311381v1)</td>
<td class="gt_row gt_center" headers="version">ASM311381v1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">ssalar_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Atlantic salmon genes (Ssal_v3.1)</td>
<td class="gt_row gt_center" headers="version">Ssal_v3.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">ssbamei_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig - Bamei genes (Bamei_pig_v1)</td>
<td class="gt_row gt_center" headers="version">Bamei_pig_v1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">ssberkshire_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig - Berkshire genes (Berkshire_pig_v1)</td>
<td class="gt_row gt_center" headers="version">Berkshire_pig_v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">sscrofa_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig genes (Sscrofa11.1)</td>
<td class="gt_row gt_center" headers="version">Sscrofa11.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">sshampshire_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig - Hampshire genes (Hampshire_pig_v1)</td>
<td class="gt_row gt_center" headers="version">Hampshire_pig_v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">ssjinhua_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig - Jinhua genes (Jinhua_pig_v1)</td>
<td class="gt_row gt_center" headers="version">Jinhua_pig_v1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">sslandrace_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig - Landrace genes (Landrace_pig_v1)</td>
<td class="gt_row gt_center" headers="version">Landrace_pig_v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">sslargewhite_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig - Largewhite genes (Large_White_v1)</td>
<td class="gt_row gt_center" headers="version">Large_White_v1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">ssmeishan_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig - Meishan genes (Meishan_pig_v1)</td>
<td class="gt_row gt_center" headers="version">Meishan_pig_v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">sspietrain_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig - Pietrain genes (Pietrain_pig_v1)</td>
<td class="gt_row gt_center" headers="version">Pietrain_pig_v1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">ssrongchang_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig - Rongchang genes (Rongchang_pig_v1)</td>
<td class="gt_row gt_center" headers="version">Rongchang_pig_v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">sstibetan_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig - Tibetan genes (Tibetan_Pig_v2)</td>
<td class="gt_row gt_center" headers="version">Tibetan_Pig_v2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">ssusmarc_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig USMARC genes (USMARCv1.0)</td>
<td class="gt_row gt_center" headers="version">USMARCv1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">sswuzhishan_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Pig - Wuzhishan genes (minipig_v1.0)</td>
<td class="gt_row gt_center" headers="version">minipig_v1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">strutta_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Brown trout genes (fSalTru1.1)</td>
<td class="gt_row gt_center" headers="version">fSalTru1.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">svulgaris_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Eurasian red squirrel genes (mSciVul1.1)</td>
<td class="gt_row gt_center" headers="version">mSciVul1.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">tbelangeri_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Tree Shrew genes (tupBel1)</td>
<td class="gt_row gt_center" headers="version">tupBel1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">tctriunguis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Three-toed box turtle genes (T_m_triunguis-2.0)</td>
<td class="gt_row gt_center" headers="version">T_m_triunguis-2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">tguttata_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Zebra finch genes (bTaeGut1_v1.p)</td>
<td class="gt_row gt_center" headers="version">bTaeGut1_v1.p</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">tnigroviridis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Tetraodon genes (TETRAODON 8.0)</td>
<td class="gt_row gt_center" headers="version">TETRAODON 8.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">trubripes_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Fugu genes (fTakRub1.2)</td>
<td class="gt_row gt_center" headers="version">fTakRub1.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">ttruncatus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Dolphin genes (turTru1)</td>
<td class="gt_row gt_center" headers="version">turTru1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">uamericanus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">American black bear genes (ASM334442v1)</td>
<td class="gt_row gt_center" headers="version">ASM334442v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">umaritimus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Polar bear genes (UrsMar_1.0)</td>
<td class="gt_row gt_center" headers="version">UrsMar_1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">uparryii_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Arctic ground squirrel genes (ASM342692v1)</td>
<td class="gt_row gt_center" headers="version">ASM342692v1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">vpacos_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Alpaca genes (vicPac1)</td>
<td class="gt_row gt_center" headers="version">vicPac1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">vursinus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Common wombat genes (bare-nosed_wombat_genome_assembly)</td>
<td class="gt_row gt_center" headers="version">bare-nosed_wombat_genome_assembly</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">vvulpes_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Red fox genes (VulVul2.2)</td>
<td class="gt_row gt_center" headers="version">VulVul2.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">xcouchianus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Monterrey platyfish genes (Xiphophorus_couchianus-4.0.1)</td>
<td class="gt_row gt_center" headers="version">Xiphophorus_couchianus-4.0.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="dataset">xmaculatus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Platyfish genes (X_maculatus-5.0-male)</td>
<td class="gt_row gt_center" headers="version">X_maculatus-5.0-male</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">xtropicalis_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Tropical clawed frog genes (UCB_Xtro_10.0)</td>
<td class="gt_row gt_center" headers="version">UCB_Xtro_10.0</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<p>A lot of stuff for a lot of species! Perhaps we want to limit it to see which ones are relevant to mouse.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a></a>mousedatasets <span class="ot">&lt;-</span> <span class="fu">filter</span>(datasets, <span class="fu">grepl</span>(<span class="st">"mmusculus"</span>, dataset))</span>
<span id="cb11-2"><a></a><span class="fu">gt</span>(mousedatasets)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="hfntqyyyoo" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#hfntqyyyoo table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#hfntqyyyoo thead, #hfntqyyyoo tbody, #hfntqyyyoo tfoot, #hfntqyyyoo tr, #hfntqyyyoo td, #hfntqyyyoo th {
  border-style: none;
}

#hfntqyyyoo p {
  margin: 0;
  padding: 0;
}

#hfntqyyyoo .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#hfntqyyyoo .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#hfntqyyyoo .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#hfntqyyyoo .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#hfntqyyyoo .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#hfntqyyyoo .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#hfntqyyyoo .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#hfntqyyyoo .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#hfntqyyyoo .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#hfntqyyyoo .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#hfntqyyyoo .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#hfntqyyyoo .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#hfntqyyyoo .gt_spanner_row {
  border-bottom-style: hidden;
}

#hfntqyyyoo .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#hfntqyyyoo .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#hfntqyyyoo .gt_from_md > :first-child {
  margin-top: 0;
}

#hfntqyyyoo .gt_from_md > :last-child {
  margin-bottom: 0;
}

#hfntqyyyoo .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#hfntqyyyoo .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#hfntqyyyoo .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#hfntqyyyoo .gt_row_group_first td {
  border-top-width: 2px;
}

#hfntqyyyoo .gt_row_group_first th {
  border-top-width: 2px;
}

#hfntqyyyoo .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#hfntqyyyoo .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#hfntqyyyoo .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#hfntqyyyoo .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#hfntqyyyoo .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#hfntqyyyoo .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#hfntqyyyoo .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#hfntqyyyoo .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#hfntqyyyoo .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#hfntqyyyoo .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#hfntqyyyoo .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#hfntqyyyoo .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#hfntqyyyoo .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#hfntqyyyoo .gt_left {
  text-align: left;
}

#hfntqyyyoo .gt_center {
  text-align: center;
}

#hfntqyyyoo .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#hfntqyyyoo .gt_font_normal {
  font-weight: normal;
}

#hfntqyyyoo .gt_font_bold {
  font-weight: bold;
}

#hfntqyyyoo .gt_font_italic {
  font-style: italic;
}

#hfntqyyyoo .gt_super {
  font-size: 65%;
}

#hfntqyyyoo .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#hfntqyyyoo .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#hfntqyyyoo .gt_indent_1 {
  text-indent: 5px;
}

#hfntqyyyoo .gt_indent_2 {
  text-indent: 10px;
}

#hfntqyyyoo .gt_indent_3 {
  text-indent: 15px;
}

#hfntqyyyoo .gt_indent_4 {
  text-indent: 20px;
}

#hfntqyyyoo .gt_indent_5 {
  text-indent: 25px;
}

#hfntqyyyoo .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#hfntqyyyoo div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="dataset" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">dataset</th>
<th id="description" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">description</th>
<th id="version" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">version</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_center" headers="dataset">mmusculus_gene_ensembl</td>
<td class="gt_row gt_center" headers="description">Mouse genes (GRCm39)</td>
<td class="gt_row gt_center" headers="version">GRCm39</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="using-biomart-1" class="slide level2 smaller">
<h2>Using biomaRt</h2>
<p>Ah so we probably want the dataset called ‘mmusculus_gene_ensembl’!</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a></a>mart <span class="ot">&lt;-</span> biomaRt<span class="sc">::</span><span class="fu">useMart</span>(</span>
<span id="cb12-2"><a></a>  <span class="st">"ENSEMBL_MART_ENSEMBL"</span>,</span>
<span id="cb12-3"><a></a>  <span class="at">dataset =</span> <span class="st">"mmusculus_gene_ensembl"</span>,</span>
<span id="cb12-4"><a></a>  <span class="at">host =</span> <span class="st">"www.ensembl.org"</span></span>
<span id="cb12-5"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Ensembl will soon enforce the use of https.
Ensure the 'host' argument includes "https://"</code></pre>
</div>
</div>
</section>
<section id="using-biomart-2" class="slide level2 smaller">
<h2>Using biomaRt</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a></a>goodies <span class="ot">&lt;-</span> <span class="fu">listAttributes</span>(mart)</span>
<span id="cb14-2"><a></a><span class="fu">gt</span>(goodies[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, ])</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="vlxymjplbm" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#vlxymjplbm table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#vlxymjplbm thead, #vlxymjplbm tbody, #vlxymjplbm tfoot, #vlxymjplbm tr, #vlxymjplbm td, #vlxymjplbm th {
  border-style: none;
}

#vlxymjplbm p {
  margin: 0;
  padding: 0;
}

#vlxymjplbm .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#vlxymjplbm .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#vlxymjplbm .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#vlxymjplbm .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#vlxymjplbm .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#vlxymjplbm .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#vlxymjplbm .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#vlxymjplbm .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#vlxymjplbm .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#vlxymjplbm .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#vlxymjplbm .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#vlxymjplbm .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#vlxymjplbm .gt_spanner_row {
  border-bottom-style: hidden;
}

#vlxymjplbm .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#vlxymjplbm .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#vlxymjplbm .gt_from_md > :first-child {
  margin-top: 0;
}

#vlxymjplbm .gt_from_md > :last-child {
  margin-bottom: 0;
}

#vlxymjplbm .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#vlxymjplbm .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#vlxymjplbm .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#vlxymjplbm .gt_row_group_first td {
  border-top-width: 2px;
}

#vlxymjplbm .gt_row_group_first th {
  border-top-width: 2px;
}

#vlxymjplbm .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#vlxymjplbm .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#vlxymjplbm .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#vlxymjplbm .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#vlxymjplbm .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#vlxymjplbm .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#vlxymjplbm .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#vlxymjplbm .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#vlxymjplbm .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#vlxymjplbm .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#vlxymjplbm .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#vlxymjplbm .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#vlxymjplbm .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#vlxymjplbm .gt_left {
  text-align: left;
}

#vlxymjplbm .gt_center {
  text-align: center;
}

#vlxymjplbm .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#vlxymjplbm .gt_font_normal {
  font-weight: normal;
}

#vlxymjplbm .gt_font_bold {
  font-weight: bold;
}

#vlxymjplbm .gt_font_italic {
  font-style: italic;
}

#vlxymjplbm .gt_super {
  font-size: 65%;
}

#vlxymjplbm .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#vlxymjplbm .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#vlxymjplbm .gt_indent_1 {
  text-indent: 5px;
}

#vlxymjplbm .gt_indent_2 {
  text-indent: 10px;
}

#vlxymjplbm .gt_indent_3 {
  text-indent: 15px;
}

#vlxymjplbm .gt_indent_4 {
  text-indent: 20px;
}

#vlxymjplbm .gt_indent_5 {
  text-indent: 25px;
}

#vlxymjplbm .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#vlxymjplbm div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">name</th>
<th id="description" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">description</th>
<th id="page" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">page</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="name">ensembl_gene_id</td>
<td class="gt_row gt_left" headers="description">Gene stable ID</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">ensembl_gene_id_version</td>
<td class="gt_row gt_left" headers="description">Gene stable ID version</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">ensembl_transcript_id</td>
<td class="gt_row gt_left" headers="description">Transcript stable ID</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">ensembl_transcript_id_version</td>
<td class="gt_row gt_left" headers="description">Transcript stable ID version</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">ensembl_peptide_id</td>
<td class="gt_row gt_left" headers="description">Protein stable ID</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">ensembl_peptide_id_version</td>
<td class="gt_row gt_left" headers="description">Protein stable ID version</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">ensembl_exon_id</td>
<td class="gt_row gt_left" headers="description">Exon stable ID</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">description</td>
<td class="gt_row gt_left" headers="description">Gene description</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">chromosome_name</td>
<td class="gt_row gt_left" headers="description">Chromosome/scaffold name</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">start_position</td>
<td class="gt_row gt_left" headers="description">Gene start (bp)</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">end_position</td>
<td class="gt_row gt_left" headers="description">Gene end (bp)</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">strand</td>
<td class="gt_row gt_left" headers="description">Strand</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">band</td>
<td class="gt_row gt_left" headers="description">Karyotype band</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">transcript_start</td>
<td class="gt_row gt_left" headers="description">Transcript start (bp)</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">transcript_end</td>
<td class="gt_row gt_left" headers="description">Transcript end (bp)</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">transcription_start_site</td>
<td class="gt_row gt_left" headers="description">Transcription start site (TSS)</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">transcript_length</td>
<td class="gt_row gt_left" headers="description">Transcript length (including UTRs and CDS)</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">transcript_tsl</td>
<td class="gt_row gt_left" headers="description">Transcript support level (TSL)</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">transcript_gencode_basic</td>
<td class="gt_row gt_left" headers="description">GENCODE basic annotation</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">transcript_primary_basic</td>
<td class="gt_row gt_left" headers="description">GENCODE primary annotation</td>
<td class="gt_row gt_left" headers="page">feature_page</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="using-biomart-3" class="slide level2 smaller">
<h2>Using biomaRt</h2>
<p>So there are 2885 rows of goodies about the mouse genome and its relationship to <em>many</em> other genomes. However, you can probably see that the ones that are most useful to us right now are right at the top: ‘ensembl_transcript_id’ and ‘ensembl_gene_id’. We can use those attributes in our mart to make a table relating genes and transcripts.</p>
<p>I’m going to through one more attribute in: external_gene_name. Those are usually more informative than ensembl IDs.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a></a>t2g <span class="ot">&lt;-</span> biomaRt<span class="sc">::</span><span class="fu">getBM</span>(</span>
<span id="cb15-2"><a></a>  <span class="at">attributes =</span> <span class="fu">c</span>(</span>
<span id="cb15-3"><a></a>    <span class="st">"ensembl_transcript_id"</span>,</span>
<span id="cb15-4"><a></a>    <span class="st">"ensembl_gene_id"</span>,</span>
<span id="cb15-5"><a></a>    <span class="st">"external_gene_name"</span></span>
<span id="cb15-6"><a></a>  ),</span>
<span id="cb15-7"><a></a>  <span class="at">mart =</span> mart</span>
<span id="cb15-8"><a></a>)</span>
<span id="cb15-9"><a></a></span>
<span id="cb15-10"><a></a><span class="co"># write_tsv(x = t2g, file = here("data","block-rna","t2g.tsv.gz"))</span></span>
<span id="cb15-11"><a></a>t2g <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"block-rna"</span>, <span class="st">"t2g.tsv.gz"</span>))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 149547 Columns: 3
── Column specification ────────────────────────────────────
Delimiter: "\t"
chr (3): ensembl_transcript_id, ensembl_gene_id, externa...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a></a><span class="fu">gt</span>(t2g[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, ])</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="gznmhnbehm" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#gznmhnbehm table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#gznmhnbehm thead, #gznmhnbehm tbody, #gznmhnbehm tfoot, #gznmhnbehm tr, #gznmhnbehm td, #gznmhnbehm th {
  border-style: none;
}

#gznmhnbehm p {
  margin: 0;
  padding: 0;
}

#gznmhnbehm .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#gznmhnbehm .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#gznmhnbehm .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#gznmhnbehm .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#gznmhnbehm .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gznmhnbehm .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gznmhnbehm .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gznmhnbehm .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#gznmhnbehm .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#gznmhnbehm .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#gznmhnbehm .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#gznmhnbehm .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#gznmhnbehm .gt_spanner_row {
  border-bottom-style: hidden;
}

#gznmhnbehm .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#gznmhnbehm .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#gznmhnbehm .gt_from_md > :first-child {
  margin-top: 0;
}

#gznmhnbehm .gt_from_md > :last-child {
  margin-bottom: 0;
}

#gznmhnbehm .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#gznmhnbehm .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#gznmhnbehm .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#gznmhnbehm .gt_row_group_first td {
  border-top-width: 2px;
}

#gznmhnbehm .gt_row_group_first th {
  border-top-width: 2px;
}

#gznmhnbehm .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gznmhnbehm .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#gznmhnbehm .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#gznmhnbehm .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gznmhnbehm .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gznmhnbehm .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#gznmhnbehm .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#gznmhnbehm .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#gznmhnbehm .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gznmhnbehm .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gznmhnbehm .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#gznmhnbehm .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gznmhnbehm .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#gznmhnbehm .gt_left {
  text-align: left;
}

#gznmhnbehm .gt_center {
  text-align: center;
}

#gznmhnbehm .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#gznmhnbehm .gt_font_normal {
  font-weight: normal;
}

#gznmhnbehm .gt_font_bold {
  font-weight: bold;
}

#gznmhnbehm .gt_font_italic {
  font-style: italic;
}

#gznmhnbehm .gt_super {
  font-size: 65%;
}

#gznmhnbehm .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#gznmhnbehm .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#gznmhnbehm .gt_indent_1 {
  text-indent: 5px;
}

#gznmhnbehm .gt_indent_2 {
  text-indent: 10px;
}

#gznmhnbehm .gt_indent_3 {
  text-indent: 15px;
}

#gznmhnbehm .gt_indent_4 {
  text-indent: 20px;
}

#gznmhnbehm .gt_indent_5 {
  text-indent: 25px;
}

#gznmhnbehm .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#gznmhnbehm div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="ensembl_transcript_id" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ensembl_transcript_id</th>
<th id="ensembl_gene_id" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ensembl_gene_id</th>
<th id="external_gene_name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">external_gene_name</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082387</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064336</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Tf</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082388</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064337</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Rnr1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082389</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064338</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Tv</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082390</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064339</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Rnr2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082391</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064340</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Tl1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082392</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064341</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Nd1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082393</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064342</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Ti</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082394</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064343</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Tq</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082395</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064344</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Tm</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082396</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064345</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Nd2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082397</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064346</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Tw</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082398</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064347</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Ta</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082399</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064348</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Tn</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082400</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064349</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Tc</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082401</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064350</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Ty</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082402</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064351</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Co1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082403</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064352</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Ts1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082404</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064353</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Td</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082405</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064354</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Co2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082406</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064355</td>
<td class="gt_row gt_left" headers="external_gene_name">mt-Tk</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="using-biomart-4" class="slide level2 smaller">
<h2>Using biomaRt</h2>
<p>Alright this looks good! We are going to split this into two tables. One that contains transcript ID and gene ID, and the other that contains gene ID and gene name.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a></a>t2g <span class="ot">&lt;-</span> t2g <span class="sc">|&gt;</span></span>
<span id="cb18-2"><a></a>  <span class="fu">select</span>(ensembl_transcript_id, ensembl_gene_id)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="getting-gene-level-expression-data-with-tximport" class="slide level2 smaller">
<h2>Getting gene level expression data with <code>tximport</code></h2>
<p>Now that we have our table relating transcripts and genes, we can give it to tximport to have it calculate gene-level expression data from our transcript-level expression data.</p>
<p>First, we have to tell it where the salmon quantification files (the <code>quant.sf.gz</code> files) are. Here’s what our directory structure that contains these files looks like:</p>

<img data-src="../img/block-rna/salmondirstructure.png" class="r-stretch"></section>
<section id="gene-expression-data-with-tximport" class="slide level2 smaller">
<h2>Gene expression data with <code>tximport</code></h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a></a><span class="co"># The directory where all of the sample-specific salmon subdirectories live</span></span>
<span id="cb19-2"><a></a></span>
<span id="cb19-3"><a></a>metadata <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb19-4"><a></a>  <span class="at">salmon_dirs =</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(</span>
<span id="cb19-5"><a></a>    <span class="fu">here</span>(<span class="st">"data/block-rna/differentiation_salmonouts"</span>),</span>
<span id="cb19-6"><a></a>    <span class="at">recurse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-7"><a></a>    <span class="at">glob =</span> <span class="st">"*quant.sf"</span></span>
<span id="cb19-8"><a></a>  )</span>
<span id="cb19-9"><a></a>) <span class="sc">|&gt;</span></span>
<span id="cb19-10"><a></a>  <span class="fu">mutate</span>(</span>
<span id="cb19-11"><a></a>    <span class="at">sample_id =</span> fs<span class="sc">::</span><span class="fu">path_file</span>(fs<span class="sc">::</span><span class="fu">path_dir</span>(salmon_dirs))</span>
<span id="cb19-12"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb19-13"><a></a>  <span class="fu">filter</span>(<span class="fu">str_starts</span>(sample_id, <span class="st">"DIV"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb19-14"><a></a>  <span class="fu">separate_wider_delim</span>(</span>
<span id="cb19-15"><a></a>    <span class="at">col =</span> sample_id,</span>
<span id="cb19-16"><a></a>    <span class="at">delim =</span> <span class="st">"."</span>,</span>
<span id="cb19-17"><a></a>    <span class="at">names =</span> <span class="fu">c</span>(<span class="st">"samp"</span>, <span class="st">"rep"</span>),</span>
<span id="cb19-18"><a></a>    <span class="at">too_few =</span> <span class="st">"align_start"</span>,</span>
<span id="cb19-19"><a></a>    <span class="at">cols_remove =</span> <span class="cn">FALSE</span></span>
<span id="cb19-20"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb19-21"><a></a>  <span class="fu">mutate</span>(<span class="at">rep =</span> <span class="fu">str_remove</span>(rep, <span class="st">"Rep"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb19-22"><a></a>  <span class="fu">column_to_rownames</span>(<span class="st">"sample_id"</span>)</span>
<span id="cb19-23"><a></a></span>
<span id="cb19-24"><a></a><span class="co"># Add the sample_id column back from rownames</span></span>
<span id="cb19-25"><a></a>metadata<span class="sc">$</span>sample_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(metadata)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="gene-expression-data-with-tximport-1" class="slide level2 smaller">
<h2>Gene expression data with <code>tximport</code></h2>
<p>You can see that we got a list of sample names and the absolute path to each sample’s quantification file.</p>
<p>Now we are ready to run <code>tximport</code></p>
<p><code>tximport</code> is going to want paths to all the quantification files (salm_dirs) and a table that relates transcripts to genes (t2g). Luckily, we happen to have those exact two things.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a></a>salmdir <span class="ot">&lt;-</span> metadata<span class="sc">$</span>salmon_dirs</span>
<span id="cb20-2"><a></a><span class="fu">names</span>(salmdir) <span class="ot">&lt;-</span> metadata<span class="sc">$</span>sample_id</span>
<span id="cb20-3"><a></a></span>
<span id="cb20-4"><a></a></span>
<span id="cb20-5"><a></a>txi <span class="ot">&lt;-</span> <span class="fu">tximport</span>(</span>
<span id="cb20-6"><a></a>  <span class="at">files =</span> salmdir,</span>
<span id="cb20-7"><a></a>  <span class="at">type =</span> <span class="st">"salmon"</span>,</span>
<span id="cb20-8"><a></a>  <span class="at">tx2gene =</span> t2g,</span>
<span id="cb20-9"><a></a>  <span class="at">dropInfReps =</span> <span class="cn">TRUE</span>,</span>
<span id="cb20-10"><a></a>  <span class="at">countsFromAbundance =</span> <span class="st">"lengthScaledTPM"</span></span>
<span id="cb20-11"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>reading in files with read_tsv</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 
transcripts missing from tx2gene: 3733
summarizing abundance
summarizing counts
summarizing length</code></pre>
</div>
</div>
</section>
<section id="gene-expression-data-with-tximport-2" class="slide level2 smaller">
<h2>Gene expression data with <code>tximport</code></h2>
<p>Notice how we chose <em>lengthscaledTPM</em> for our abundance measurement. This is going to give us TPM values (transcripts per million) for expression in the $abundance slot. Let’s check out what we have now.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a></a>tpms <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(txi<span class="sc">$</span>abundance) <span class="sc">|&gt;</span></span>
<span id="cb23-2"><a></a>  <span class="fu">set_names</span>(<span class="fu">rownames</span>(metadata)) <span class="sc">|&gt;</span></span>
<span id="cb23-3"><a></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"ensembl_gene_id"</span>)</span>
<span id="cb23-4"><a></a></span>
<span id="cb23-5"><a></a><span class="fu">gt</span>(tpms[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>, ])</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fglmzksapk" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#fglmzksapk table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#fglmzksapk thead, #fglmzksapk tbody, #fglmzksapk tfoot, #fglmzksapk tr, #fglmzksapk td, #fglmzksapk th {
  border-style: none;
}

#fglmzksapk p {
  margin: 0;
  padding: 0;
}

#fglmzksapk .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#fglmzksapk .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#fglmzksapk .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#fglmzksapk .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#fglmzksapk .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#fglmzksapk .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fglmzksapk .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#fglmzksapk .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#fglmzksapk .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#fglmzksapk .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#fglmzksapk .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#fglmzksapk .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#fglmzksapk .gt_spanner_row {
  border-bottom-style: hidden;
}

#fglmzksapk .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#fglmzksapk .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#fglmzksapk .gt_from_md > :first-child {
  margin-top: 0;
}

#fglmzksapk .gt_from_md > :last-child {
  margin-bottom: 0;
}

#fglmzksapk .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#fglmzksapk .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#fglmzksapk .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#fglmzksapk .gt_row_group_first td {
  border-top-width: 2px;
}

#fglmzksapk .gt_row_group_first th {
  border-top-width: 2px;
}

#fglmzksapk .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#fglmzksapk .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#fglmzksapk .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#fglmzksapk .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fglmzksapk .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#fglmzksapk .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#fglmzksapk .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#fglmzksapk .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#fglmzksapk .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fglmzksapk .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#fglmzksapk .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#fglmzksapk .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#fglmzksapk .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#fglmzksapk .gt_left {
  text-align: left;
}

#fglmzksapk .gt_center {
  text-align: center;
}

#fglmzksapk .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#fglmzksapk .gt_font_normal {
  font-weight: normal;
}

#fglmzksapk .gt_font_bold {
  font-weight: bold;
}

#fglmzksapk .gt_font_italic {
  font-style: italic;
}

#fglmzksapk .gt_super {
  font-size: 65%;
}

#fglmzksapk .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#fglmzksapk .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#fglmzksapk .gt_indent_1 {
  text-indent: 5px;
}

#fglmzksapk .gt_indent_2 {
  text-indent: 10px;
}

#fglmzksapk .gt_indent_3 {
  text-indent: 15px;
}

#fglmzksapk .gt_indent_4 {
  text-indent: 20px;
}

#fglmzksapk .gt_indent_5 {
  text-indent: 25px;
}

#fglmzksapk .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#fglmzksapk div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="ensembl_gene_id" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ensembl_gene_id</th>
<th id="DIV0.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV0.Rep1</th>
<th id="DIV0.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV0.Rep2</th>
<th id="DIV0.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV0.Rep3</th>
<th id="DIV1.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV1.Rep1</th>
<th id="DIV1.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV1.Rep2</th>
<th id="DIV1.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV1.Rep3</th>
<th id="DIV1.Rep4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV1.Rep4</th>
<th id="DIV16.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV16.Rep1</th>
<th id="DIV16.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV16.Rep2</th>
<th id="DIV16.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV16.Rep3</th>
<th id="DIV16.Rep4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV16.Rep4</th>
<th id="DIV21.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV21.Rep1</th>
<th id="DIV21.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV21.Rep2</th>
<th id="DIV21.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV21.Rep3</th>
<th id="DIV21.Rep4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV21.Rep4</th>
<th id="DIV28.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV28.Rep1</th>
<th id="DIV28.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV28.Rep2</th>
<th id="DIV28.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV28.Rep3</th>
<th id="DIV28.Rep4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV28.Rep4</th>
<th id="DIV7.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV7.Rep1</th>
<th id="DIV7.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV7.Rep2</th>
<th id="DIV7.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV7.Rep3</th>
<th id="DIV7.Rep4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV7.Rep4</th>
<th id="DIVminus4.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus4.Rep1</th>
<th id="DIVminus4.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus4.Rep2</th>
<th id="DIVminus4.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus4.Rep3</th>
<th id="DIVminus8.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus8.Rep1</th>
<th id="DIVminus8.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus8.Rep2</th>
<th id="DIVminus8.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus8.Rep3</th>
<th id="DIVminus8.Rep4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus8.Rep4</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000001</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">148.004511</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">145.539529</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">164.680183</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">142.713005</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">138.042672</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">137.027499</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">123.894516</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">41.689798</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">41.554126</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">41.002841</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">39.522314</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">28.162815</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">29.694158</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">29.333682</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">27.956845</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">17.960041</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">17.979483</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">18.102228</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">19.590551</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">76.088940</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">76.669016</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">79.505738</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">76.038479</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">106.634101</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">106.395094</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">109.997598</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">92.799612</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">93.563524</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">95.500337</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">101.951963</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000003</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000028</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">34.180035</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">36.215924</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">39.486489</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">16.894844</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">15.391383</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">18.171282</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">16.654062</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">1.067147</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">1.270428</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">1.109391</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">1.136970</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">1.939558</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">1.003108</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">1.245186</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">1.086912</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">1.303824</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.935135</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.864451</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.862209</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">6.582942</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">6.752942</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">5.547107</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">5.510259</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">45.614077</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">47.851930</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">47.516125</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">59.718765</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">57.981259</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">57.935839</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">41.663343</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000031</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">7.213808</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">6.431925</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">13.253322</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">3.301713</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">2.539161</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">3.591509</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">2.798164</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">93.312776</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">56.314466</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">62.958855</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">68.565195</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">32.037271</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">37.465959</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">40.581328</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">24.780757</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">67.818115</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">64.333586</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">30.408290</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">60.198406</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">151.244212</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">137.227925</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">169.615059</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">150.599359</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.914481</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">1.402704</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">1.317939</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.177770</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.215726</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.320948</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">36.425779</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000037</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">5.877237</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">5.001733</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">7.579149</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">2.861842</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">3.176129</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">2.457556</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">3.591170</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.319281</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.708167</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.578467</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.513434</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.186198</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.260270</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.258075</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.585622</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.615895</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.133394</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.197050</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">1.005331</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">2.750903</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">1.827190</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">2.446732</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">2.894751</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">2.196733</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">2.309187</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">2.213591</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">3.187886</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">2.507155</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">2.624140</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">2.281871</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000049</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.046208</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.411807</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.213589</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.118890</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">1.024154</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.628760</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.303218</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.449928</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.262019</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.105047</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.567535</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.649720</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.515246</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.360877</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.188254</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.050654</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.277837</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.189814</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000056</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">31.553583</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">28.610456</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">19.741875</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">59.046908</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">51.324548</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">54.015893</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">43.897451</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">36.575044</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">47.171198</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">41.459018</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">38.403920</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">55.170239</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">37.441141</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">44.043899</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">45.852392</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">38.906362</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">40.362321</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">44.933634</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">42.356811</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">41.828967</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">46.812515</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">44.172289</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">46.834670</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">14.509458</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">18.656028</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">16.335093</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">10.671419</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">10.666954</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">12.386288</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">8.077939</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000058</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.420304</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.507728</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.602757</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.357594</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.504091</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.498006</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.565744</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.931596</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.632998</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.606530</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.503290</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.292719</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.224968</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.401341</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.428359</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.490118</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.200389</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.038911</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.399079</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.194399</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.102993</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.063154</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.130644</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.163821</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.162316</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.251082</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.693347</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.201366</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.527565</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.482299</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000078</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">52.604855</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">64.802703</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">76.892914</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">60.446343</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">60.861211</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">61.059568</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">50.159693</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">32.474649</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">28.287563</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">27.391212</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">29.511742</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">19.632508</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">19.942257</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">20.182975</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">19.363735</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">16.845197</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">16.128839</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">18.828771</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">15.732413</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">52.939918</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">52.709453</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">54.334181</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">50.804497</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">51.427093</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">49.900737</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">53.274561</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">19.669879</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">19.472875</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">20.216492</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">21.591729</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000085</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">60.617248</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">59.182914</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">61.623794</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">72.657499</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">71.506733</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">75.816856</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">63.507273</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">36.498680</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">31.450681</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">36.983697</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">34.558348</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">31.893252</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">35.692047</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">31.095232</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">34.751542</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">36.339896</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">32.474173</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">38.456832</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">39.182603</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">52.494911</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">43.715591</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">45.720249</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">46.351620</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">9.949776</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">11.703136</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">9.320286</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">13.843137</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">11.699411</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">13.486194</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">10.366294</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000088</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">372.546920</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">364.764405</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">381.963748</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">389.617591</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">388.204349</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">413.080094</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">533.720843</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">882.254461</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">918.864762</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">881.567068</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">952.656933</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">1064.561469</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">1018.564149</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">1023.266589</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">935.493316</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">944.628320</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">1102.492902</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">1011.428801</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">993.369153</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">703.366663</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">749.229000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">711.055019</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">744.360315</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">602.694088</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">539.095644</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">568.529023</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">715.919453</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">723.040511</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">713.534783</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">795.564123</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000093</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">18.748818</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">26.367729</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">28.740489</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">18.171262</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">17.023064</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">19.681817</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">21.696499</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">1.748123</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">1.159551</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">1.244226</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">1.286737</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">1.859175</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">1.536283</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">1.490581</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">1.446475</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">3.138448</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">3.362518</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">3.197714</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">3.445495</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">1.960168</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">1.812696</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">2.626110</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">2.401155</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">1.641559</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">1.120127</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">1.520160</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.123672</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.064585</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.118030</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.062646</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000094</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.114764</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.197628</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.237980</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.067150</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.050183</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.025762</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.020946</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.023273</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.020613</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.020391</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.253309</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.359092</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.191826</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.280499</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.644665</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.508711</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.638906</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">1.610289</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">2.043695</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">1.513517</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">1.061169</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000103</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.019350</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.019932</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.024280</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.027595</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.022570</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.020990</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.049946</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000120</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">18.187330</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">17.621063</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">16.764680</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">65.095814</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">63.742121</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">70.050947</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">65.437767</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">7.138900</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">6.728019</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">6.471314</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">7.243701</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">3.043655</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">3.779647</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">4.010854</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">3.292345</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">1.055312</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">1.336741</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">1.401802</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">1.323000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">7.879629</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">8.472306</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">7.826161</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">8.344422</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">1.302789</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">1.494076</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">1.100786</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">2.720008</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">2.666598</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">3.080750</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">7.171139</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000125</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">9.426741</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">9.885487</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">5.404755</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">2.614957</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">1.888344</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">2.419112</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">2.543413</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">1.836920</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">2.186343</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">2.081065</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">2.033197</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">1.815480</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">1.714785</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">1.591194</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">1.707502</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">1.388731</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">1.150249</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">1.311930</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">1.489738</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">2.613648</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">2.668347</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">2.770951</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">2.336912</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">1.482033</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">1.217406</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">1.473157</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.168666</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.267824</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.162984</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.439311</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000126</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">1.215801</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.746274</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.814387</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.893695</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.726967</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.795409</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.811422</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">1.698931</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.816782</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">1.232844</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">1.175287</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.776041</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.671299</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.527801</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.856885</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.881084</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.587611</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.480369</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.599140</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">1.920124</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">1.992736</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">2.643161</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">2.442442</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">1.243716</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.787857</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">1.154595</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">1.480120</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">1.685512</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">1.662030</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">1.757216</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000127</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">13.113348</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">14.260035</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">14.074681</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">16.162555</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">17.427828</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">17.565289</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">11.863558</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">12.271221</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">14.241964</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">13.551247</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">14.347367</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">12.813797</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">12.260097</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">11.685471</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">12.283357</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">8.849005</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">8.757533</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">8.775956</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">9.371477</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">16.306853</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">15.553249</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">15.896542</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">15.505971</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">11.131771</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">11.417468</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">12.892073</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">12.916885</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">12.422329</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">12.138589</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">11.783150</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000131</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">55.485697</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">54.448611</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">62.549063</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">71.704310</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">71.301569</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">71.997625</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">85.422041</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">53.653944</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">54.569370</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">54.041498</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">53.203853</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">56.560171</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">55.599344</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">57.503726</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">56.142399</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">56.062512</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">54.730314</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">56.114317</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">55.657729</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">64.772660</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">60.737488</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">64.559072</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">65.908557</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">73.304601</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">82.616101</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">81.113216</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">72.032652</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">72.128976</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">70.421851</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">64.545350</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000134</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">33.480962</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">36.683455</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">38.983454</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">39.609181</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">45.133664</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">47.577079</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">38.693647</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">19.425951</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">20.279171</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">18.258503</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">19.466904</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">20.759852</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">19.514517</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">20.453484</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">23.945999</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">24.145265</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">23.148770</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">24.341792</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">20.775309</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">26.254515</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">25.201055</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">25.298568</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">24.448624</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">70.194375</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">70.469434</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">72.039951</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">62.710030</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">65.426174</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">62.960782</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">52.892113</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000142</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">34.556890</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">33.051268</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">31.249676</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">26.910897</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">25.255541</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">26.237510</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">24.678957</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">4.314812</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">3.851946</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">5.201375</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">4.297327</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">3.944854</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">3.709407</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">4.698792</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">4.180144</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">4.008057</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">4.408832</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">5.413859</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">4.183791</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">12.235464</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">12.443578</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">13.543412</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">12.406199</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">5.611935</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">4.639913</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">5.140094</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">3.517411</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">3.338406</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">3.652103</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">5.866998</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000148</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">12.975017</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">12.265602</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">13.479207</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">15.929561</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">15.998245</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">15.251781</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">16.439377</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">11.187122</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">11.765327</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">12.255481</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">11.805331</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">9.727571</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">10.811203</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">10.842452</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">9.828969</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">9.057784</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">8.261829</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">8.418036</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">8.912263</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">13.212425</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">11.883479</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">13.518251</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">13.482611</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">10.102323</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">10.474313</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">9.593967</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">7.725926</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">8.271340</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">8.221738</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">8.493223</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000149</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">62.786926</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">60.575599</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">64.868495</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">81.035249</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">77.612600</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">80.007932</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">79.501053</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">19.769167</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">19.068923</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">18.725772</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">19.826181</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">17.829466</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">16.300954</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">17.513712</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">15.040879</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">14.602690</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">13.900191</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">14.489153</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">13.408245</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">37.005858</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">35.055680</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">36.987634</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">36.668932</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">36.844849</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">38.723659</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">38.975206</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">27.466288</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">29.294849</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">29.622090</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">30.355845</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000154</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.333055</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.115844</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.432376</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.325385</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.332173</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.555266</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.274624</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.315355</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.635393</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.370886</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.117576</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.114637</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.156711</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.361648</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.316280</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.277836</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.297838</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.396606</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.245718</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.372255</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.472991</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.428795</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.411684</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">1.678056</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">1.611784</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">1.901671</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">3.967049</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">5.087375</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">4.540421</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">5.049930</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000157</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.019660</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.038957</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000159</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.066624</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.120014</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.192008</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.069363</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.465487</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.133256</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.105527</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.609742</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.232339</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.034548</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.047125</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.116238</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.039351</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.258603</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.235119</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.181094</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.033496</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000167</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">7.167658</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">7.111275</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">4.562918</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">4.611452</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">5.165249</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">3.980233</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">5.810633</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">3.014607</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">2.165885</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">1.420868</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">2.402707</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">1.174127</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">1.731610</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">2.442734</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">1.718371</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">1.983554</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">1.487518</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.867104</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">1.640726</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">2.877345</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">4.086346</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">2.980909</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">2.788065</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">2.412113</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">1.093266</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">2.134765</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">3.075103</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">2.544267</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">4.048963</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">1.194254</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000168</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">46.898364</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">51.544129</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">50.803506</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">46.728899</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">46.416810</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">43.668901</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">42.770089</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">135.597979</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">142.174183</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">140.466776</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">155.267598</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">176.558772</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">177.449941</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">173.533809</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">177.441385</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">140.085372</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">135.955772</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">149.522280</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">139.289347</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">77.093701</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">65.875673</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">72.031057</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">69.113490</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">65.284363</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">59.589716</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">67.122029</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">66.846252</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">65.352465</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">66.011315</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">60.027737</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000171</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">183.510028</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">191.339737</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">185.366419</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">146.485489</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">136.055305</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">134.774362</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">144.419003</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">250.624457</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">258.445965</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">254.599788</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">264.936389</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">297.223823</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">290.955520</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">287.995065</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">284.714197</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">228.171116</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">236.310898</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">224.511224</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">221.322408</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">180.849179</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">173.186896</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">175.786723</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">175.691677</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">147.487631</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">156.118401</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">153.095971</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">219.561253</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">219.763695</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">210.634119</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">196.601985</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000182</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.103193</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.032365</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.029772</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.109766</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.257429</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.112760</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.211044</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.082172</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.037902</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.110168</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.111682</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.380279</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.139486</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.081007</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.246593</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.169495</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.356141</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.264438</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.269860</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.040837</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.041344</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.036939</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.030240</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.203803</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000183</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.010340</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.010401</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.016126</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.015047</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.012499</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.012861</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.020984</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000184</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">142.747177</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">147.826378</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">149.197724</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">66.761615</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">63.335841</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">65.070470</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">48.113890</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">21.348042</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">17.384228</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">15.935255</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">15.653769</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">16.292530</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">14.504585</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">17.797905</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">15.544638</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">17.701942</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">13.309930</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">15.605609</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">17.667904</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">28.101151</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">33.067019</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">38.044163</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">34.866641</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">7.046037</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">5.154655</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">5.428973</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">9.002990</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">11.536859</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">8.437719</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">2.867368</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000194</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">36.103417</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">38.818908</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">42.800890</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">36.817530</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">38.667199</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">36.778395</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">26.187279</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">33.002131</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">37.727627</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">36.606405</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">37.474231</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">36.703396</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">34.536615</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">36.214178</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">34.713799</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">33.020334</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">26.550992</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">28.899042</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">31.118602</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">34.300556</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">32.621947</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">33.363838</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">34.012681</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">30.084083</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">26.911440</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">26.198168</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">28.291703</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">30.043730</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">28.726287</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">29.337298</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000197</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.724701</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.864508</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.538785</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">1.073368</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">1.074458</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.792837</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.844630</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">25.556904</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">27.172721</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">27.223684</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">27.307364</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">23.469370</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">25.193771</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">24.809004</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">25.595933</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">22.134882</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">17.538889</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">21.238627</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">22.714445</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">11.528675</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">11.227085</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">11.667388</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">11.717720</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.717834</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.544315</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.843883</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.333168</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.479806</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.295036</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.135287</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000202</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">46.873410</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">38.860603</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">20.595529</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">21.277139</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">19.304139</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">22.041307</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">20.637502</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">1.998099</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">2.664860</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">2.147623</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">1.671951</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.707986</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">1.171352</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">2.805967</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.762410</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">1.275708</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.275591</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.391964</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.762712</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">3.703893</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">4.510447</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">4.937764</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">4.553973</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">1.186139</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">1.366082</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.695439</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">1.176641</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">2.441639</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.983961</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">3.130113</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000204</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.077812</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.036758</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.087585</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.043964</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.018414</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.038316</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.050157</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.041803</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.040884</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.152281</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000214</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.316821</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.376734</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.580427</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.074393</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.077525</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.232658</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.047076</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.195838</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.050855</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.059494</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.028719</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.080425</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.133600</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.051463</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.062615</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.191165</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.206326</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.128695</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.070217</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000215</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.201293</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.079614</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.125601</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.060125</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.339217</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.526094</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000216</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.055074</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.051534</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.015706</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.031705</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.039904</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.022940</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.019738</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.020080</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.018491</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.038279</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.017680</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.017147</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.017771</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.022180</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.065627</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.033124</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.015572</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000223</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">1.490207</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">1.396739</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">1.636890</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">1.736106</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">1.971587</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">1.840778</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">1.365031</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">2.454484</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">2.567922</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">2.499309</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">2.498017</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">2.092696</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">1.983752</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">2.066465</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">1.717463</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">2.653865</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">1.954384</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">1.841953</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">2.723702</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">2.868183</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">2.988016</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">3.715324</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">3.097974</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.185306</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.237818</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.133284</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.229376</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.237803</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.222297</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.190730</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000244</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.083692</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.054689</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.130041</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.090199</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.193026</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.120789</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.083430</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.218939</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.038914</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.099444</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.015106</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.013459</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.154390</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.067568</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.270073</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.127634</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.108335</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.113842</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.088430</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.052551</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.638876</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.595938</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.471887</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.533838</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.596377</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.453464</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.350619</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000247</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">3.581850</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">3.499372</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">3.528227</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">2.710486</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">3.101592</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">2.767801</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">2.565610</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">1.225779</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.659762</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">1.060458</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.978868</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.964519</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.809062</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.716354</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.546474</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.468901</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.592905</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.252756</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.383217</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">9.739157</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">11.367096</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">12.204956</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">9.050891</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">1.025635</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">1.058490</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.835800</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.563193</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.906406</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.423688</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.336354</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000248</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.307460</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.275004</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.037225</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.726755</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.032926</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.094631</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.027475</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.264993</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.027614</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.035735</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000253</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">5.002777</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">4.525335</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">4.179621</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">5.048770</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">4.947710</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">4.666987</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">4.505384</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">23.658215</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">24.517931</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">25.751619</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">26.217720</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">35.486540</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">36.812237</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">35.029205</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">34.464815</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">47.977399</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">51.307180</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">44.104269</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">43.421043</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">18.585852</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">19.591512</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">21.542503</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">19.528295</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">10.439729</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">10.042434</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">12.184047</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">20.222381</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">19.607781</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">19.986088</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">25.072968</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000263</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">2.981638</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">1.970632</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">2.113011</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">1.671629</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">1.341986</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">2.387899</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.558066</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">43.069555</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">53.099363</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">55.631124</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">57.681339</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">85.004519</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">87.323404</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">79.087829</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">83.067736</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">94.715692</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">60.793459</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">96.932514</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">95.108836</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">3.780805</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">3.893267</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">3.860449</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">4.027712</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.037352</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.037589</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.032486</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.056548</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.079220</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.058177</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.109525</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000266</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">4.207659</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">3.886346</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">3.670926</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">5.967348</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">5.313591</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">5.564873</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">3.908537</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">18.434130</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">19.178063</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">18.998609</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">19.210836</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">22.110790</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">21.232814</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">21.788512</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">20.701333</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">21.877453</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">18.154458</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">24.446192</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">24.809788</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">14.273999</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">14.445732</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">13.674834</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">13.532920</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.403795</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.353641</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.384367</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.045010</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.072257</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.094877</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.126511</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000275</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">15.337964</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">15.667032</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">22.966957</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">6.220162</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">6.831170</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">6.443813</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">6.385162</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">1.564104</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">1.618634</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">1.642415</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">1.574792</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">1.154512</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">1.511644</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">1.017289</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.828477</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">1.030165</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.983866</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.759897</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">1.061297</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">3.119988</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">3.554271</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">3.629310</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">3.486077</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">44.342670</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">44.274541</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">45.952267</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">68.204400</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">66.534540</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">67.546482</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">65.233133</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000276</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">6.707699</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">6.335487</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">5.715570</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">11.190985</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">11.445650</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">10.558276</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">8.857024</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">23.079722</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">24.157868</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">24.513742</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">24.306458</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">22.041553</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">22.124420</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">21.233254</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">22.742842</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">28.209391</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">23.221775</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">27.581952</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">27.546310</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">20.847164</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">20.360543</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">19.165914</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">19.357199</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">7.914284</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">8.048394</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">8.766309</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">11.622329</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">11.302601</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">11.791095</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">7.286859</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000278</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">90.617387</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">96.681487</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">113.010324</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">37.035350</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">35.574970</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">37.008561</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">35.981529</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">53.884610</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">53.796869</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">52.070052</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">55.294585</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">71.168345</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">71.952532</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">69.563003</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">72.057016</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">80.149296</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">90.857958</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">89.634870</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">77.269610</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">38.043631</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">38.898064</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">39.409246</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">38.148701</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">174.002655</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">172.278617</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">171.035085</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">162.763884</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">161.548267</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">159.085443</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">194.152958</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000000282</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">28.352789</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">28.584065</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">28.490442</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">51.536750</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">48.043432</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">53.508818</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">41.034434</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">34.048141</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">39.276596</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">38.718342</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">42.472822</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">33.504992</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">34.355608</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">38.392612</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">31.313721</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">28.164786</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">28.592877</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">31.653213</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">29.223527</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">46.532177</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">45.670087</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">48.495228</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">50.032227</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">27.125277</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">25.598833</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">24.739579</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">18.692252</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">16.789268</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">15.102999</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">17.659619</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="tpm-as-an-expression-metric" class="slide level2 smaller">
<h2>TPM as an expression metric</h2>
<p>Alright, not bad!</p>
<p>Let’s stop and think for a minute about what <code>tximport</code> did and the metric we are using (TPM). What does <em>transcripts per million</em> mean? Well, it means pretty much what it sounds like. For every million transcripts in the cell, X of them are this particular transcript. Importantly, this means when this TPM value was calculated from the number of <em>counts</em> a transcript received, this number had to be adjusted for both the total number of counts in the library and the length of a transcript.</p>
<p>If sample A had twice the number of total counts as sample B (i.e.&nbsp;was sequenced twice as deeply), then you would expect every transcript to have approximately twice the number of counts in sample A as it has in sample B. Similarly, if transcript X is twice as long as transcript Y, then you would expect that if they were equally expressed (i.e.&nbsp;the same number of transcript X and transcript Y molecules were present in the sample) that X would have approximately twice the counts that Y does. Working with expression units of TPM incorporates both of these normalizations.</p>
<p>So, if a TPM of X means that for every million transcripts in the sample that X of them were the transcript of interest, then the sum of TPM values across all species should equal one million, right?</p>
<p>Let’s check and see if that’s true.</p>
</section>
<section id="tpm-as-an-expression-metric-1" class="slide level2 smaller">
<h2>TPM as an expression metric</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a></a><span class="fu">sum</span>(tpms<span class="sc">$</span>DIVminus8.Rep1)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 989007</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a></a><span class="fu">sum</span>(tpms<span class="sc">$</span>DIVminus8.Rep2)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 988591.2</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a></a><span class="fu">sum</span>(tpms<span class="sc">$</span>DIVminus8.Rep3)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 988471</code></pre>
</div>
</div>
<p>OK, not quite one million, but pretty darn close.</p>
<p>This notion that TPMs represent proportions of a whole also leads to another interesting insight into what <code>tximport</code> is doing here. If all transcripts belong to genes, then the TPM for a gene must be the sum of the TPMs of its transcripts. Can we verify that that is true?</p>
</section>
<section id="tpm-as-an-expression-metric-2" class="slide level2 smaller">
<h2>TPM as an expression metric</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a></a><span class="co"># Redefine for clarity in comparisons</span></span>
<span id="cb30-2"><a></a>tpms.genes <span class="ot">&lt;-</span> tpms</span>
<span id="cb30-3"><a></a></span>
<span id="cb30-4"><a></a><span class="co"># Make a new tximport object, but this time instead</span></span>
<span id="cb30-5"><a></a><span class="co"># of giving gene expression values, give transcript expression values</span></span>
<span id="cb30-6"><a></a><span class="co"># This is controlled by the `txOut` argument</span></span>
<span id="cb30-7"><a></a>txi.transcripts <span class="ot">&lt;-</span> <span class="fu">tximport</span>(</span>
<span id="cb30-8"><a></a>  salmdir,</span>
<span id="cb30-9"><a></a>  <span class="at">type =</span> <span class="st">"salmon"</span>,</span>
<span id="cb30-10"><a></a>  <span class="at">tx2gene =</span> t2g,</span>
<span id="cb30-11"><a></a>  <span class="at">dropInfReps =</span> <span class="cn">TRUE</span>,</span>
<span id="cb30-12"><a></a>  <span class="at">countsFromAbundance =</span> <span class="st">"lengthScaledTPM"</span>,</span>
<span id="cb30-13"><a></a>  <span class="at">txOut =</span> <span class="cn">TRUE</span></span>
<span id="cb30-14"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>reading in files with read_tsv</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 </code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb33"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a></a><span class="co"># Make a table of tpm values for every transcript</span></span>
<span id="cb33-2"><a></a>tpms.txs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(txi.transcripts<span class="sc">$</span>abundance) <span class="sc">|&gt;</span></span>
<span id="cb33-3"><a></a>  <span class="fu">set_names</span>(<span class="fu">rownames</span>(metadata)) <span class="sc">|&gt;</span></span>
<span id="cb33-4"><a></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"ensembl_transcript_id"</span>) <span class="sc">|&gt;</span></span>
<span id="cb33-5"><a></a>  <span class="fu">inner_join</span>(t2g, ., <span class="at">by =</span> <span class="st">"ensembl_transcript_id"</span>)</span>
<span id="cb33-6"><a></a></span>
<span id="cb33-7"><a></a><span class="fu">gt</span>(tpms.txs[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, ])</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="zdburukdtm" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#zdburukdtm table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#zdburukdtm thead, #zdburukdtm tbody, #zdburukdtm tfoot, #zdburukdtm tr, #zdburukdtm td, #zdburukdtm th {
  border-style: none;
}

#zdburukdtm p {
  margin: 0;
  padding: 0;
}

#zdburukdtm .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#zdburukdtm .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#zdburukdtm .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#zdburukdtm .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#zdburukdtm .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#zdburukdtm .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#zdburukdtm .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#zdburukdtm .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#zdburukdtm .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#zdburukdtm .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#zdburukdtm .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#zdburukdtm .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#zdburukdtm .gt_spanner_row {
  border-bottom-style: hidden;
}

#zdburukdtm .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#zdburukdtm .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#zdburukdtm .gt_from_md > :first-child {
  margin-top: 0;
}

#zdburukdtm .gt_from_md > :last-child {
  margin-bottom: 0;
}

#zdburukdtm .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#zdburukdtm .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#zdburukdtm .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#zdburukdtm .gt_row_group_first td {
  border-top-width: 2px;
}

#zdburukdtm .gt_row_group_first th {
  border-top-width: 2px;
}

#zdburukdtm .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#zdburukdtm .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#zdburukdtm .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#zdburukdtm .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#zdburukdtm .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#zdburukdtm .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#zdburukdtm .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#zdburukdtm .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#zdburukdtm .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#zdburukdtm .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#zdburukdtm .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#zdburukdtm .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#zdburukdtm .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#zdburukdtm .gt_left {
  text-align: left;
}

#zdburukdtm .gt_center {
  text-align: center;
}

#zdburukdtm .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#zdburukdtm .gt_font_normal {
  font-weight: normal;
}

#zdburukdtm .gt_font_bold {
  font-weight: bold;
}

#zdburukdtm .gt_font_italic {
  font-style: italic;
}

#zdburukdtm .gt_super {
  font-size: 65%;
}

#zdburukdtm .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#zdburukdtm .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#zdburukdtm .gt_indent_1 {
  text-indent: 5px;
}

#zdburukdtm .gt_indent_2 {
  text-indent: 10px;
}

#zdburukdtm .gt_indent_3 {
  text-indent: 15px;
}

#zdburukdtm .gt_indent_4 {
  text-indent: 20px;
}

#zdburukdtm .gt_indent_5 {
  text-indent: 25px;
}

#zdburukdtm .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#zdburukdtm div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="ensembl_transcript_id" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ensembl_transcript_id</th>
<th id="DIV0.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV0.Rep1</th>
<th id="DIV0.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV0.Rep2</th>
<th id="DIV0.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV0.Rep3</th>
<th id="DIV1.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV1.Rep1</th>
<th id="DIV1.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV1.Rep2</th>
<th id="DIV1.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV1.Rep3</th>
<th id="DIV1.Rep4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV1.Rep4</th>
<th id="DIV16.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV16.Rep1</th>
<th id="DIV16.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV16.Rep2</th>
<th id="DIV16.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV16.Rep3</th>
<th id="DIV16.Rep4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV16.Rep4</th>
<th id="DIV21.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV21.Rep1</th>
<th id="DIV21.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV21.Rep2</th>
<th id="DIV21.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV21.Rep3</th>
<th id="DIV21.Rep4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV21.Rep4</th>
<th id="DIV28.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV28.Rep1</th>
<th id="DIV28.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV28.Rep2</th>
<th id="DIV28.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV28.Rep3</th>
<th id="DIV28.Rep4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV28.Rep4</th>
<th id="DIV7.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV7.Rep1</th>
<th id="DIV7.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV7.Rep2</th>
<th id="DIV7.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV7.Rep3</th>
<th id="DIV7.Rep4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIV7.Rep4</th>
<th id="DIVminus4.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus4.Rep1</th>
<th id="DIVminus4.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus4.Rep2</th>
<th id="DIVminus4.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus4.Rep3</th>
<th id="DIVminus8.Rep1" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus8.Rep1</th>
<th id="DIVminus8.Rep2" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus8.Rep2</th>
<th id="DIVminus8.Rep3" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus8.Rep3</th>
<th id="DIVminus8.Rep4" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">DIVminus8.Rep4</th>
<th id="ensembl_gene_id" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ensembl_gene_id</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000119854</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">2.286511</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">2.475351</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">2.778492</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">3.752931</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">4.290970</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">3.606815</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">2.289835</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">4.753009</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">6.407206</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">6.093523</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">8.073422</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">4.114218</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">3.854100</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">4.504607</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">4.233634</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">3.062028</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">2.657892</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">4.551532</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">7.708040</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">5.771621</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">4.913897</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">5.733164</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.821894</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.791545</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.452225</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.283685</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.253438</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000037736</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000147408</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000052516</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000070080</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">17.294246</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">16.642412</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">20.145840</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">22.849371</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">23.111531</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">23.055949</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">18.922899</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">85.742059</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">92.095921</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">92.061439</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">90.117232</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">80.253664</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">82.962866</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">82.837521</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">83.218155</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">68.286276</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">59.179619</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">65.655743</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">67.027119</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">61.090866</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">60.764495</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">60.258015</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">58.545413</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">22.716225</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">23.042382</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">25.299848</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">18.289952</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">19.497363</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">17.597384</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">22.126199</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000056124</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000033908</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.277638</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.771701</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.734236</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.835938</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">1.130580</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">1.774933</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.262763</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.231432</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.209738</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.302866</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.342360</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.394527</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.196355</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.088545</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.170725</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.088308</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.148182</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.037937</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.361527</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.124041</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.411704</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.216135</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.384765</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.627833</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.412680</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.393967</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.177899</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.097658</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.091573</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.404344</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000031511</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000174016</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">8.545914</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">6.122679</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">4.978280</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">13.429257</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">13.448801</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">13.789999</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">9.139241</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">3.772152</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">6.493719</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">6.454788</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">7.620481</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">3.964981</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">3.769810</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">4.272302</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">4.711036</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">4.523138</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">4.025177</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">6.303732</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">5.551957</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">20.404943</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">18.886851</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">26.036222</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">24.703049</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">15.936284</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">17.060844</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">18.859078</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">9.623711</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">12.255722</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">10.591347</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">2.846295</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000036036</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000082868</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">5.240588</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">7.435392</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">21.577246</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">3.157229</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">8.363363</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">2.106274</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">7.631468</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">4.013754</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000064802</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000187399</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.112263</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000090925</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000159512</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">1.197789</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">1.783770</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">1.592837</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">3.586005</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">2.904509</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">3.830286</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">1.389749</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">1.361614</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">1.953179</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">1.801017</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">2.061461</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">2.504494</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">1.787667</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">1.311947</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">1.985731</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">2.389696</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.801072</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">1.741500</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">1.538186</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.880757</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">2.094465</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.624705</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">1.609800</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.248865</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.954488</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">1.411125</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">1.029973</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.838168</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.218119</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.135338</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000029207</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000194793</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.068955</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.061078</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.064820</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.078959</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.177114</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.299510</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.089493</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.181351</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.080516</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.310951</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.222825</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.324322</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.070346</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.163243</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.468088</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.067476</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.360582</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.332645</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.068630</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.155780</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.076557</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000103497</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000221298</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.077941</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.100916</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.090854</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.309668</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.244962</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.239357</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.299259</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.174734</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.291636</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.193253</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.231279</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.296867</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.117060</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.200170</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.109928</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.266300</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.103878</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.301950</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.395391</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.569643</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.215417</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.462677</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.353049</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.028680</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.050635</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000113688</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000138551</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.119676</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000034164</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000210944</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000110312</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000194146</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.066535</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.185273</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.055779</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.070382</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.078488</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.129887</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.162839</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.143064</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.058096</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000102375</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000112725</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">1.270589</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">1.049486</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">1.271776</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">1.294794</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">1.619005</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">1.511612</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.893824</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.688353</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.884466</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.635204</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.709094</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.928998</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.846144</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.829175</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.696928</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.989624</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">1.018670</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.818393</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">1.100337</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">1.270363</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.878781</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">1.156467</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">1.281586</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">1.667594</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">1.633202</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">1.868757</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">1.477511</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">1.274160</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">1.640671</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">1.306233</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000025269</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000181391</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">2.222061</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">1.416381</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">2.497798</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">3.006018</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">3.142753</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">1.571165</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">3.167750</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">3.530355</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">2.965603</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">4.649879</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">3.558337</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">3.465148</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">3.203126</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">3.106205</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">2.862499</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">1.333758</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">2.039999</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">3.282519</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">5.710520</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">3.937353</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">6.847600</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">3.991743</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">1.733049</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.496494</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">1.601051</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">1.270107</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">1.507055</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">1.477476</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000030446</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000223290</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.026984</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.032844</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.033378</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.033128</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.031158</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.072496</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.032133</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.031972</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.029910</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.036096</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.029574</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000035954</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000116191</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.044746</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.104528</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.168647</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.046020</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.048814</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.102273</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.042109</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.069164</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.111104</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.170086</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000080078</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000060782</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.267962</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.170311</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.391316</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.446440</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.389636</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.087233</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.046855</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.110542</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.048208</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.044187</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.083067</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.328807</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000051716</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000141479</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.960027</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.541858</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.473587</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">1.080745</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.274066</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.359059</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">1.129914</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.300957</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.388096</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.801308</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">1.141492</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">1.285565</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.314952</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">1.575395</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.348874</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.398863</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.970236</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">1.421541</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.351342</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000049119</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="ensembl_transcript_id">ENSMUST00000211096</td>
<td class="gt_row gt_right" headers="DIV0.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV0.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV1.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV16.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV21.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV28.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIV7.Rep4">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus4.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep1">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep2">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep3">0.000000</td>
<td class="gt_row gt_right" headers="DIVminus8.Rep4">0.000000</td>
<td class="gt_row gt_left" headers="ensembl_gene_id">ENSMUSG00000110353</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<p>OK so lets look at the expression of ENSMUSG00000020634 in the first sample (DIVminus8.Rep1).</p>
</section>
<section id="tpm-as-an-expression-metric-3" class="slide level2 smaller">
<h2>TPM as an expression metric</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a></a><span class="co"># Get sum of tpm values for transcripts that belong to ENSMUSG00000020634</span></span>
<span id="cb34-2"><a></a>tpms.tx.ENSMUSG00000020634 <span class="ot">&lt;-</span> <span class="fu">filter</span>(</span>
<span id="cb34-3"><a></a>  tpms.txs,</span>
<span id="cb34-4"><a></a>  ensembl_gene_id <span class="sc">==</span> <span class="st">"ENSMUSG00000020634"</span></span>
<span id="cb34-5"><a></a>)</span>
<span id="cb34-6"><a></a></span>
<span id="cb34-7"><a></a>sumoftxtpm <span class="ot">&lt;-</span> <span class="fu">sum</span>(tpms.tx.ENSMUSG00000020634<span class="sc">$</span>DIVminus8.Rep1)</span>
<span id="cb34-8"><a></a></span>
<span id="cb34-9"><a></a><span class="co"># Get gene level tpm value of ENSMUSG00000020634</span></span>
<span id="cb34-10"><a></a>genetpm <span class="ot">&lt;-</span> <span class="fu">filter</span>(</span>
<span id="cb34-11"><a></a>  tpms.genes,</span>
<span id="cb34-12"><a></a>  ensembl_gene_id <span class="sc">==</span> <span class="st">"ENSMUSG00000020634"</span></span>
<span id="cb34-13"><a></a>) <span class="sc">|&gt;</span></span>
<span id="cb34-14"><a></a>  <span class="fu">pull</span>(DIVminus8.Rep1)</span>
<span id="cb34-15"><a></a></span>
<span id="cb34-16"><a></a><span class="co"># Are they the same?</span></span>
<span id="cb34-17"><a></a>sumoftxtpm</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 46.71414</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a></a>genetpm</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 46.71414</code></pre>
</div>
</div>
</section>
<section id="basic-rnaseq-qc" class="slide level2 smaller">
<h2>Basic RNAseq QC</h2>
<p>OK now that we’ve got expression values for all genes, we now might want to use these expression values to learn a little bit about our samples. One simple question is &gt; Are replicates similar to each other, or at least more similar to each other than to other samples?</p>
<p>If our data is worth anything at all, we would hope that differences between replicates, which are supposed to be drawn from the same condition, are smaller than differences between samples drawn from different conditions. If that’s not true, it could indicate that one replicate is very different from other replciates (in which case we might want to remove it), or that the data in general is of poor quality.</p>
<p>Another question is:</p>
<blockquote>
<p>How similar is each sample to every other sample?</p>
</blockquote>
<p>In our timecourse, we might expect that samples drawn from adjacent timepoints might be more similar to each other than samples from more distant timepoints.</p>
</section>
<section id="hierarchical-clustering" class="slide level2 smaller">
<h2>Hierarchical clustering</h2>
<p>A simple way to think about this is to simply correlate TPM values for genes between samples. For plotting purposes here, let’s plot the log(TPM) of two samples against each other. However, for the actual correlation coefficient we are going to be using the <em>Spearman</em> correlation method, which uses ranks, not absolute values. This means that whether or not you take the log will have no effect on the Spearman correlation coefficient.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb38"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a></a><span class="co"># DIVminus8.Rep1 vs DIVminus8.Rep2</span></span>
<span id="cb38-2"><a></a></span>
<span id="cb38-3"><a></a><span class="co"># Since we are plotting log TPM values, we need to add a pseudocount to all samples.</span></span>
<span id="cb38-4"><a></a><span class="co"># log(0) is a problem.</span></span>
<span id="cb38-5"><a></a></span>
<span id="cb38-6"><a></a><span class="co"># Add pseudocounts and take log within ggplot function call</span></span>
<span id="cb38-7"><a></a>r.spearman <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(</span>
<span id="cb38-8"><a></a>  tpms<span class="sc">$</span>DIVminus8.Rep1,</span>
<span id="cb38-9"><a></a>  tpms<span class="sc">$</span>DIVminus8.Rep2,</span>
<span id="cb38-10"><a></a>  <span class="at">method =</span> <span class="st">"spearman"</span></span>
<span id="cb38-11"><a></a>) <span class="sc">|&gt;</span></span>
<span id="cb38-12"><a></a>  broom<span class="sc">::</span><span class="fu">tidy</span>() <span class="sc">|&gt;</span></span>
<span id="cb38-13"><a></a>  <span class="fu">pull</span>(estimate)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="hierarchical-clustering-output" class="slide level2 smaller output-location-slide"><h2>Hierarchical clustering</h2><div class="cell output-location-slide">
<div class="cell-output cell-output-stderr">
<pre><code>Warning in cor.test.default(tpms$DIVminus8.Rep1,
tpms$DIVminus8.Rep2, method = "spearman"): Cannot compute
exact p-value with ties</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb40"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a></a>r.spearman <span class="ot">&lt;-</span> <span class="fu">signif</span>(r.spearman, <span class="dv">2</span>)</span>
<span id="cb40-2"><a></a></span>
<span id="cb40-3"><a></a><span class="fu">ggplot</span>(</span>
<span id="cb40-4"><a></a>  tpms,</span>
<span id="cb40-5"><a></a>  <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log10</span>(DIVminus8.Rep1 <span class="sc">+</span> <span class="fl">1e-3</span>), <span class="at">y =</span> <span class="fu">log10</span>(DIVminus8.Rep2 <span class="sc">+</span> <span class="fl">1e-3</span>))</span>
<span id="cb40-6"><a></a>) <span class="sc">+</span></span>
<span id="cb40-7"><a></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb40-8"><a></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb40-9"><a></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">2</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"R = "</span>, r.spearman))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-23_files/figure-revealjs/cor-to-samples-1.png" alt="Description of the plot - PLEASE FILL IN" width="960"></p>
</figure>
</div>
</div>
</div></section><section id="hierarchical-clustering-1" class="slide level2 smaller">
<h2>Hierarchical clustering</h2>
<p>With RNAseq data, the variance of a gene’s expression increases as the expression increases. However, using a pseudocount and taking the log of the expression value actually reverses this trend. Now, genes with the lowest expression have the most variance. Why is this a problem? Well, the genes with the most variance are going to be the ones that contribute the most to intersample differences. Ideally, we would like to therefore remove the relationship between expression and variance.</p>
<p>There are transformations, notably <code>rlog</code> and <code>vst</code>, that are made to deal with this, but they are best used when dealing with normalized <strong>count</strong> data, while here we are dealing with TPMs. We will talk about counts later, but not here.</p>
<p>So, for now, we will take another approach of simply using an expression threshold. Any gene that does not meet our threshold will be excluded from the analysis. Obviously where to set this threshold is a bit subjective. For now, we will set this cutoff at 1 TPM.</p>
</section>
<section id="hierarchical-clustering-2" class="slide level2 smaller">
<h2>Hierarchical clustering</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb41"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a></a><span class="co"># DIVminus8.Rep1 vs DIVminus8.Rep2</span></span>
<span id="cb41-2"><a></a></span>
<span id="cb41-3"><a></a><span class="co"># Since we are plotting log TPM values we will only keep for genes that have expression of at least 1 TPM in both samples</span></span>
<span id="cb41-4"><a></a>tpms<span class="fl">.2</span>samplecor <span class="ot">&lt;-</span></span>
<span id="cb41-5"><a></a>  <span class="fu">select</span>(tpms, <span class="fu">c</span>(ensembl_gene_id, DIVminus8.Rep1, DIVminus8.Rep2)) <span class="sc">|&gt;</span></span>
<span id="cb41-6"><a></a>  <span class="fu">filter</span>(DIVminus8.Rep1 <span class="sc">&gt;=</span> <span class="dv">1</span> <span class="sc">&amp;</span> DIVminus8.Rep2 <span class="sc">&gt;=</span> <span class="dv">1</span>)</span>
<span id="cb41-7"><a></a></span>
<span id="cb41-8"><a></a><span class="co"># pull the correlation coefficient</span></span>
<span id="cb41-9"><a></a>r.spearman <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(</span>
<span id="cb41-10"><a></a>  tpms<span class="fl">.2</span>samplecor<span class="sc">$</span>DIVminus8.Rep1,</span>
<span id="cb41-11"><a></a>  tpms<span class="fl">.2</span>samplecor<span class="sc">$</span>DIVminus8.Rep2,</span>
<span id="cb41-12"><a></a>  <span class="at">method =</span> <span class="st">"spearman"</span></span>
<span id="cb41-13"><a></a>) <span class="sc">|&gt;</span></span>
<span id="cb41-14"><a></a>  broom<span class="sc">::</span><span class="fu">tidy</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-15"><a></a>  <span class="fu">pull</span>(estimate)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="hierarchical-clustering-2-output" class="slide level2 smaller output-location-slide"><h2>Hierarchical clustering</h2><div class="cell output-location-slide">
<div class="cell-output cell-output-stderr">
<pre><code>Warning in cor.test.default(tpms.2samplecor$DIVminus8.Rep1,
tpms.2samplecor$DIVminus8.Rep2, : Cannot compute exact
p-value with ties</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb43"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a></a><span class="co"># round/set sig digits</span></span>
<span id="cb43-2"><a></a>r.spearman <span class="ot">&lt;-</span> <span class="fu">signif</span>(r.spearman, <span class="dv">2</span>)</span>
<span id="cb43-3"><a></a></span>
<span id="cb43-4"><a></a><span class="co"># plot</span></span>
<span id="cb43-5"><a></a><span class="fu">ggplot</span>(</span>
<span id="cb43-6"><a></a>  tpms<span class="fl">.2</span>samplecor,</span>
<span id="cb43-7"><a></a>  <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log10</span>(DIVminus8.Rep1 <span class="sc">+</span> <span class="fl">1e-3</span>), <span class="at">y =</span> <span class="fu">log10</span>(DIVminus8.Rep2 <span class="sc">+</span> <span class="fl">1e-3</span>))</span>
<span id="cb43-8"><a></a>) <span class="sc">+</span></span>
<span id="cb43-9"><a></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb43-10"><a></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb43-11"><a></a>  <span class="fu">annotate</span>(</span>
<span id="cb43-12"><a></a>    <span class="st">"text"</span>,</span>
<span id="cb43-13"><a></a>    <span class="at">x =</span> <span class="dv">2</span>,</span>
<span id="cb43-14"><a></a>    <span class="at">y =</span> <span class="dv">1</span>,</span>
<span id="cb43-15"><a></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"R = "</span>, r.spearman)</span>
<span id="cb43-16"><a></a>  )</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-23_files/figure-revealjs/cor-to-sample-exp-1.png" alt="Description of the plot - PLEASE FILL IN" width="960"></p>
</figure>
</div>
</div>
</div></section><section id="filtering-lowly-expressed-genes" class="slide level2 smaller">
<h2>Filtering lowly expressed genes</h2>
<p>OK that’s two samples compared to each other, but now we want to see how <strong>all</strong> samples compare to <strong>all</strong> other samples. Before we do this we need to decide how to apply our expression cutoff across many samples. Should a gene have to meet the cutoff in only one sample? In all samples? Let’s start by saying it has to meet the cutoff in at least half of the 30 samples.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb44"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a></a><span class="co"># Make a new column in tpms that is the number of samples in which the value is at least 1</span></span>
<span id="cb44-2"><a></a>tpms.cutoff <span class="ot">&lt;-</span></span>
<span id="cb44-3"><a></a>  <span class="fu">mutate</span>(tpms, <span class="at">nSamples =</span> <span class="fu">rowSums</span>(tpms[, <span class="dv">2</span><span class="sc">:</span><span class="dv">31</span>] <span class="sc">&gt;</span> <span class="dv">1</span>)) <span class="sc">|&gt;</span></span>
<span id="cb44-4"><a></a>  <span class="co"># Now filter for rows where nSamples is at least 15</span></span>
<span id="cb44-5"><a></a>  <span class="co"># Meaning that at least 15 samples passed the threshold</span></span>
<span id="cb44-6"><a></a>  <span class="fu">filter</span>(nSamples <span class="sc">&gt;=</span> <span class="dv">15</span>) <span class="sc">|&gt;</span></span>
<span id="cb44-7"><a></a>  <span class="co"># Get rid of the nSamples column</span></span>
<span id="cb44-8"><a></a>  <span class="fu">select</span>(<span class="sc">-</span>nSamples)</span>
<span id="cb44-9"><a></a></span>
<span id="cb44-10"><a></a><span class="fu">nrow</span>(tpms)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 49972</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb46"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a></a><span class="fu">nrow</span>(tpms.cutoff)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 14861</code></pre>
</div>
</div>
</section>
<section id="correlating-gene-expression-values" class="slide level2 smaller">
<h2>Correlating gene expression values</h2>
<p>Now we can use the <code>cor</code> function to calculate pairwise correlations in a .red[<em>matrix</em>] of TPM values.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb48"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a></a>tpms.cutoff.matrix <span class="ot">&lt;-</span></span>
<span id="cb48-2"><a></a>  <span class="fu">select</span>(tpms.cutoff, <span class="sc">-</span>ensembl_gene_id) <span class="sc">|&gt;</span></span>
<span id="cb48-3"><a></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb48-4"><a></a></span>
<span id="cb48-5"><a></a>tpms.cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(tpms.cutoff.matrix, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb48-6"><a></a><span class="fu">head</span>(tpms.cor)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>          DIV0.Rep1 DIV0.Rep2 DIV0.Rep3 DIV1.Rep1 DIV1.Rep2
DIV0.Rep1 1.0000000 0.9922453 0.9857834 0.9348484 0.9324194
DIV0.Rep2 0.9922453 1.0000000 0.9895779 0.9294218 0.9273449
DIV0.Rep3 0.9857834 0.9895779 1.0000000 0.9175882 0.9153932
DIV1.Rep1 0.9348484 0.9294218 0.9175882 1.0000000 0.9933149
DIV1.Rep2 0.9324194 0.9273449 0.9153932 0.9933149 1.0000000
DIV1.Rep3 0.9321014 0.9273853 0.9164250 0.9915135 0.9913159
          DIV1.Rep3 DIV1.Rep4 DIV16.Rep1 DIV16.Rep2
DIV0.Rep1 0.9321014 0.9344590  0.6680186  0.6549872
DIV0.Rep2 0.9273853 0.9304106  0.6614933  0.6465640
DIV0.Rep3 0.9164250 0.9177944  0.6380326  0.6218017
DIV1.Rep1 0.9915135 0.9858579  0.7550841  0.7423875
DIV1.Rep2 0.9913159 0.9853574  0.7571130  0.7443644
DIV1.Rep3 1.0000000 0.9865138  0.7564301  0.7425891
          DIV16.Rep3 DIV16.Rep4 DIV21.Rep1 DIV21.Rep2
DIV0.Rep1  0.6555308  0.6562956  0.6434782  0.6441234
DIV0.Rep2  0.6480255  0.6483917  0.6369659  0.6371799
DIV0.Rep3  0.6229298  0.6233139  0.6126546  0.6136543
DIV1.Rep1  0.7434990  0.7440758  0.7237427  0.7238040
DIV1.Rep2  0.7460841  0.7461203  0.7255483  0.7256123
DIV1.Rep3  0.7448212  0.7445382  0.7244319  0.7242751
          DIV21.Rep3 DIV21.Rep4 DIV28.Rep1 DIV28.Rep2
DIV0.Rep1  0.6461267  0.6426477  0.6390216  0.6421053
DIV0.Rep2  0.6391854  0.6342137  0.6316861  0.6352214
DIV0.Rep3  0.6143880  0.6096429  0.6085960  0.6117384
DIV1.Rep1  0.7263283  0.7222009  0.7164272  0.7143724
DIV1.Rep2  0.7282509  0.7249480  0.7185337  0.7169766
DIV1.Rep3  0.7263667  0.7224427  0.7167805  0.7144093
          DIV28.Rep3 DIV28.Rep4 DIV7.Rep1 DIV7.Rep2
DIV0.Rep1  0.6344281  0.6355676 0.7742758 0.7792020
DIV0.Rep2  0.6266482  0.6285669 0.7671712 0.7722675
DIV0.Rep3  0.6034204  0.6050626 0.7436562 0.7494654
DIV1.Rep1  0.7108448  0.7130572 0.8503631 0.8519662
DIV1.Rep2  0.7134794  0.7153368 0.8521781 0.8532264
DIV1.Rep3  0.7115690  0.7131355 0.8506309 0.8521874
          DIV7.Rep3 DIV7.Rep4 DIVminus4.Rep1 DIVminus4.Rep2
DIV0.Rep1 0.7772718 0.7758796      0.8317332      0.8289624
DIV0.Rep2 0.7701116 0.7693336      0.8360947      0.8336978
DIV0.Rep3 0.7468141 0.7458387      0.8518416      0.8504546
DIV1.Rep1 0.8521139 0.8515258      0.7370960      0.7355276
DIV1.Rep2 0.8537124 0.8523451      0.7335145      0.7310733
DIV1.Rep3 0.8526356 0.8523464      0.7344799      0.7318674
          DIVminus4.Rep3 DIVminus8.Rep1 DIVminus8.Rep2
DIV0.Rep1      0.8301511      0.7957126      0.7965574
DIV0.Rep2      0.8346960      0.8022918      0.8031270
DIV0.Rep3      0.8519312      0.8186949      0.8190062
DIV1.Rep1      0.7360888      0.6994479      0.7000579
DIV1.Rep2      0.7326038      0.6962439      0.6965824
DIV1.Rep3      0.7343008      0.6973234      0.6975552
          DIVminus8.Rep3 DIVminus8.Rep4
DIV0.Rep1      0.7963088      0.7849475
DIV0.Rep2      0.8015284      0.7899987
DIV0.Rep3      0.8192264      0.8069155
DIV1.Rep1      0.7003414      0.6908235
DIV1.Rep2      0.6959884      0.6867789
DIV1.Rep3      0.6974818      0.6878911</code></pre>
</div>
</div>
</section>
<section id="hierarchical-clustering-3" class="slide level2 smaller">
<h2>Hierarchical clustering</h2>
<p>Now we need to plot these and have similar samples (i.e.&nbsp;those that are highly correlated with each other) be clustered near to each other. We will use <code>pheatmap</code> to do this.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb50"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a></a><span class="co"># let's pull information that we want to add as categories</span></span>
<span id="cb50-2"><a></a><span class="fu">pheatmap</span>(</span>
<span id="cb50-3"><a></a>  tpms.cor,</span>
<span id="cb50-4"><a></a>  <span class="at">annotation_col =</span> metadata[, <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>],</span>
<span id="cb50-5"><a></a>  <span class="at">fontsize =</span> <span class="dv">7</span>,</span>
<span id="cb50-6"><a></a>  <span class="at">show_colnames =</span> <span class="cn">FALSE</span></span>
<span id="cb50-7"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

<p>This looks pretty good! There are two main points to takeaway here. First, all replicates for a given timepoint are clustering with each other. Second, you can kind of derive the order of the timepoints from the clustering. The biggest separation is between early (DIVminus8 to DIV1) and late (DIV7 to DIV28). After that you can then see finer-grained structure.</p>
</section>
<section id="hierarchical-clustering-3-output" class="slide level2 smaller output-location-slide"><h2>Hierarchical clustering</h2><div class="cell output-location-slide">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-23_files/figure-revealjs/make-heatmap-1.png" width="960"></p>
</figure>
</div>
</div>
</div></section><section id="pca-analysis" class="slide level2 smaller">
<h2>PCA analysis</h2>
<p>Another way to visualize relationships is using a dimensionality reduction technique called Principal Component Analysis (PCA). Let’s watch this short video. It focuses more on how to interpret them rather than the math behind their creation.</p>
<iframe data-external="1" src="https://www.youtube.com/embed/HMOI_lkzW08" title="" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe>
</section>
<section id="pca-analysis-1" class="slide level2 smaller">
<h2>PCA analysis</h2>
<p>PCA works best when values are approximately normally distributed, so we will first take the log of our expression values.</p>
<p>With our cutoff as it is now (genes have to have expression of at least 1 TPM in half the samples), it is possible that we will have some 0 values. Taking the log of 0 might cause a problem, so we will add a pseudocount.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb51"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a></a>tpms.cutoff.matrix <span class="ot">&lt;-</span></span>
<span id="cb51-2"><a></a>  <span class="fu">select</span>(tpms.cutoff, <span class="sc">-</span>ensembl_gene_id) <span class="sc">|&gt;</span></span>
<span id="cb51-3"><a></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb51-4"><a></a></span>
<span id="cb51-5"><a></a><span class="co"># Add pseudocount and take log2</span></span>
<span id="cb51-6"><a></a>tpms.cutoff.matrix <span class="ot">&lt;-</span> <span class="fu">log2</span>(tpms.cutoff.matrix <span class="sc">+</span> <span class="fl">1e-3</span>)</span>
<span id="cb51-7"><a></a></span>
<span id="cb51-8"><a></a><span class="co"># scale - annoying double transpose</span></span>
<span id="cb51-9"><a></a>tpms.cutoff.matrix <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(tpms.cutoff.matrix)))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="pca-analysis-2" class="slide level2 smaller">
<h2>PCA analysis</h2>
<p>Very similar interpretation as before (heatmap of correlation).</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb52"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a></a><span class="co"># prcomp expects samples to be rownames, right now they are columns</span></span>
<span id="cb52-2"><a></a><span class="co"># so we need to transpose the matrix using `t`</span></span>
<span id="cb52-3"><a></a>tpms.pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(tpms.cutoff.matrix))</span>
<span id="cb52-4"><a></a></span>
<span id="cb52-5"><a></a><span class="co"># The coordinates of samples on the principle components are stored in the $x slot</span></span>
<span id="cb52-6"><a></a><span class="co"># These are what we are going to use to plot</span></span>
<span id="cb52-7"><a></a><span class="co"># We can also also some data about the samples here so that our plot is a little more interesting</span></span>
<span id="cb52-8"><a></a></span>
<span id="cb52-9"><a></a><span class="do">### Tricky piping!!</span></span>
<span id="cb52-10"><a></a>tpms.pca.pc <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(tpms.pca<span class="sc">$</span>x) <span class="sc">|&gt;</span></span>
<span id="cb52-11"><a></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_id"</span>) <span class="sc">|&gt;</span></span>
<span id="cb52-12"><a></a>  <span class="fu">left_join</span>(metadata[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">by =</span> <span class="st">"sample_id"</span>)</span>
<span id="cb52-13"><a></a></span>
<span id="cb52-14"><a></a></span>
<span id="cb52-15"><a></a><span class="co"># We can see how much of the total variance is explained by each PC using the summary function</span></span>
<span id="cb52-16"><a></a>tpms.pca.summary <span class="ot">&lt;-</span> <span class="fu">summary</span>(tpms.pca)<span class="sc">$</span>importance</span>
<span id="cb52-17"><a></a></span>
<span id="cb52-18"><a></a><span class="co"># The amount of variance explained by PC1 is the second row, first column of this table</span></span>
<span id="cb52-19"><a></a><span class="co"># It's given as a fraction of 1, so we multiply it by 100 to get a percentage</span></span>
<span id="cb52-20"><a></a>pc1var <span class="ot">&lt;-</span> <span class="fu">round</span>(tpms.pca.summary[<span class="dv">2</span>, <span class="dv">1</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>)</span>
<span id="cb52-21"><a></a></span>
<span id="cb52-22"><a></a><span class="co"># The amount of variance explained by PC2 is the second row, second column of this table</span></span>
<span id="cb52-23"><a></a>pc2var <span class="ot">&lt;-</span> <span class="fu">round</span>(tpms.pca.summary[<span class="dv">2</span>, <span class="dv">2</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>)</span>
<span id="cb52-24"><a></a></span>
<span id="cb52-25"><a></a><span class="co"># Reorder timepoints explicitly for plotting purposes</span></span>
<span id="cb52-26"><a></a></span>
<span id="cb52-27"><a></a>tpms.pca.pc<span class="sc">$</span>samp <span class="ot">&lt;-</span></span>
<span id="cb52-28"><a></a>  <span class="fu">factor</span>(</span>
<span id="cb52-29"><a></a>    tpms.pca.pc<span class="sc">$</span>samp,</span>
<span id="cb52-30"><a></a>    <span class="at">levels =</span> <span class="fu">c</span>(</span>
<span id="cb52-31"><a></a>      <span class="st">"DIVminus8"</span>,</span>
<span id="cb52-32"><a></a>      <span class="st">"DIVminus4"</span>,</span>
<span id="cb52-33"><a></a>      <span class="st">"DIV0"</span>,</span>
<span id="cb52-34"><a></a>      <span class="st">"DIV1"</span>,</span>
<span id="cb52-35"><a></a>      <span class="st">"DIV7"</span>,</span>
<span id="cb52-36"><a></a>      <span class="st">"DIV16"</span>,</span>
<span id="cb52-37"><a></a>      <span class="st">"DIV21"</span>,</span>
<span id="cb52-38"><a></a>      <span class="st">"DIV28"</span></span>
<span id="cb52-39"><a></a>    )</span>
<span id="cb52-40"><a></a>  )</span>
<span id="cb52-41"><a></a></span>
<span id="cb52-42"><a></a><span class="co"># Plot results</span></span>
<span id="cb52-43"><a></a><span class="fu">ggplot</span>(</span>
<span id="cb52-44"><a></a>  <span class="at">data =</span> tpms.pca.pc,</span>
<span id="cb52-45"><a></a>  <span class="fu">aes</span>(</span>
<span id="cb52-46"><a></a>    <span class="at">x =</span> PC1,</span>
<span id="cb52-47"><a></a>    <span class="at">y =</span> PC2,</span>
<span id="cb52-48"><a></a>    <span class="at">color =</span> samp,</span>
<span id="cb52-49"><a></a>    <span class="at">label =</span> sample_id</span>
<span id="cb52-50"><a></a>  )</span>
<span id="cb52-51"><a></a>) <span class="sc">+</span></span>
<span id="cb52-52"><a></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb52-53"><a></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb52-54"><a></a>  <span class="fu">theme_cowplot</span>(<span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb52-55"><a></a>  <span class="fu">labs</span>(</span>
<span id="cb52-56"><a></a>    <span class="at">x =</span> <span class="fu">paste</span>(<span class="st">"PC1,"</span>, pc1var, <span class="st">"% explained var."</span>),</span>
<span id="cb52-57"><a></a>    <span class="at">y =</span> <span class="fu">paste</span>(<span class="st">"PC2,"</span>, pc2var, <span class="st">"% explained var."</span>)</span>
<span id="cb52-58"><a></a>  ) <span class="sc">+</span></span>
<span id="cb52-59"><a></a>  <span class="fu">geom_text_repel</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>

</div>


<img data-src="slides-23_files/figure-revealjs/make-pca-1.png" width="960" class="r-stretch"></section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">
<p><a href="https://rnabioco.github.io/molb-7950">Course 🔗</a></p>
</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': true,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"8\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdn.jsdelivr.net/npm/mathjax@2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
          const outerScaffold = trigger.parentElement.cloneNode(true);
          const codeEl = outerScaffold.querySelector('code');
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp("https:\/\/rnabioco\.github\.io\/molb-7950\/");
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>