<!DOCTYPE html>
<html lang="en"><head>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/tabby.min.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-8f4e3aabaa80e3e650ae20f2c24f966b.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.8.24">

  <meta name="author" content="Jay Hesselberth">
  <meta name="dcterms.date" content="2025-09-23">
  <title>MOLB 7950 – Mapping chromatin structure and transactions</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    html { -webkit-text-size-adjust: 100%; }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #97947a;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #97947a;  padding-left: 4px; }
    div.sourceCode
      { color: #f8f8f2; background-color: #2b2b2b; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #f8f8f2; } /* Normal */
    code span.al { color: #dcc6e0; } /* Alert */
    code span.an { color: #d4d0ab; } /* Annotation */
    code span.at { color: #ffd700; } /* Attribute */
    code span.bn { color: #dcc6e0; } /* BaseN */
    code span.bu { color: #f5ab35; } /* BuiltIn */
    code span.cf { color: #ffa07a; } /* ControlFlow */
    code span.ch { color: #abe338; } /* Char */
    code span.cn { color: #ffa07a; } /* Constant */
    code span.co { color: #d4d0ab; } /* Comment */
    code span.cv { color: #d4d0ab; font-style: italic; } /* CommentVar */
    code span.do { color: #d4d0ab; font-style: italic; } /* Documentation */
    code span.dt { color: #dcc6e0; } /* DataType */
    code span.dv { color: #dcc6e0; } /* DecVal */
    code span.er { color: #dcc6e0; } /* Error */
    code span.ex { color: #ffd700; } /* Extension */
    code span.fl { color: #f5ab35; } /* Float */
    code span.fu { color: #ffd700; } /* Function */
    code span.im { } /* Import */
    code span.in { color: #d4d0ab; } /* Information */
    code span.kw { color: #ffa07a; } /* Keyword */
    code span.op { color: #00e0e0; } /* Operator */
    code span.ot { color: #ffa07a; } /* Other */
    code span.pp { color: #dcc6e0; } /* Preprocessor */
    code span.sc { color: #00e0e0; } /* SpecialChar */
    code span.ss { color: #abe338; } /* SpecialString */
    code span.st { color: #abe338; } /* String */
    code span.va { color: #f5ab35; } /* Variable */
    code span.vs { color: #abe338; } /* VerbatimString */
    code span.wa { color: #d4d0ab; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="../site_libs/revealjs/dist/theme/quarto-8fafb62820c35118272e29c1027a293f.css">
  <link href="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
<meta property="og:title" content="Mapping chromatin structure and transactions – MOLB 7950">
<meta property="og:site_name" content="MOLB 7950">
<meta name="twitter:title" content="Mapping chromatin structure and transactions – MOLB 7950">
<meta name="twitter:image" content="https://rnabioco.github.io/molb-7950/slides/images/twitter-card.png">
<meta name="twitter:creator" content="@rnabioco">
<meta name="twitter:card" content="summary_large_image">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Mapping chromatin structure and transactions</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Jay Hesselberth 
</div>
        <p class="quarto-title-affiliation">
            RNA Bioscience Initiative | CU Anschutz
          </p>
    </div>
</div>

  <p class="date">2025-09-23</p>
</section>
<section id="final-projects" class="slide level2">
<h2>Final Projects</h2>
<p>Throughout the DNA and RNA blocks we will refer to several data sets that might be good starting points for your final project (worth 20% of your grade).</p>
<p>We will ask for a sketch of the rough plan for a final project by the end of week 7 (Wed Oct 8). In addition, if you plan to work in a group, we’d like to know who you will be working with.</p>
<p><strong>Final projects will be due Monday Oct 27. We will schedule short (5 minute) talks by each group on Oct 27 and Oct 29.</strong></p>
<ul>
<li><a href="https://rnabioco.github.io/molb-7950/course-info/final-projects.html">Final project overview</a></li>
<li><a href="https://rnabioco.github.io/molb-7950/resources/block-dna-resources.html">DNA block resources</a></li>
</ul>
</section>
<section>
<section id="gene-regulation-by-chromatin" class="title-slide slide level1 center">
<h1>Gene regulation by chromatin</h1>

</section>
<section id="chromatin-organization-regulates-genome-compaction-and-gene-expression" class="slide level2">
<h2>Chromatin organization regulates genome compaction and gene expression</h2>
<div class="quarto-layout-panel" data-layout-ncol="3">
<div class="quarto-layout-row quarto-layout-valign-center">
<div class="quarto-layout-cell" style="flex-basis: 33.3%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../img/block-dna/ctcf.jpg"></p>
<figcaption>Genome organization</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 33.3%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../img/block-dna/promoter.png"></p>
<figcaption>Cell signaling</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 33.3%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../img/block-dna/org-dev.jpg"></p>
<figcaption>Organism development</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="dna-accessibility-drives-all-biochemical-transactions-on-the-genome" class="slide level2">
<h2>DNA accessibility drives all biochemical transactions on the genome</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li class="fragment">Transcription Initiation</li>
<li class="fragment">Transcription Elongation</li>
<li class="fragment">DNA Repair</li>
<li class="fragment">Initiation of DNA Replication</li>
<li class="fragment">Recombination</li>
<li class="fragment">Viral Integration</li>
</ul>
</div><div class="column" style="width:50%;">
<p><img data-src="../img/block-dna/domino.gif"></p>
</div></div>
</section>
<section id="mapping-measurement-are-first-steps-toward-understanding." class="slide level2 smaller">
<h2>Mapping &amp; measurement are first steps toward understanding.</h2>
<div class="columns">
<div class="column" style="width:60%;">
<p>Before genome-wide DNA accessibility measurements, we knew about chromatin transactions at only a handful of loci.</p>
<p>This was a classic “keys under the lamppost” situation, leading to general models of chromatin-based gene regulation.</p>
</div><div class="column" style="width:40%;">
<p><img data-src="../img/block-dna/street-lamp.jpg" style="width:100.0%"></p>
</div></div>

<aside><div>
<p>Photo by <a href="https://unsplash.com/@justusmenke?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">Justus Menke</a> on <a href="https://unsplash.com/photos/PvQtCNm4teI?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">Unsplash</a></p>
</div></aside></section>
<section id="using-micrococcal-nuclease-mnase-to-map-chromatin" class="slide level2 smaller">
<h2>Using micrococcal nuclease (MNase) to map chromatin</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>Micrococcal nuclease is an endo/exonuclease from <em>Staphylococcus aureus</em></li>
<li>Efficient, <strong>multiple turnover</strong> enzyme that digests accessible DNA (&amp; RNA)</li>
<li>Dependent on calcium ions</li>
</ul>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../img/block-dna/mnase-overview.png" class="quarto-figure quarto-figure-center"></p>
</figure>
</div>
</div></div>
</section>
<section id="using-atac-to-map-chromatin" class="slide level2 smaller">
<h2>Using ATAC to map chromatin</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>Tn5 transposase catalyzes “cut-and-paste” insertion of DNA into a target</li>
<li>In ATAC (“Assay for Transposase-Accessible Chromatin”), the transposase enzymes are loaded with DNA sequencing adaptors (blue and red in the image), so the products of transposition are ready to PCR.</li>
<li><strong>Single turnover enzyme</strong> that acts on accessible DNA.</li>
<li>Requires about ~60 bp of accessible for transposition.</li>
</ul>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="../img/block-dna/atac-explain.png" class="quarto-figure quarto-figure-center"></p>
</figure>
</div>
</div></div>
</section>
<section id="chromatin-mapping-experimental-summary" class="slide level2 smaller">
<h2><strong>Chromatin mapping experimental summary</strong></h2>
<table class="caption-top">
<colgroup>
<col style="width: 12%">
<col style="width: 23%">
<col style="width: 46%">
<col style="width: 16%">
</colgroup>
<thead>
<tr class="header">
<th></th>
<th style="text-align: center;">DNase-seq</th>
<th style="text-align: center;">ATAC-seq</th>
<th style="text-align: center;">MNase-seq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Genome representation</strong></td>
<td style="text-align: center;">Most active regions</td>
<td style="text-align: center;">Most active regions</td>
<td style="text-align: center;">Whole genome</td>
</tr>
<tr class="even">
<td><strong>Ease of experiment</strong></td>
<td style="text-align: center;">Very difficult</td>
<td style="text-align: center;">Easy peasy</td>
<td style="text-align: center;">One day’s work</td>
</tr>
<tr class="odd">
<td><strong>What is profiled?</strong></td>
<td style="text-align: center;">Accessible DNA, “footprints” at low cut frequency</td>
<td style="text-align: center;">Accessible DNA. Not really “footprints”, single turnover enzyme, so fragments are not informative</td>
<td style="text-align: center;">Protections of TFs and nucleosomes</td>
</tr>
</tbody>
</table>
</section></section>
<section>
<section id="experimental-details" class="title-slide slide level1 center">
<h1>Experimental details</h1>

</section>
<section id="section" class="slide level2">
<h2></h2>

<img data-src="../img/block-dna/footprinting-exp-overview.png" class="r-stretch"></section>
<section id="section-1" class="slide level2">
<h2></h2>

<img data-src="../img/block-dna/atac-input-comparison.png" class="r-stretch"></section></section>
<section>
<section id="data-structures-and-tools-used-for-analysis-of-chromatin-mapping-experiments" class="title-slide slide level1 center">
<h1>Data structures and tools used for analysis of chromatin mapping experiments</h1>

</section>
<section id="experimental-workflow" class="slide level2">
<h2>Experimental workflow</h2>

<img data-src="../img/block-dna/acc-workflow.png" class="r-stretch"></section>
<section id="bed-format" class="slide level2 smaller">
<h2>BED format</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>Contains information about genomic intervals.</li>
<li>Used to represent genomic features (exons, introns, transcription start sites)</li>
<li>First 3 columns are: <code>chrom</code>, <code>start</code>, <code>end</code></li>
<li>Next 3 are: <code>name</code>, <code>score</code>, <code>strand</code>. Strand can be <code>+</code>, <code>-</code>, or <code>.</code> (no strand)</li>
</ul>
</div><div class="column" style="width:50%;">
<pre><code>chr7 127473530 127474697 Pos3 0  +
chr7 127474697 127475864 Pos4 0  +
chr7 127475864 127477031 Neg1 0  -
chr7 127477031 127478198 Neg2 0  -</code></pre>
</div></div>
</section>
<section id="wig-bedgraph" class="slide level2 smaller">
<h2>WIG / bedGraph</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li><p>WIG and bedGraph store interval <em>signals</em>.</p>
<ul>
<li>Genome sequencing coverage</li>
<li>Coverage of reads from MNase / ATAC-seq</li>
</ul></li>
<li><p>Many studies will provide genome-scale data in these formats</p></li>
</ul>
</div><div class="column" style="width:50%;">
<pre><code>chr19 49302000 49302300 -1.0
chr19 49302300 49302600 -0.75
chr19 49302600 49302900 -0.50
chr19 49302900 49303200 -0.25</code></pre>
</div></div>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p>bigWig is a binary form of WIG, used to store large amounts of signal in a compressed, indexed format.</p>
</div>
</div>
</div>
</section>
<section id="interval-analysis" class="slide level2 smaller">
<h2>Interval analysis</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>The primary tool in the genome interval analysis is <a href="https://bedtools.readthedocs.io/en/latest/">BEDtools</a> – it’s the Swiss-army knife of internal analysis.</p>
<p>We will use an R package called <a href="https://rnabioco.github.io/valr/">valr</a> that provides the same tools, but you don’t need to leave RStudio. valr provides the same tools for reading and manipulating genome intervals.</p>
<p><code>bed_intersect()</code> is a fundamental operation. It identifies intervals from two tibbles that intersect and reports their overlaps.</p>
<p>Let’s take a look at that it does.</p>
</div><div class="column" style="width:50%;">
<p><img data-src="../img/block-dna/intersect-glyph.png"></p>
</div></div>
</section>
<section id="bed_intersect-example" class="slide level2 smaller">
<h2><code>bed_intersect()</code> example</h2>
<p>First, we’ll define some example intervals in <code>x</code> and <code>y</code>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a></a><span class="fu">library</span>(valr)</span>
<span id="cb3-2"><a></a><span class="fu">library</span>(dplyr)</span>
<span id="cb3-3"><a></a></span>
<span id="cb3-4"><a></a>x <span class="ot">&lt;-</span> <span class="fu">tribble</span>(</span>
<span id="cb3-5"><a></a>  <span class="sc">~</span>chrom, <span class="sc">~</span>start, <span class="sc">~</span>end,</span>
<span id="cb3-6"><a></a>  <span class="st">"chr1"</span>, <span class="dv">25</span>, <span class="dv">50</span>,</span>
<span id="cb3-7"><a></a>  <span class="st">"chr1"</span>, <span class="dv">100</span>, <span class="dv">125</span></span>
<span id="cb3-8"><a></a>)</span>
<span id="cb3-9"><a></a></span>
<span id="cb3-10"><a></a>y <span class="ot">&lt;-</span> <span class="fu">tribble</span>(</span>
<span id="cb3-11"><a></a>  <span class="sc">~</span>chrom, <span class="sc">~</span>start, <span class="sc">~</span>end,</span>
<span id="cb3-12"><a></a>  <span class="st">"chr1"</span>, <span class="dv">30</span>,     <span class="dv">75</span></span>
<span id="cb3-13"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section class="slide level2">

<p>Visual representation of the intersecting intervals.</p>

<img data-src="slides-16_files/figure-revealjs/bed_intersect-glyph-1.png" width="960" class="r-stretch"></section>
<section class="slide level2">

<p>Tibble of the intersecting intervals.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a></a><span class="fu">bed_intersect</span>(x, y)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 6
  chrom start.x end.x start.y end.y .overlap
  &lt;chr&gt;   &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;    &lt;int&gt;
1 chr1       25    50      30    75       20</code></pre>
</div>
</div>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p>Note the addition of <code>.x</code> and <code>.y</code> suffixes to disamguate the input sources.</p>
</div>
</div>
</div>
</section>
<section id="read_bed-example" class="slide level2">
<h2><code>read_bed()</code> example</h2>
<p>Rse <code>read_bed()</code> to load genome annotations and signals.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a></a>snps <span class="ot">&lt;-</span> <span class="fu">read_bed</span>(</span>
<span id="cb6-2"><a></a>  <span class="fu">valr_example</span>(<span class="st">"hg19.snps147.chr22.bed.gz"</span>)</span>
<span id="cb6-3"><a></a>)</span>
<span id="cb6-4"><a></a></span>
<span id="cb6-5"><a></a>genes <span class="ot">&lt;-</span> <span class="fu">read_bed</span>(</span>
<span id="cb6-6"><a></a>  <span class="fu">valr_example</span>(<span class="st">"genes.hg19.chr22.bed.gz"</span>)</span>
<span id="cb6-7"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p>You’ll also use <code>read_bedgraph()</code> and <code>read_bigwig()</code> on your problem set.</p>
</div>
</div>
</div>
</section>
<section id="what-is-in-snps-and-genes" class="slide level2 smaller">
<h2>What is in <code>snps</code> and <code>genes</code>?</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a></a>snps</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10,000 × 6
   chrom    start      end name        score strand
   &lt;chr&gt;    &lt;int&gt;    &lt;int&gt; &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt; 
 1 chr22 16053247 16053248 rs587721086 0     +     
 2 chr22 16053443 16053444 rs80167676  0     +     
 3 chr22 16055964 16055965 rs587706951 0     +     
 4 chr22 16069373 16069374 rs2154787   0     +     
 5 chr22 16069782 16069783 rs1963212   0     +     
 6 chr22 16100513 16100514 rs8140563   0     +     
 7 chr22 16114622 16114623 rs558374111 0     +     
 8 chr22 16115181 16115182 rs567521927 0     +     
 9 chr22 16115377 16115378 rs538733846 0     +     
10 chr22 16121639 16121640 rs544560983 0     +     
# ℹ 9,990 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a></a>genes</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 732 × 6
   chrom    start      end name      score strand
   &lt;chr&gt;    &lt;int&gt;    &lt;int&gt; &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt; 
 1 chr22 16150259 16193004 AK022914  8     -     
 2 chr22 16162065 16172265 LINC00516 3     +     
 3 chr22 16179617 16181004 BC017398  1     -     
 4 chr22 16239287 16239327 DQ590589  1     +     
 5 chr22 16240245 16240277 DQ573684  1     -     
 6 chr22 16240300 16240340 DQ595048  1     -     
 7 chr22 16241085 16241125 DQ590589  1     +     
 8 chr22 16242753 16242785 DQ573684  1     -     
 9 chr22 16243382 16243414 DQ573684  1     -     
10 chr22 16243447 16243487 DQ587539  1     -     
# ℹ 722 more rows</code></pre>
</div>
</div>
</section>
<section id="interval-manipulation" class="slide level2 smaller">
<h2>Interval manipulation</h2>
<p>Let’s find and characterize intergenic SNPs. We’ll use the tools <a href="https://rnabioco.github.io/valr/reference/bed_subtract.html"><code>bed_substract()</code></a> and <a href="https://rnabioco.github.io/valr/reference/bed_closest.html"><code>bed_closest()</code></a>. Take a look and their examples in the valr documentation to see what they do.</p>
<p>Here, we’ll ask the question: <strong>“Which SNPs are intergenic, and how far are they from the nearest gene?”</strong></p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a></a><span class="co"># find snps in intergenic regions</span></span>
<span id="cb11-2"><a></a>intergenic <span class="ot">&lt;-</span> <span class="fu">bed_subtract</span>(snps, genes)</span>
<span id="cb11-3"><a></a></span>
<span id="cb11-4"><a></a><span class="co"># find distance from intergenic snps to nearest gene</span></span>
<span id="cb11-5"><a></a>nearby <span class="ot">&lt;-</span> <span class="fu">bed_closest</span>(intergenic, genes)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Take a look at the <code>intergenic</code> and <code>nearby</code> objects in the console.</p>
</section>
<section class="slide level2">

<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a></a>intergenic</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4,786 × 6
   chrom    start      end name        score strand
   &lt;chr&gt;    &lt;int&gt;    &lt;int&gt; &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt; 
 1 chr22 16053247 16053248 rs587721086 0     +     
 2 chr22 16053443 16053444 rs80167676  0     +     
 3 chr22 16055964 16055965 rs587706951 0     +     
 4 chr22 16069373 16069374 rs2154787   0     +     
 5 chr22 16069782 16069783 rs1963212   0     +     
 6 chr22 16100513 16100514 rs8140563   0     +     
 7 chr22 16114622 16114623 rs558374111 0     +     
 8 chr22 16115181 16115182 rs567521927 0     +     
 9 chr22 16115377 16115378 rs538733846 0     +     
10 chr22 16121639 16121640 rs544560983 0     +     
# ℹ 4,776 more rows</code></pre>
</div>
</div>
</section>
<section class="slide level2">

<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a></a>nearby</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4,798 × 13
   chrom  start.x    end.x name.x   score.x strand.x start.y
   &lt;chr&gt;    &lt;int&gt;    &lt;int&gt; &lt;chr&gt;    &lt;chr&gt;   &lt;chr&gt;      &lt;int&gt;
 1 chr22 16053247 16053248 rs58772… 0       +         1.62e7
 2 chr22 16053443 16053444 rs80167… 0       +         1.62e7
 3 chr22 16055964 16055965 rs58770… 0       +         1.62e7
 4 chr22 16069373 16069374 rs21547… 0       +         1.62e7
 5 chr22 16069782 16069783 rs19632… 0       +         1.62e7
 6 chr22 16100513 16100514 rs81405… 0       +         1.62e7
 7 chr22 16114622 16114623 rs55837… 0       +         1.62e7
 8 chr22 16115181 16115182 rs56752… 0       +         1.62e7
 9 chr22 16115377 16115378 rs53873… 0       +         1.62e7
10 chr22 16121639 16121640 rs54456… 0       +         1.62e7
# ℹ 4,788 more rows
# ℹ 6 more variables: end.y &lt;int&gt;, name.y &lt;chr&gt;,
#   score.y &lt;chr&gt;, strand.y &lt;chr&gt;, .overlap &lt;int&gt;,
#   .dist &lt;int&gt;</code></pre>
</div>
</div>
</section>
<section id="interval-manipulation-1" class="slide level2 smaller">
<h2>Interval manipulation</h2>
<p>Now that you have overlaps and distances between SNPs and genes, you can go back to dplyr tools to generate reports.</p>
<div class="cell columns column-output-location">
<div class="column">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a></a>nearby <span class="sc">|&gt;</span></span>
<span id="cb16-2"><a></a>  <span class="fu">select</span>(</span>
<span id="cb16-3"><a></a>    <span class="fu">starts_with</span>(<span class="st">"name"</span>),</span>
<span id="cb16-4"><a></a>    .overlap,</span>
<span id="cb16-5"><a></a>    .dist</span>
<span id="cb16-6"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb16-7"><a></a>  <span class="fu">filter</span>(<span class="fu">abs</span>(.dist) <span class="sc">&lt;</span> <span class="dv">5000</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div><div class="column fragment">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1,047 × 4
   name.x      name.y   .overlap .dist
   &lt;chr&gt;       &lt;chr&gt;       &lt;int&gt; &lt;int&gt;
 1 rs530458610 P704P           0  2579
 2 rs2261631   P704P           0  -268
 3 rs570770556 POTEH           0  -913
 4 rs538163832 POTEH           0  -953
 5 rs190224195 POTEH           0 -1399
 6 rs2379966   DQ571479        0  4750
 7 rs142687051 DQ571479        0  3558
 8 rs528403095 DQ571479        0  3309
 9 rs555126291 DQ571479        0  2745
10 rs5747567   DQ571479        0 -1778
# ℹ 1,037 more rows</code></pre>
</div>
</div></div>
</section>
<section id="bed_map-example" class="slide level2">
<h2><code>bed_map()</code> example</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><code>bed_map()</code> does two things in order:</p>
<ol type="1">
<li>It identifies intersecting intervals between <code>x</code> and <code>y</code></li>
<li>Calculates summary statistics based on the intersection</li>
</ol>
</div><div class="column" style="width:50%;">
<p><img data-src="../img/block-dna/map-glyph.png"></p>
</div></div>
<p>A typical use is to count up signals (e.g., coverage from an MNase-seq experiment) over specific regions (e.g., promoter regions).</p>
</section>
<section id="bed_map-example-1" class="slide level2 smaller">
<h2><code>bed_map()</code> example</h2>
<p>Let’s setup some example data.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a></a>x <span class="ot">&lt;-</span> <span class="fu">tribble</span>(</span>
<span id="cb18-2"><a></a>  <span class="sc">~</span>chrom, <span class="sc">~</span>start, <span class="sc">~</span>end,</span>
<span id="cb18-3"><a></a>  <span class="st">"chr1"</span>, <span class="dv">100</span>,    <span class="dv">250</span>,</span>
<span id="cb18-4"><a></a>  <span class="st">"chr1"</span>, <span class="dv">250</span>,    <span class="dv">500</span></span>
<span id="cb18-5"><a></a>)</span>
<span id="cb18-6"><a></a></span>
<span id="cb18-7"><a></a>y <span class="ot">&lt;-</span> <span class="fu">tribble</span>(</span>
<span id="cb18-8"><a></a>  <span class="sc">~</span>chrom, <span class="sc">~</span>start, <span class="sc">~</span>end, <span class="sc">~</span>value,</span>
<span id="cb18-9"><a></a>  <span class="st">"chr1"</span>, <span class="dv">100</span>,    <span class="dv">250</span>,  <span class="dv">10</span>,</span>
<span id="cb18-10"><a></a>  <span class="st">"chr1"</span>, <span class="dv">150</span>,    <span class="dv">250</span>,  <span class="dv">20</span>,</span>
<span id="cb18-11"><a></a>  <span class="st">"chr1"</span>, <span class="dv">250</span>,    <span class="dv">500</span>,  <span class="dv">500</span></span>
<span id="cb18-12"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="bed_map-example-continued" class="slide level2 smaller">
<h2><code>bed_map()</code> example continued</h2>
<p>First examine the intersecting intervals.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a></a><span class="fu">bed_intersect</span>(x, y)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 7
  chrom start.x end.x start.y end.y value.y .overlap
  &lt;chr&gt;   &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;    &lt;int&gt;
1 chr1      100   250     100   250      10      150
2 chr1      100   250     150   250      20      100
3 chr1      100   250     250   500     500        0
4 chr1      250   500     100   250      10        0
5 chr1      250   500     150   250      20        0
6 chr1      250   500     250   500     500      250</code></pre>
</div>
</div>
</section>
<section class="slide level2">

<p>Visual representation of <code>bed_map()</code></p>

<img data-src="slides-16_files/figure-revealjs/bed-map-glyph-1.png" width="960" class="r-stretch"></section>
<section class="slide level2">

<p>Computing multiple summaries with <code>bed_map()</code></p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a></a><span class="fu">bed_map</span>(</span>
<span id="cb21-2"><a></a>  x, y,</span>
<span id="cb21-3"><a></a>  <span class="at">.sum =</span> <span class="fu">sum</span>(value),</span>
<span id="cb21-4"><a></a>  <span class="at">.count =</span> <span class="fu">length</span>(value)</span>
<span id="cb21-5"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 5
  chrom start   end  .sum .count
  &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;int&gt;
1 chr1    100   250    30      2
2 chr1    250   500   500      1</code></pre>
</div>
</div>


</section></section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">
<p><a href="https://rnabioco.github.io/molb-7950">Course 🔗</a></p>
</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': true,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"8\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdn.jsdelivr.net/npm/mathjax@2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
          const outerScaffold = trigger.parentElement.cloneNode(true);
          const codeEl = outerScaffold.querySelector('code');
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp("https:\/\/rnabioco\.github\.io\/molb-7950\/");
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>