---
title: "Long-read sequencing platforms & applications"
author: "Jay Hesselberth"
---

## Business

- No homework this weekend.
- Single-cell mRNA-seq next week. 1 problem set.
- Final project presentations.

  * Would like to complete in 1 day. 5-7 minutes each.
  * Make Google Slides and send us a link.
  * How many groups?

- Survey at the end. Please fill it out, we want and need your feedback.

## Short & long read DNA sequencing {.smaller}

::: {layout-ncol=2 .nostretch}
![See the forest](../img/block-rna/leaves.png)

![for the trees](../img/block-rna/birds-eye.jpg)
:::

We have entered a new era (sorry, Swifties) wherein routine application of long-read sequencing is revealing new principles of gene regulation.


## Comparing data from short- and long-read experiments {.smaller}

![](../img/block-rna/short-long-read.jpg){fig-align="center"}

## Things that are invisible to short-read sequencing {.smaller}

**Genomic features greater 500 bp in size (i.e., the size of an Illumina fragment):**

- Patterns of pre-mRNA splicing (which exons / introns in a molecule)
- Structural genomic variants
- Chromatin structure across a single locus.

. . .

Patterns we have examined in class rely on patterns that emerge from *many, small* DNA or RNA fragments.

Long-read experiments examine *single molecules* and can capture patterns captured in *single* DNA fragments.

## Major platforms ‚Äî Pacific Biosciences {.smaller}

:::: {.columns}

::: {.column width="30%"}
- \>25 kb reads
- Low error rates achieved by "circular consensus"
- Only DNA (RNA used to be available üò¢)
:::

::: {.column width="70%"}
![](../img/block-rna/pacbio-seq.png){fig-align="center"}
:::

::::

::: aside
What enzyme would you immobilize to analyze RNA?
:::

## Major platforms - Oxford Nanopore {.smaller}

:::: {.columns}

::: {.column width="30%"}
- \>150 kb reads
- higher error rates (0.1-1%)
- detection of modified bases (5mC, RNA mods)
:::

::: {.column width="70%"}
![](../img/block-rna/nanopore-seq.png){width="80%" fig-align="center"}
:::

::::

::: aside
I ‚ù§Ô∏è nanopore sequencing -- data can be collected within minutes of generating a library.
:::

## DNA case study -- T2T {.smaller}

The T2T (telomere-to-telomere) consortium used long-read sequencing to define the first complete draft of a human genome, including telomeric and centromeric repeats.

![](../img/block-rna/long-read-dna-case-study.png){.absolute top=150 right=250 width="60%"}

## DNA case study - Fiber-seq {.smaller}

Stergachis et al. (2020) Nature

![](../img/block-rna/fiber-seq-fig-1.jpeg){.absolute top=100 right=200 width="70%"}

## RNA case study -- splicing patterns

![](../img/block-rna/churchman-fig-1.png){fig-align="center"}

## RNA case study -- co-transcriptional communication

![](../img/block-rna/hilgers-overview.jpg){.absolute top=120 right=250 width="50%"}

## RNA case study -- co-transcriptional communication

![](../img/block-rna/hilgers-fig-3.png){fig-align="center"}

::: aside
Alfonso-Gonzalez et al. (2023) Cell
:::

## Applications beyond long-read sequencing {.smaller}

The nanopore sequencing technology has been adapted for more than DNA sequencing. For example,
we can now analyze modifications on tRNA, where the modifiations leave unique signatures in 
the current traces.

::: {layout-ncol=2}

![](../img/block-rna/trna-seq.png)

![](../img/block-rna/aa-tRNA-sequencing.png)

:::



