<!DOCTYPE html>
<html lang="en"><head>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/tabby.min.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-8f4e3aabaa80e3e650ae20f2c24f966b.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.8.24">

  <meta name="author" content="Jay Hesselberth">
  <meta name="dcterms.date" content="2025-09-13">
  <title>MOLB 7950 – Factor-centric chromatin analysis</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    html { -webkit-text-size-adjust: 100%; }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #97947a;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #97947a;  padding-left: 4px; }
    div.sourceCode
      { color: #f8f8f2; background-color: #2b2b2b; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #f8f8f2; } /* Normal */
    code span.al { color: #dcc6e0; } /* Alert */
    code span.an { color: #d4d0ab; } /* Annotation */
    code span.at { color: #ffd700; } /* Attribute */
    code span.bn { color: #dcc6e0; } /* BaseN */
    code span.bu { color: #f5ab35; } /* BuiltIn */
    code span.cf { color: #ffa07a; } /* ControlFlow */
    code span.ch { color: #abe338; } /* Char */
    code span.cn { color: #ffa07a; } /* Constant */
    code span.co { color: #d4d0ab; } /* Comment */
    code span.cv { color: #d4d0ab; font-style: italic; } /* CommentVar */
    code span.do { color: #d4d0ab; font-style: italic; } /* Documentation */
    code span.dt { color: #dcc6e0; } /* DataType */
    code span.dv { color: #dcc6e0; } /* DecVal */
    code span.er { color: #dcc6e0; } /* Error */
    code span.ex { color: #ffd700; } /* Extension */
    code span.fl { color: #f5ab35; } /* Float */
    code span.fu { color: #ffd700; } /* Function */
    code span.im { } /* Import */
    code span.in { color: #d4d0ab; } /* Information */
    code span.kw { color: #ffa07a; } /* Keyword */
    code span.op { color: #00e0e0; } /* Operator */
    code span.ot { color: #ffa07a; } /* Other */
    code span.pp { color: #dcc6e0; } /* Preprocessor */
    code span.sc { color: #00e0e0; } /* SpecialChar */
    code span.ss { color: #abe338; } /* SpecialString */
    code span.st { color: #abe338; } /* String */
    code span.va { color: #f5ab35; } /* Variable */
    code span.vs { color: #abe338; } /* VerbatimString */
    code span.wa { color: #d4d0ab; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="../site_libs/revealjs/dist/theme/quarto-8fafb62820c35118272e29c1027a293f.css">
  <link href="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
<meta property="og:title" content="Factor-centric chromatin analysis – MOLB 7950">
<meta property="og:site_name" content="MOLB 7950">
<meta name="twitter:title" content="Factor-centric chromatin analysis – MOLB 7950">
<meta name="twitter:image" content="https://rnabioco.github.io/molb-7950/slides/images/twitter-card.png">
<meta name="twitter:creator" content="@rnabioco">
<meta name="twitter:card" content="summary_large_image">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Factor-centric chromatin analysis</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Jay Hesselberth 
</div>
        <p class="quarto-title-affiliation">
            RNA Bioscience Initiative | CU Anschutz
          </p>
    </div>
</div>

  <p class="date">2025-09-13</p>
</section>
<section id="where-do-transcription-factors-bind-in-the-genome" class="slide level2">
<h2>Where do transcription factors bind in the genome?</h2>
<p>Today we’ll look at where two yeast transcription factors bind in the genome using CUT&amp;RUN.</p>
</section>
<section id="where-do-transcription-factors-bind-in-the-genome-1" class="slide level2">
<h2>Where do transcription factors bind in the genome?</h2>
<p>Techniques like CUT&amp;RUN require an affinity reagent (e.g., an antibody) that uniquely recognizes a transcription factor in the cell.</p>
<p>This antibody is added to permeabilized cells, and the antibody associates with the epitope. A separate reagent, a fusion of Protein A (which binds IgG) and micrococcal nuclease (MNase) then associates with the antibody. Addition of calcium activates MNase, and nearby DNA is digested. These DNA fragments are then isolated and sequenced to identify sites of TF association in the genome.</p>
</section>
<section id="where-do-transcription-factors-bind-in-the-genome-2" class="slide level2">
<h2>Where do transcription factors bind in the genome?</h2>

<img data-src="../img/block-dna/skene-fig-1a.png" class="r-stretch quarto-figure-center"><p class="caption">Fig 1a, Skene et al.</p></section>
<section id="data-download-and-pre-processing" class="slide level2">
<h2>Data download and pre-processing</h2>
<p>CUT&amp;RUN data were downloaded from the <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE84474">NCBI GEO page</a> for Skene et al.</p>
<p>I selected the 16 second time point for <em>S. cerevisiae</em> Abf1 and Reb1 (note the paper combined data from the 1-32 second time points).</p>
<p>BED files containing mapped DNA fragments were separated by size and converted to bigWig with:</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb1"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a></a><span class="co"># separate fragments by size</span></span>
<span id="cb1-2"><a></a><span class="fu">awk</span> <span class="st">'($3 - $2 &lt;= 120)'</span> Abf1.bed <span class="op">&gt;</span> CutRun_Abf1_lt120.bed</span>
<span id="cb1-3"><a></a><span class="fu">awk</span> <span class="st">'($3 - $2 =&gt; 150)'</span> Abf1.bed <span class="op">&gt;</span> CutRun_Abf1_gt150.bed</span>
<span id="cb1-4"><a></a></span>
<span id="cb1-5"><a></a><span class="co"># for each file with the different sizes</span></span>
<span id="cb1-6"><a></a><span class="ex">bedtools</span> genomecov <span class="at">-i</span> Abf1.bed <span class="at">-g</span> sacCer3.chrom.sizes <span class="at">-bg</span> <span class="op">&gt;</span> Abf1.bg</span>
<span id="cb1-7"><a></a><span class="ex">bedGraphToBigWig</span> Abf1.bg sacCer3.chrom.sizes Abf1.bw</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<p>The bigWig files are available here in the <code>data/</code> directory.</p>
</section>
<section>
<section id="cutrun-analysis" class="title-slide slide level1 center">
<h1>CUT&amp;RUN analysis</h1>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-2"><a></a><span class="fu">library</span>(here)</span>
<span id="cb2-3"><a></a><span class="fu">library</span>(valr)</span>
<span id="cb2-4"><a></a></span>
<span id="cb2-5"><a></a><span class="co"># genome viz</span></span>
<span id="cb2-6"><a></a><span class="fu">library</span>(TxDb.Scerevisiae.UCSC.sacCer3.sgdGene)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import
'S4Arrays::makeNindexFromArrayViewport' by
'DelayedArray::makeNindexFromArrayViewport' when loading
'SummarizedExperiment'</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a></a><span class="fu">library</span>(Gviz)</span>
<span id="cb4-2"><a></a><span class="fu">library</span>(rtracklayer)</span>
<span id="cb4-3"><a></a></span>
<span id="cb4-4"><a></a><span class="co"># motif discovery and viz</span></span>
<span id="cb4-5"><a></a><span class="fu">library</span>(BSgenome.Scerevisiae.UCSC.sacCer3)</span>
<span id="cb4-6"><a></a><span class="fu">library</span>(memes)</span>
<span id="cb4-7"><a></a><span class="fu">library</span>(ggseqlogo)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a></a>track_start <span class="ot">&lt;-</span> <span class="dv">90000</span></span>
<span id="cb5-2"><a></a>track_end <span class="ot">&lt;-</span> <span class="dv">150000</span></span>
<span id="cb5-3"><a></a></span>
<span id="cb5-4"><a></a><span class="co"># genes track</span></span>
<span id="cb5-5"><a></a>sgd_genes_trk <span class="ot">&lt;-</span></span>
<span id="cb5-6"><a></a>  <span class="fu">GeneRegionTrack</span>(</span>
<span id="cb5-7"><a></a>    TxDb.Scerevisiae.UCSC.sacCer3.sgdGene,</span>
<span id="cb5-8"><a></a>    <span class="at">chromosome =</span> <span class="st">"chrII"</span>,</span>
<span id="cb5-9"><a></a>    <span class="at">start =</span> track_start,</span>
<span id="cb5-10"><a></a>    <span class="at">end =</span> track_end,</span>
<span id="cb5-11"><a></a>    <span class="at">background.title =</span> <span class="st">"white"</span>,</span>
<span id="cb5-12"><a></a>    <span class="at">col.title =</span> <span class="st">"black"</span>,</span>
<span id="cb5-13"><a></a>    <span class="at">fontsize =</span> <span class="dv">16</span></span>
<span id="cb5-14"><a></a>  )</span>
<span id="cb5-15"><a></a></span>
<span id="cb5-16"><a></a><span class="co"># signal tracks</span></span>
<span id="cb5-17"><a></a>track_info <span class="ot">&lt;-</span></span>
<span id="cb5-18"><a></a>  <span class="fu">tibble</span>(</span>
<span id="cb5-19"><a></a>    <span class="at">file_name =</span> <span class="fu">c</span>(</span>
<span id="cb5-20"><a></a>      <span class="st">"CutRun_Reb1_lt120.bw"</span>,</span>
<span id="cb5-21"><a></a>      <span class="st">"CutRun_Abf1_lt120.bw"</span>,</span>
<span id="cb5-22"><a></a>      <span class="st">"CutRun_Reb1_gt150.bw"</span>,</span>
<span id="cb5-23"><a></a>      <span class="st">"CutRun_Abf1_gt150.bw"</span></span>
<span id="cb5-24"><a></a>    ),</span>
<span id="cb5-25"><a></a>    <span class="at">sample_type =</span> <span class="fu">c</span>(</span>
<span id="cb5-26"><a></a>      <span class="st">"Reb1_Short"</span>,</span>
<span id="cb5-27"><a></a>      <span class="st">"Abf1_Short"</span>,</span>
<span id="cb5-28"><a></a>      <span class="st">"Reb1_Long"</span>,</span>
<span id="cb5-29"><a></a>      <span class="st">"Abf1_Long"</span></span>
<span id="cb5-30"><a></a>    )</span>
<span id="cb5-31"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb5-32"><a></a>  <span class="fu">mutate</span>(</span>
<span id="cb5-33"><a></a>    <span class="at">file_path =</span> <span class="fu">here</span>(<span class="st">"data/block-dna"</span>, file_name),</span>
<span id="cb5-34"><a></a>    <span class="at">big_wig =</span> purrr<span class="sc">::</span><span class="fu">map</span>(</span>
<span id="cb5-35"><a></a>      file_path,</span>
<span id="cb5-36"><a></a>      <span class="sc">~</span> <span class="fu">import.bw</span>(.x, <span class="at">as =</span> <span class="st">"GRanges"</span>)</span>
<span id="cb5-37"><a></a>    ),</span>
<span id="cb5-38"><a></a>    <span class="at">data_track =</span> purrr<span class="sc">::</span><span class="fu">map2</span>(</span>
<span id="cb5-39"><a></a>      big_wig,</span>
<span id="cb5-40"><a></a>      sample_type,</span>
<span id="cb5-41"><a></a>      \(x, y) {</span>
<span id="cb5-42"><a></a>        <span class="fu">DataTrack</span>(</span>
<span id="cb5-43"><a></a>          x,</span>
<span id="cb5-44"><a></a>          <span class="at">name =</span> y,</span>
<span id="cb5-45"><a></a>          <span class="at">background.title =</span> <span class="st">"white"</span>,</span>
<span id="cb5-46"><a></a>          <span class="at">col.title =</span> <span class="st">"black"</span>,</span>
<span id="cb5-47"><a></a>          <span class="at">col.axis =</span> <span class="st">"black"</span>,</span>
<span id="cb5-48"><a></a>          <span class="at">fontsize =</span> <span class="dv">16</span></span>
<span id="cb5-49"><a></a>        )</span>
<span id="cb5-50"><a></a>      }</span>
<span id="cb5-51"><a></a>    )</span>
<span id="cb5-52"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb5-53"><a></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample_type, big_wig, data_track)</span>
<span id="cb5-54"><a></a></span>
<span id="cb5-55"><a></a><span class="co"># x-axis track</span></span>
<span id="cb5-56"><a></a>x_axis_trk <span class="ot">&lt;-</span> <span class="fu">GenomeAxisTrack</span>(</span>
<span id="cb5-57"><a></a>  <span class="at">col =</span> <span class="st">"black"</span>,</span>
<span id="cb5-58"><a></a>  <span class="at">col.axis =</span> <span class="st">"black"</span>,</span>
<span id="cb5-59"><a></a>  <span class="at">fontsize =</span> <span class="dv">16</span></span>
<span id="cb5-60"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a></a><span class="fu">plotTracks</span>(</span>
<span id="cb6-2"><a></a>  <span class="fu">c</span>(</span>
<span id="cb6-3"><a></a>    sgd_genes_trk,</span>
<span id="cb6-4"><a></a>    track_info<span class="sc">$</span>data_track,</span>
<span id="cb6-5"><a></a>    x_axis_trk</span>
<span id="cb6-6"><a></a>  ),</span>
<span id="cb6-7"><a></a>  <span class="at">from =</span> track_start,</span>
<span id="cb6-8"><a></a>  <span class="at">to =</span> track_end,</span>
<span id="cb6-9"><a></a>  <span class="at">chromosome =</span> <span class="st">"chrII"</span>,</span>
<span id="cb6-10"><a></a>  <span class="at">transcriptAnnotation =</span> <span class="st">"gene"</span>,</span>
<span id="cb6-11"><a></a>  <span class="at">shape =</span> <span class="st">"arrow"</span>,</span>
<span id="cb6-12"><a></a>  <span class="at">type =</span> <span class="st">"histogram"</span></span>
<span id="cb6-13"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a></a>abf1_tbl <span class="ot">&lt;-</span> <span class="fu">read_bigwig</span>(<span class="fu">here</span>(<span class="st">"data/block-dna/CutRun_Abf1_lt120.bw"</span>))</span>
<span id="cb7-2"><a></a>total_reads <span class="ot">&lt;-</span> <span class="fl">16e6</span></span>
<span id="cb7-3"><a></a></span>
<span id="cb7-4"><a></a>genome <span class="ot">&lt;-</span> <span class="fu">read_genome</span>(<span class="fu">here</span>(<span class="st">"data/block-dna/sacCer3.chrom.sizes"</span>))</span>
<span id="cb7-5"><a></a>genome_size <span class="ot">&lt;-</span> <span class="fu">sum</span>(genome<span class="sc">$</span>size)</span>
<span id="cb7-6"><a></a></span>
<span id="cb7-7"><a></a>genome_lambda <span class="ot">&lt;-</span> total_reads <span class="sc">/</span> genome_size</span>
<span id="cb7-8"><a></a>peak_calls <span class="ot">&lt;-</span></span>
<span id="cb7-9"><a></a>  abf1_tbl <span class="sc">|&gt;</span></span>
<span id="cb7-10"><a></a>  <span class="co"># define single-base sites</span></span>
<span id="cb7-11"><a></a>  <span class="fu">mutate</span>(</span>
<span id="cb7-12"><a></a>    <span class="at">midpoint =</span> start <span class="sc">+</span> <span class="fu">round</span>((end <span class="sc">-</span> start) <span class="sc">/</span> <span class="dv">2</span>),</span>
<span id="cb7-13"><a></a>    <span class="at">start =</span> midpoint,</span>
<span id="cb7-14"><a></a>    <span class="at">end =</span> start <span class="sc">+</span> <span class="dv">1</span>,</span>
<span id="cb7-15"><a></a>    <span class="co"># use the poisson to calculate a p-value with the genome-wide lambda</span></span>
<span id="cb7-16"><a></a>    <span class="at">pval =</span> <span class="fu">dpois</span>(value, genome_lambda),</span>
<span id="cb7-17"><a></a>    <span class="co"># convert p-values to FDR</span></span>
<span id="cb7-18"><a></a>    <span class="at">fdr =</span> <span class="fu">p.adjust</span>(pval, <span class="at">method =</span> <span class="st">"fdr"</span>)</span>
<span id="cb7-19"><a></a>  )</span>
<span id="cb7-20"><a></a></span>
<span id="cb7-21"><a></a>peak_calls_sig <span class="ot">&lt;-</span></span>
<span id="cb7-22"><a></a>  <span class="fu">filter</span>(</span>
<span id="cb7-23"><a></a>    peak_calls,</span>
<span id="cb7-24"><a></a>    fdr <span class="sc">==</span> <span class="dv">0</span></span>
<span id="cb7-25"><a></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb7-26"><a></a>  <span class="co"># collapse neighboring, significant sites</span></span>
<span id="cb7-27"><a></a>  <span class="fu">bed_merge</span>(<span class="at">max_dist =</span> <span class="dv">20</span>)</span>
<span id="cb7-28"><a></a></span>
<span id="cb7-29"><a></a><span class="fu">filter</span>(</span>
<span id="cb7-30"><a></a>  peak_calls_sig,</span>
<span id="cb7-31"><a></a>  chrom <span class="sc">==</span> <span class="st">"chrII"</span> <span class="sc">&amp;</span></span>
<span id="cb7-32"><a></a>    start <span class="sc">&gt;=</span> track_start <span class="sc">&amp;</span></span>
<span id="cb7-33"><a></a>    end <span class="sc">&lt;=</span> track_end</span>
<span id="cb7-34"><a></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</section>
<section id="cutrun-analysis-output" class="title-slide slide level1 output-location-slide center"><h1>CUT&amp;RUN analysis</h1><div class="cell output-location-slide">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 3
  chrom  start    end
  &lt;chr&gt;  &lt;int&gt;  &lt;int&gt;
1 chrII 100248 100289
2 chrII 101292 101393
3 chrII 124916 124949
4 chrII 136181 136264
5 chrII 141070 141121</code></pre>
</div>
</div></section><section id="cutrun-analysis-output" class="title-slide slide level1 output-location-slide center"><h1>CUT&amp;RUN analysis</h1><div class="cell output-location-slide">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-21_files/figure-revealjs/plot-tracks-1.png" alt="Genome browser tracks for Abf1 and Reb1 CUT&amp;RUN data." width="960"></p>
</figure>
</div>
</div>
</div></section><section id="how-do-proteins-recognize-specific-locations-in-the-genome-to-bind" class="slide level2">
<h2>How do proteins recognize specific locations in the genome to bind?</h2>
</section>
<section id="motif-discovery" class="slide level2">
<h2>Motif discovery</h2>
</section>
<section id="theory" class="slide level2">
<h2>Theory</h2>
<p>There are two major approaches to defining sequence motifs enriched in a sample: enumerative and probabilistic approaches.</p>
</section>
<section id="theory-1" class="slide level2">
<h2>Theory</h2>
<p>Here we’ll apply a probabilistic approach (MEME) to discover motifs in a collection of DNA sequences. During the RNA block, you’ll learn about k-mer analysis, which is a form of enumerative approach.</p>
<p>In each case, the goal is to define a set of sequence motifs that are encriched in a set of provided sequences (i.e., peaks from CUT&amp;RUN data) relative to a genomic background.</p>
</section>
<section id="theory-2" class="slide level2">
<h2>Theory</h2>
<p>Motifs are expressed in a <a href="https://en.wikipedia.org/wiki/Position_weight_matrix">Position Weight Matrix</a>, which captures the propensities for a position to be a particular nucleotide in a sequence motif.</p>
<p>These PWMs can be represented as sequence logos, visually represent the amount of information provided by the motif, typically using “information content”, expressed in bits.</p>
</section>
<section id="theory-3" class="slide level2">
<h2>Theory</h2>

<img data-src="../img/block-dna/lexa-motif.png" class="r-stretch quarto-figure-center"><p class="caption">LexA sequence motif</p></section>
<section id="practice" class="slide level2">
<h2>Practice</h2>
<p>We’ll use the <a href="https://bioconductor.org/packages/release/bioc/html/memes.html">memes</a> package from Bioconductor to derive sequence motifs from the peaks we called above. This is a straightforward process:</p>
<ol type="1">
<li class="fragment">Collect the DNA sequences within the peak windows using the BSgenome for <em>S. cerevisiae</em></li>
<li class="fragment">Provide those sequences and the genomic background to <code>runDreme()</code>, which runs uses an Expectation-Maximization (EM) approach to identify and refine motifs.</li>
<li class="fragment">Examine the discovered motifs, and plot as a logo using <code>ggseqlogo</code>.</li>
</ol>
<div class="fragment">
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a></a>peak_calls_gr <span class="ot">&lt;-</span></span>
<span id="cb9-2"><a></a>  <span class="fu">GRanges</span>(</span>
<span id="cb9-3"><a></a>    <span class="at">seqnames =</span> peak_calls_sig<span class="sc">$</span>chrom,</span>
<span id="cb9-4"><a></a>    <span class="at">ranges =</span> <span class="fu">IRanges</span>(peak_calls_sig<span class="sc">$</span>start, peak_calls_sig<span class="sc">$</span>end)</span>
<span id="cb9-5"><a></a>  )</span>
<span id="cb9-6"><a></a></span>
<span id="cb9-7"><a></a>peak_seqs <span class="ot">&lt;-</span> BSgenome<span class="sc">::</span><span class="fu">getSeq</span>(</span>
<span id="cb9-8"><a></a>  <span class="co"># provided by BSgenome.Scerevisiae.UCSC.sacCer3</span></span>
<span id="cb9-9"><a></a>  Scerevisiae,</span>
<span id="cb9-10"><a></a>  peak_calls_gr</span>
<span id="cb9-11"><a></a>)</span>
<span id="cb9-12"><a></a></span>
<span id="cb9-13"><a></a>motif_results <span class="ot">&lt;-</span> <span class="fu">runStreme</span>(</span>
<span id="cb9-14"><a></a>  <span class="at">input =</span> peak_seqs,</span>
<span id="cb9-15"><a></a>  <span class="at">control =</span> <span class="st">"shuffle"</span>, <span class="co"># use shuffled sequences as control</span></span>
<span id="cb9-16"><a></a>  <span class="at">minw =</span> <span class="dv">6</span>, <span class="co"># minimum motif width</span></span>
<span id="cb9-17"><a></a>  <span class="at">maxw =</span> <span class="dv">12</span>, <span class="co"># maximum motif width</span></span>
<span id="cb9-18"><a></a>  <span class="at">nmotifs =</span> <span class="dv">5</span>, <span class="co"># number of motifs to find</span></span>
<span id="cb9-19"><a></a>  <span class="at">seed =</span> <span class="dv">42</span></span>
<span id="cb9-20"><a></a>)</span>
<span id="cb9-21"><a></a></span>
<span id="cb9-22"><a></a><span class="co"># look at the consensus motifs</span></span>
<span id="cb9-23"><a></a>motif_results</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

<p>Now let’s look at the sequence logo for the top hit.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a></a>motif_1 <span class="ot">&lt;-</span> motif_results[[<span class="dv">1</span>]]</span>
<span id="cb10-2"><a></a>pwm <span class="ot">&lt;-</span> motif_1[[<span class="dv">1</span>]]<span class="sc">@</span>motif</span>
<span id="cb10-3"><a></a></span>
<span id="cb10-4"><a></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb10-5"><a></a>  <span class="fu">geom_logo</span>(pwm) <span class="sc">+</span></span>
<span id="cb10-6"><a></a>  <span class="fu">theme_logo</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>

</div>
</section>
<section id="practice-output" class="slide level2 output-location-slide"><h2>Practice</h2><div class="cell output-location-slide">

</div></section><section id="practice-output" class="slide level2 output-location-slide"><h2>Practice</h2><div class="cell output-location-slide">

</div></section><section id="questions" class="slide level2">
<h2>Questions</h2>
<ol type="1">
<li class="fragment"><p>Does this motif make sense, based on what you know about the requirements and specificity of DNA binding by transcription factors?</p></li>
<li class="fragment"><p>How might you confirm that a specific sequence (that conforms to a motif) is bound directly by a transcription factor?</p></li>
</ol>


</section></section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">
<p><a href="https://rnabioco.github.io/molb-7950">Course 🔗</a></p>
</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': true,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"8\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdn.jsdelivr.net/npm/mathjax@2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
          const outerScaffold = trigger.parentElement.cloneNode(true);
          const codeEl = outerScaffold.querySelector('code');
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp("https:\/\/rnabioco\.github\.io\/molb-7950\/");
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>