<!DOCTYPE html>
<html lang="en"><head>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/tabby.min.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-8f4e3aabaa80e3e650ae20f2c24f966b.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.8.24">

  <meta name="author" content="Neel Mukherjee">
  <meta name="dcterms.date" content="2025-09-28">
  <title>MOLB 7950 – class 22</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    html { -webkit-text-size-adjust: 100%; }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #97947a;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #97947a;  padding-left: 4px; }
    div.sourceCode
      { color: #f8f8f2; background-color: #2b2b2b; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #f8f8f2; } /* Normal */
    code span.al { color: #dcc6e0; } /* Alert */
    code span.an { color: #d4d0ab; } /* Annotation */
    code span.at { color: #ffd700; } /* Attribute */
    code span.bn { color: #dcc6e0; } /* BaseN */
    code span.bu { color: #f5ab35; } /* BuiltIn */
    code span.cf { color: #ffa07a; } /* ControlFlow */
    code span.ch { color: #abe338; } /* Char */
    code span.cn { color: #ffa07a; } /* Constant */
    code span.co { color: #d4d0ab; } /* Comment */
    code span.cv { color: #d4d0ab; font-style: italic; } /* CommentVar */
    code span.do { color: #d4d0ab; font-style: italic; } /* Documentation */
    code span.dt { color: #dcc6e0; } /* DataType */
    code span.dv { color: #dcc6e0; } /* DecVal */
    code span.er { color: #dcc6e0; } /* Error */
    code span.ex { color: #ffd700; } /* Extension */
    code span.fl { color: #f5ab35; } /* Float */
    code span.fu { color: #ffd700; } /* Function */
    code span.im { } /* Import */
    code span.in { color: #d4d0ab; } /* Information */
    code span.kw { color: #ffa07a; } /* Keyword */
    code span.op { color: #00e0e0; } /* Operator */
    code span.ot { color: #ffa07a; } /* Other */
    code span.pp { color: #dcc6e0; } /* Preprocessor */
    code span.sc { color: #00e0e0; } /* SpecialChar */
    code span.ss { color: #abe338; } /* SpecialString */
    code span.st { color: #abe338; } /* String */
    code span.va { color: #f5ab35; } /* Variable */
    code span.vs { color: #abe338; } /* VerbatimString */
    code span.wa { color: #d4d0ab; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="../site_libs/revealjs/dist/theme/quarto-8fafb62820c35118272e29c1027a293f.css">
  <link href="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
<meta property="og:title" content="class 22 – MOLB 7950">
<meta property="og:description" content="RNA-sequencing intro">
<meta property="og:site_name" content="MOLB 7950">
<meta name="twitter:title" content="class 22 – MOLB 7950">
<meta name="twitter:description" content="RNA-sequencing intro">
<meta name="twitter:image" content="https://rnabioco.github.io/molb-7950/slides/images/twitter-card.png">
<meta name="twitter:creator" content="@rnabioco">
<meta name="twitter:card" content="summary_large_image">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">class 22</h1>
  <p class="subtitle">RNA-sequencing intro</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Neel Mukherjee 
</div>
        <p class="quarto-title-affiliation">
            RNA Bioscience Initiative | CU Anschutz
          </p>
    </div>
</div>

  <p class="date">2025-09-28</p>
</section>
<section id="lecture-overview" class="slide level2">
<h2>Lecture Overview</h2>
<ul>
<li class="fragment">mRNA life cycle</li>
<li class="fragment">RNA-seq</li>
<li class="fragment">Alignment</li>
<li class="fragment">Transcript quantification</li>
<li class="fragment">DESeq model</li>
<li class="fragment">Additional statistical considerations</li>
</ul>
</section>
<section id="rna-in-the-cell" class="slide level2">
<h2>RNA in the cell</h2>

<img data-src="../img/block-rna/rrna.png" class="r-stretch"></section>
<section id="life-fo-an-mrna" class="slide level2">
<h2>life fo an mRNA</h2>
<p>Messenger RNA (mRNA) carries genetic information encoded in DNA required for making proteins.</p>

<img data-src="../img/block-rna/RNA_life_cycle.png" style="width:5in" class="r-stretch"></section>
<section id="steps-of-rna-regulation" class="slide level2 smaller">
<h2>Steps of RNA regulation</h2>
<ul>
<li class="fragment"><strong>Transcription</strong>: A pre-mRNA still containing intron sequences is transcribed from DNA.</li>
<li class="fragment"><strong>Capping</strong>: A 7-methyl-guanosine “cap” is attached the 5’ end of the nascent RNA.</li>
<li class="fragment"><strong>Splicing</strong>: The excision intronic sequences.</li>
<li class="fragment"><strong>Cleavage</strong>: The nascent RNA is cleaved from DNA.</li>
<li class="fragment"><strong>Polyadenylation</strong>: The addition of a polyA tail to the 3’ end of the now RNA.</li>
<li class="fragment"><strong>Export</strong>: The mature mRNA is transported from the nucleus to the cytoplasm.</li>
<li class="fragment"><strong>Localization</strong>: The mRNA is localized to specific sub-regions or organelles within the cell.</li>
<li class="fragment"><strong>Translation</strong>: The production of specific protein based on the codons present within the mRNA.</li>
<li class="fragment"><strong>Decay</strong>: The enzymatic degradation of mRNA molecules.</li>
</ul>
</section>
<section id="rna-seq" class="slide level2 smaller">
<h2>RNA-seq</h2>
<p>Typically refers to “long” RNAs i.e.&nbsp;mRNA and long non-coding RNA (lncRNA). Specifically, we capture the steady-state pool of mature mRNA and to a lesser degree pre-mRNA. Thus, we can easily assess the abundance and isoforms expressed in the sample of interest. Of course, long read RNA sequencing (Nanopore, PacBio) enable better detection of continuity of exons and full-length isoforms.</p>

<img data-src="https://raw.githubusercontent.com/Sydney-Informatics-Hub/training-RNAseq-slides/master/01_IntroductionToRNASeq/assets/tracy1.png" class="r-stretch"></section>
<section id="selecting-rna-populations-for-sequencing" class="slide level2 smaller">
<h2>Selecting RNA populations for sequencing</h2>
<p>Need to determine which population of RNA you are interested in sequencing. The vast majority (~80%) of RNA in the cell is from ribosomal RNA. Smaller regulatory non-coding RNA are typically excluded due to size selection (snRNA, snoRNA, tRNA, miRNA).</p>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li class="fragment"><p><strong>polyA selection</strong>: uses oligo dT to hybridize to poly A tails of mRNA (and many long non-coding RNA)</p></li>
<li class="fragment"><p><strong>depletion of rRNA</strong>: uses DNA oligos complementary to portions of rRNA to either remove (purification) or degrade (RNaseH) to avoid rRNA getting into the library.</p></li>
<li class="fragment"><p><strong>size selection</strong>: sequence a population of RNAs that have a specific length such as microRNAs, which are ~21 nt regulatory non-coding RNAs.</p></li>
</ul>
</div><div class="column" style="width:50%;">
<p><img data-src="../img/block-rna/polyA_rRNA_size.png"></p>
</div></div>
</section>
<section id="importance-of-stand-specificity" class="slide level2 smaller">
<h2>Importance of stand-specificity</h2>
<div class="columns">
<div class="column" style="width:60%;">
<p><img data-src="../img/block-rna/strandedness.png"></p>
</div><div class="column" style="width:40%;">
<p>…and the genome has complex organization.</p>
<p>strand-specificity is crucial</p>
<p><img data-src="../img/block-rna/overlapping_genes.png"></p>
</div></div>
</section>
<section id="achieving-strand-specificity" class="slide level2">
<h2>Achieving strand-specificity</h2>

<img data-src="../img/block-rna/strand_protocol.png" class="r-stretch"></section>
<section id="rna-seq-read-alignment" class="slide level2 smaller">
<h2>RNA-seq read alignment</h2>
<p>You get your data back and align the reads to the genome, right? Nope, we need to deal with reads that will need to be “split” - spliced exons - to properly align. There are two strategies to deal with this: <strong>1) Spliced alignments</strong> and <strong>2) Pseudoalignment (transcripts)</strong>.</p>

<img data-src="https://upload.wikimedia.org/wikipedia/commons/0/01/RNA-Seq-alignment.png" class="r-stretch"></section>
<section id="spliced-alignment-workflow" class="slide level2 smaller">
<h2>Spliced alignment workflow</h2>
<div class="columns">
<div class="column" style="width:60%;">
<p><img data-src="https://github.com/hbctraining/Intro-to-rnaseq-hpc-salmon/blob/master/img/RNAseqWorkflow.png?raw=true" style="width:75.0%"></p>
</div><div class="column" style="width:40%;">
<p>Splice-aware aligners:</p>
<ul>
<li><p><a href="http://daehwankimlab.github.io/hisat2/">HiSat2</a></p></li>
<li><p><a href="http://ccb.jhu.edu/software/tophat/index.shtml">Tophat2</a></p></li>
<li><p><a href="https://github.com/alexdobin/STAR">STAR</a> <em>&lt;- use this</em></p></li>
</ul>
</div></div>
</section>
<section id="how-does-star-work" class="slide level2 smaller">
<h2>How does STAR work?</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><img data-src="../img/block-rna/alignment_STAR_step1.png"></p>
<p><img data-src="../img/block-rna/alignment_STAR_step2.png"></p>
</div><div class="column" style="width:50%;">
<p><img data-src="../img/block-rna/alignment_STAR_step5.png"></p>
</div></div>
</section>
<section id="pseudoalignment-and-transcript-quantification" class="slide level2 smaller">
<h2>Pseudoalignment and transcript quantification</h2>
<p>OR align/quantify in the same step. Fast and accurate…but you need to provide the transcripts (cannot discover new isoforms).</p>
<div class="columns">
<div class="column" style="width:50%;">
<p><img data-src="https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon/img/rnaseq_salmon_workflow.png"></p>
</div><div class="column" style="width:50%;">
<p><img data-src="https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon/img/salmon_workflow_subset.png"></p>
<p>Software:<br>
<a href="https://combine-lab.github.io/salmon/">Salmon</a> <em>&lt;- use this</em>.<br>
<a href="https://pachterlab.github.io/kallisto/about">Kallisto</a>.</p>
</div></div>
</section>
<section id="how-does-salmon-work" class="slide level2 smaller">
<h2>How does Salmon work?</h2>
<p>Create an index to evaluate the sequences for all possible unique sequences of length k (k-mer) in the transcriptome from known transcripts (splice isoforms for all genes).</p>
<div class="columns">
<div class="column" style="width:50%;">
<p>The Salmon index has two components:</p>
<ul>
<li class="fragment">a suffix array (SA) of the reference transcriptome</li>
<li class="fragment">a hash table to map each transcript in the reference transcriptome to it’s location in the SA</li>
</ul>
<p>The quasi-mapping approach estimates where the reads best map to on the transcriptome through identifying where informative sequences within the read map to instead of performing base-by-base alignment.</p>
</div><div class="column" style="width:50%;">
<p><img data-src="https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon/img/salmon_quasialignment.png"></p>
</div></div>
</section>
<section id="how-does-salmon-work-cont." class="slide level2 smaller">
<h2>How does Salmon work? (cont.)</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ol type="1">
<li class="fragment">The read is scanned from left to right until a k-mer that appears in the hash table is discovered.</li>
<li class="fragment">The k-mer is looked up in the hash table and the SA intervals are retrieved, giving all suffixes containing that k-mer</li>
<li class="fragment">Similar to STAR, the maximal matching prefix (MMP) is identified by finding the longest read sequence that exactly matches the reference suffixes.</li>
</ol>
</div><div class="column" style="width:50%;">
<p><img data-src="https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon/img/salmon_quasialignment.png"></p>
</div></div>
</section>
<section id="how-does-salmon-work-cont.-1" class="slide level2 smaller">
<h2>How does Salmon work? (cont.)</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ol start="4" type="1">
<li class="fragment">We could search for the next MMP at the position following the MMP, but often natural variation or a sequencing error in the read is the cause of the mismatch from the reference, so the beginning the search at this position would likely return the same set of transcripts. Therefore, Salmon identifies the next informative position (NIP), by skipping ahead 1 k-mer.</li>
<li class="fragment">This process is repeated until the end of the read.</li>
<li class="fragment">The final mappings are generated by determining the set of transcripts appearing in all MMPs for the read. The transcripts, orientation and transcript location are output for each read.</li>
</ol>
</div><div class="column" style="width:50%;">
<p><img data-src="https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon/img/salmon_quasialignment.png"></p>
<p>After determining the best mapping for each read/fragment, salmon will generate the final transcript abundance estimates after modeling sample-specific parameters and biases. Note that reads/fragments that map equally well to more than one transcript will have the count divided between all of the mappings; thereby not losing information for the various gene isoforms.</p>
</div></div>
</section>
<section id="accounting-for-biases" class="slide level2">
<h2>Accounting for biases</h2>
<p>Salmon and Kallisto account for:</p>
<ul>
<li class="fragment"><p>GC bias</p></li>
<li class="fragment"><p>positional coverage biases</p></li>
<li class="fragment"><p>sequence biases at 5’ and 3’ ends of the fragments</p></li>
<li class="fragment"><p>fragment length distribution</p></li>
<li class="fragment"><p>strand-specificity</p></li>
</ul>
</section>
<section id="transcript-quantification-metrics" class="slide level2 smaller">
<h2>Transcript quantification metrics</h2>
<p>Need to deal with systematic differences within/between samples such as:</p>
<ul>
<li class="fragment">sequencing depth</li>
<li class="fragment">gene/transcript length</li>
<li class="fragment">composition</li>
</ul>
</section>
<section id="sequencing-depth" class="slide level2 smaller">
<h2>Sequencing Depth</h2>

<img data-src="https://hbctraining.github.io/DGE_workshop/img/normalization_methods_depth.png" style="width:50.0%" class="r-stretch"></section>
<section id="gene-length" class="slide level2 smaller">
<h2>Gene length</h2>

<img data-src="https://hbctraining.github.io/DGE_workshop/img/normalization_methods_length.png" style="width:50.0%" class="r-stretch"></section>
<section id="composition" class="slide level2 smaller">
<h2>Composition</h2>

<img data-src="https://hbctraining.github.io/DGE_workshop/img/normalization_methods_composition.png" style="width:50.0%" class="r-stretch"></section>
<section id="popular-metrics" class="slide level2 smaller">
<h2>Popular metrics:</h2>
<div class="columns">
<div class="column" style="width:60%;">
<p>CPM — (read) Counts Per Million: <span class="math display">\[CPM =    \displaystyle \frac {\#\ reads\ mapped\ to\ gene*10^6}{Total\ \#\ mapped\ reads}\]</span></p>
<p>[F|R]PKM — [Fragments|Reads] Per Kilobase per Million: <span class="math display">\[FPKM =     \displaystyle \frac {\#\ fragments\ mapped\ to\ gene*10^6}{Total\ \#\ mapped\ reads*transcript\ length}\]</span></p>
<p>TPM — Transcripts Per Million: <span class="math display">\[TPM =  \displaystyle A* \frac {1}{\sum_{}A}\]</span> <span class="math display">\[A = \displaystyle \frac {\#\ fragments\ mapped\ to\ gene*10^6}{transcript\ length}\]</span></p>
</div><div class="column" style="width:40%;">
<ul>
<li class="fragment"><p>RPKM and FPKM are pretty much the same thing. FPKM is for fragments (paired-ends), not reads (single-end).</p></li>
<li class="fragment"><p>TPM and [F|R]PKM account for length differences between transcripts.</p></li>
<li class="fragment"><p>The sum of all [F|R]PKMs may not be the same across samples.</p></li>
<li class="fragment"><p>TPM normalizes to gene length first and then normalize for sequencing depth. <strong>Thus the sum of all TPMs is the same across samples</strong>. A better measure of relative transcript “concentration” in your sample than [F|R]PKM.</p></li>
</ul>
</div></div>
</section>
<section id="appropriate-use-and-caveats" class="slide level2 smaller">
<h2>Appropriate use and caveats</h2>
<p>Can I compare TPM / *PKM / CPM across samples?</p>
<blockquote>
<p>It depends what you mean by “compare”. Because these measures are purely relative, you cannot reliably use a metric like TPM to directly assess differences in transcript abundance across samples. Specifically, changes in the composition (e.g.&nbsp;polyA vs rRNA-depleted) of a sample can lead to changes in TPM, even if transcripts are, in reality, expressed at the same true level across samples. Metrics like this can be useful for “high-level” comparisons (e.g.&nbsp;visualizing samples etc.). However, whenever using a relative metric like this, one should be aware of its relative nature and the potential caveats that go along with interpreting them between samples.</p>
</blockquote>
<p><strong>We do NOT use TPM differences for differential expression.</strong></p>
<p><a href="http://robpatro.com/blog/?p=235">The RNA-seq abundance zoo</a></p>
<p><a href="https://rnajournal.cshlp.org/content/26/8/903.full">Misuse of RPKM or TPM normalization when comparing across samples and sequencing protocols</a></p>
<p><a href="https://www.youtube.com/watch?v=TTUrtCY2k-w">RPKM, FPKM and TPM, Clearly Explained!!!</a></p>
</section>
<section id="differential-expression-analysis" class="slide level2 smaller">
<h2>Differential expression analysis</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>DO NOT USE TPM (or anything we just talked about) to perform differential expression analysis.</p>
<p>RNA-seq data are discrete non-negative integers (counts per transcripts).</p>
<p>Remember the reads are (pseudo-)aligned and we <strong>COUNT</strong> how many are assigned to a specific transcript in a given sample.</p>
</div><div class="column" style="width:50%;">
<p><img data-src="../img/block-rna/de_workflow_salmon.png"></p>
</div></div>
</section>
<section id="distributions-for-count-data" class="slide level2 smaller">
<h2>Distributions for count data</h2>
<blockquote>
<p>Reads are count based and not normally distributed. Two distributions for count based data are poisson (which presumes the variance and mean are equal) or negative binomial (which does not). This is especially a problem when the number of biological replicates are low because it is hard to accurately model variance of count based data if you are looking at only that gene and making the assumptions of normally distributed continuous data (ie a t-test).</p>
</blockquote>
</section>
<section id="overdispersion" class="slide level2 smaller">
<h2>Overdispersion</h2>
<blockquote>
<p><strong>Overdispersion</strong> the variance of counts is generally greater than their mean, especially for genes expressed at a higher level.</p>
</blockquote>
<div class="columns">
<div class="column" style="width:50%;">
<p><img data-src="../img/block-rna/nb_mean_var.png"></p>
</div><div class="column" style="width:50%;">
<p>The total number of reads for each sample tends to be in the millions, while the counts per gene are much lower (many zeros, tens/hundreds) and vary considerably. While the Poisson distribution seems appropriate for sampling out of a large pool with low probability. Poisson does not handle <strong>overdispersion</strong>, enter the Negative Binomial distribution.</p>
</div></div>
</section>
<section id="examine-count-data" class="slide level2 smaller">
<h2>Examine count data</h2>
<div class="cell columns column-output-location">
<div class="column">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a></a>d <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(</span>
<span id="cb1-2"><a></a>  <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"unfilt_counts.csv.gz"</span>)</span>
<span id="cb1-3"><a></a>) <span class="sc">|&gt;</span></span>
<span id="cb1-4"><a></a>  <span class="fu">as.matrix</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div><div class="column fragment">
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 57808 Columns: 6
── Column specification ────────────────────────────────────
Delimiter: ","
dbl (6): mock_rna_A, mock_rna_B, mock_rna_C, 8430_rna_A,...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a></a>df <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb3-2"><a></a>  <span class="at">variance =</span> <span class="fu">rowVars</span>(d),</span>
<span id="cb3-3"><a></a>  <span class="at">mean =</span> <span class="fu">rowMeans</span>(d)</span>
<span id="cb3-4"><a></a>)</span>
<span id="cb3-5"><a></a></span>
<span id="cb3-6"><a></a><span class="fu">ggplot</span>(df) <span class="sc">+</span></span>
<span id="cb3-7"><a></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> mean, <span class="at">y =</span> variance)) <span class="sc">+</span></span>
<span id="cb3-8"><a></a>  <span class="fu">scale_y_log10</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1e9</span>)) <span class="sc">+</span></span>
<span id="cb3-9"><a></a>  <span class="fu">scale_x_log10</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1e9</span>)) <span class="sc">+</span></span>
<span id="cb3-10"><a></a>  <span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb3-11"><a></a>  <span class="fu">theme_cowplot</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in scale_y_log10(limits = c(1, 1e+09)): log-10
transformation introduced infinite values.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in scale_x_log10(limits = c(1, 1e+09)): log-10
transformation introduced infinite values.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 6367 rows containing missing values or values
outside the scale range (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="slides-22_files/figure-revealjs/plt-mean-var-1.png" alt="examine counts" width="960"></p>
</figure>
</div>
</div>
</div></div>
<ul>
<li class="fragment"><p>data points do not fall on the diagonal, <em>mean</em> != <em>var</em></p></li>
<li class="fragment"><p>for highly expressed genes, <em>var</em> &gt; <em>mean</em></p></li>
<li class="fragment"><p>lowly expressed genes have more scatter i.e.&nbsp;“heteroscedasticity”.</p></li>
</ul>
<p><a href="http://bridgeslab.sph.umich.edu/posts/why-do-we-use-the-negative-binomial-distribution-for-rnaseq">Why do we use the negative binomial distribution for analysing RNAseq data?</a></p>
<p><a href="https://bioramble.wordpress.com/2016/01/30/why-sequencing-data-is-modeled-as-negative-binomial/">Why sequencing data is modeled as a negative binomial</a></p>
</section>
<section id="deseq-model" class="slide level2 smaller">
<h2>DEseq Model</h2>

<img data-src="https://hbctraining.github.io/DGE_workshop/img/NB_model_formula.png" class="r-stretch"><p>where counts <span class="math inline">\(K_{ij}\)</span> for gene <code>i</code>, sample <code>j</code> are modeled using a negative binomial distribution with fitted mean <span class="math inline">\(\mu_{ij}\)</span> and a gene-specific dispersion parameter <span class="math inline">\(\alpha_i\)</span>. The fitted mean is composed of a sample-specific size factor <span class="math inline">\(s_{j}\)</span> and a parameter <span class="math inline">\(q_{ij}\)</span> proportional to the expected true concentration of fragments for sample <code>j</code>.</p>
</section>
<section id="deseq2" class="slide level2 smaller">
<h2>DESeq2</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><span class="math display">\[\log_2(q_{ij}) = x_{j.} \beta_i\]</span> The coefficients <span class="math inline">\(\beta_{i}\)</span> give log2 fold changes for gene <code>i</code> `for each column of the model matrix X. Note that the model can be generalized to use sample- and gene-dependent normalization factors <span class="math inline">\(s_{ij}\)</span>.</p>
</div><div class="column" style="width:50%;">
<p><img data-src="https://hbctraining.github.io/DGE_workshop/img/deseq2_workflow_separate.png"></p>
</div></div>
</section>
<section id="scaling-between-samples" class="slide level2 smaller">
<h2>Scaling between samples</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>The counts divided by sample-specific size factors determined by median ratio of gene counts relative to geometric mean per gene.</p>
<ul>
<li class="fragment"><p>Step 1: creates a pseudo-reference sample (row-wise geometric mean)</p></li>
<li class="fragment"><p>Step 2: calculates ratio of each sample to the reference</p></li>
<li class="fragment"><p>Step 3: calculate the normalization factor for each sample (size factor)</p></li>
</ul>
</div><div class="column" style="width:50%;">
<p><img data-src="https://hbctraining.github.io/DGE_workshop/img/normalization_methods_composition.png"></p>
</div></div>
</section>
<section id="estimatesizefactors" class="slide level2 smaller">
<h2>estimateSizeFactors</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a></a>d <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"unfilt_counts.csv.gz"</span>)) <span class="sc">|&gt;</span> <span class="fu">as.matrix</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell fragment">
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 57808 Columns: 6
── Column specification ────────────────────────────────────
Delimiter: ","
dbl (6): mock_rna_A, mock_rna_B, mock_rna_C, 8430_rna_A,...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a></a><span class="co"># estimateSizeFactorsForMatrix(counts = d)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="gene-wise-dispersion" class="slide level2 smaller">
<h2>Gene-wise dispersion</h2>
<p>We need to generate accurate estimates of within-group variation for each gene…but usually have only 3 replicates making it hard to estimate reliably.</p>
<p>DESeq2 shares information across genes to generate more accurate estimates of variation based on the mean expression level of the gene using a method called ‘shrinkage’. DESeq2 assumes that genes with similar expression levels have similar dispersion.</p>
<p>Estimating the dispersion for each gene separately:</p>
<p>To model the dispersion based on expression level (mean counts of replicates), the dispersion for each gene is estimated using maximum likelihood estimation. In other words, given the count values of the replicates, the most likely estimate of dispersion is calculated.</p>
<p><a href="https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html">Count normalization</a></p>
<p><a href="https://hbctraining.github.io/DGE_workshop/lessons/04_DGE_DESeq2_analysis.html">Comprehensive explanation of DESeq2 steps</a></p>
</section>
<section id="design-considerations" class="slide level2">
<h2>Design considerations</h2>

<img data-src="../img/block-rna/auer.jpg" class="r-stretch"></section>
<section id="remember-encode" class="slide level2 smaller">
<h2>Remember ENCODE</h2>
<p>A study compared mRNA expression profiles of many human and mouse tissues. One of their key findings:</p>
<blockquote>
<p>GENE EXPRESSION IS MORE SIMILAR AMONG TISSUES WITHIN A SPECIES THAN BETWEEN CORRESPONDING TISSUES OF THE TWO SPECIES</p>
</blockquote>

<img data-src="https://f1000researchdata.s3.amazonaws.com/manuscripts/7019/9f5f4330-d81d-46b8-9a3f-d8cb7aaf577e_figure1.gif" class="r-stretch"></section>
<section id="power-depth-vs-reps" class="slide level2 smaller">
<h2>Power: depth vs reps</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>Replicates allow us to:</p>
<ul>
<li>estimate variation for each gene</li>
<li>randomize out unknown covariates</li>
<li>spot outliers</li>
<li>improve precision of expression and fold-change estimates</li>
</ul>
<p><a href="https://cqs-vumc.shinyapps.io/rnaseqsamplesizeweb/">RNA-seq power calculation</a></p>
</div><div class="column" style="width:50%;">
<p><img data-src="../img/block-rna/de_replicates_img2.png"></p>
</div></div>


</section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">
<p><a href="https://rnabioco.github.io/molb-7950">Course 🔗</a></p>
</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': true,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"8\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdn.jsdelivr.net/npm/mathjax@2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
          const outerScaffold = trigger.parentElement.cloneNode(true);
          const codeEl = outerScaffold.querySelector('code');
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp("https:\/\/rnabioco\.github\.io\/molb-7950\/");
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>