---
title: "Resources for the RNA block"
author: "Neelanjan Mukherjee"
date: last-modified
---

## Foundational work

These studies below laid the methodological groundwork for RNA-sequencing, ribosome profiling, and mapping binding sites of RNA-binding proteins using UV-crosslinking based approaches (CLIP-seq/PAR-CLIP).


#### RNA-seq

Mortazavi A, Williams BA, McCue K, Schaeffer L, Wold B. Mapping and quantifying mammalian transcriptomes by RNA-Seq. Nat Methods. 2008 Jul;5(7):621-8. doi: 10.1038/nmeth.1226. Epub 2008 May 30. PMID: 18516045. [\[Link\]](https://pubmed.ncbi.nlm.nih.gov/18516045/)


#### Ribosome profiling

Ingolia NT, Ghaemmaghami S, Newman JR, Weissman JS. Genome-wide analysis in vivo of translation with nucleotide resolution using ribosome profiling. Science. 2009 Apr 10;324(5924):218-23. doi: 10.1126/science.1168978. Epub 2009 Feb 12. PMID: 19213877; PMCID: PMC2746483.[\[Link\]](https://pubmed.ncbi.nlm.nih.gov/19213877/)


#### CLIP-seq

Ule J, Jensen KB, Ruggiu M, Mele A, Ule A, Darnell RB. CLIP identifies Nova-regulated RNA networks in the brain. Science. 2003 Nov 14;302(5648):1212-5. doi: 10.1126/science.1090095. PMID: 14615540. [\[Link\]](https://pubmed.ncbi.nlm.nih.gov/14615540/)

Hafner M, Landthaler M, Burger L, Khorshid M, Hausser J, Berninger P, Rothballer A, Ascano M Jr, Jungkamp AC, Munschauer M, Ulrich A, Wardle GS, Dewell S, Zavolan M, Tuschl T. Transcriptome-wide identification of RNA-binding protein and microRNA target sites by PAR-CLIP. Cell. 2010 Apr 2;141(1):129-41. doi: 10.1016/j.cell.2010.03.009. PMID: 20371350; PMCID: PMC2861495. [\[Link\]](https://pubmed.ncbi.nlm.nih.gov/20371350/)


## Software

### Alignment software

[STAR](https://github.com/alexdobin/STAR) and [minimap2](https://github.com/lh3/minimap2) are popular choices for short read alignment. They are fast, free, and well-maintained.

### Transcript Quantification  

[Salmon](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5600148/) is a pseuodoalignment based approach for quantifying transcripts from RNA-seq data. See the salmon [documentation](https://combine-lab.github.io/salmon/) and documentation for importing salmon data into R using [txipmort](https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html)

### Peak calling

[PARalyzer](https://pubmed.ncbi.nlm.nih.gov/21851591/) is the gold-standard in peak calling for PAR-CLIP data. It models read coverage and nucleotide conversions using a kernel density estimate classification to generate a high-resolution map of RNA-protein interaction sites.

