<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Neelanjan Mukherjee">
<meta name="dcterms.date" content="2024-10-02">
<title>Stats Bootcamp - class 15 – MOLB 7950</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>
<meta property="og:title" content="Stats Bootcamp - class 15 – MOLB 7950">
<meta property="og:description" content="Dealing with big data">
<meta property="og:site_name" content="MOLB 7950">
<meta name="twitter:title" content="Stats Bootcamp - class 15 – MOLB 7950">
<meta name="twitter:description" content="Dealing with big data">
<meta name="twitter:image" content="https://rnabioco.github.io/molb-7950/exercises/images/twitter-card.png">
<meta name="twitter:creator" content="@rnabioco">
<meta name="twitter:card" content="summary_large_image">
</head>
<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Stats Bootcamp - class 15</li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto"><div class="pt-lg-2 mt-2 text-center sidebar-header">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../img/logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none"></a>
      <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/rnabioco/molb-7950" title="Course GitHub page" class="quarto-navigation-tool px-1" aria-label="Course GitHub page"><i class="bi bi-github"></i></a>
    <a href="https://sso.posit.cloud/cu-anschutz" title="Posit Cloud SSO login" class="quarto-navigation-tool px-1" aria-label="Posit Cloud SSO login"><i class="bi bi-cloud"></i></a>
    <a href="https://calendly.com/molb7950" title="Office hours scheduling" class="quarto-navigation-tool px-1" aria-label="Office hours scheduling"><i class="bi bi-calendar"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Course information</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Schedule</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../course-info/syllabus.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Syllabus</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../course-info/support.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Support</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../course-info/team.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Teaching team</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Resources</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/bootcamp-resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bootcamp resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/block-dna-resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DNA Block resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/block-rna-resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNA Block resources</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Assignments</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../course-info/problem-sets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Problet Set Overview</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../course-info/final-projects.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Final Project Overview</span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#learning-objectives" id="toc-learning-objectives" class="nav-link active" data-scroll-target="#learning-objectives">Learning objectives</a></li>
  <li><a href="#types-i-and-ii-error" id="toc-types-i-and-ii-error" class="nav-link" data-scroll-target="#types-i-and-ii-error">Types I and II error</a></li>
  <li><a href="#different-visualization" id="toc-different-visualization" class="nav-link" data-scroll-target="#different-visualization">Different visualization</a></li>
  <li><a href="#genomics---lots-of-data---lots-of-hypothesis-tests" id="toc-genomics---lots-of-data---lots-of-hypothesis-tests" class="nav-link" data-scroll-target="#genomics---lots-of-data---lots-of-hypothesis-tests">Genomics -&gt; Lots of Data -&gt; Lots of Hypothesis Tests</a></li>
  <li><a href="#multiple-test-corrections" id="toc-multiple-test-corrections" class="nav-link" data-scroll-target="#multiple-test-corrections">Multiple test corrections</a></li>
  <li>
<a href="#bonferroni-correction" id="toc-bonferroni-correction" class="nav-link" data-scroll-target="#bonferroni-correction">Bonferroni Correction</a>
  <ul class="collapse">
<li><a href="#sidak-correction" id="toc-sidak-correction" class="nav-link" data-scroll-target="#sidak-correction">Sidak Correction</a></li>
  </ul>
</li>
  <li><a href="#holms-step-down-procedure" id="toc-holms-step-down-procedure" class="nav-link" data-scroll-target="#holms-step-down-procedure">Holm’s Step-Down Procedure</a></li>
  <li><a href="#hochbergs-step-up-procedure" id="toc-hochbergs-step-up-procedure" class="nav-link" data-scroll-target="#hochbergs-step-up-procedure">Hochberg’s Step-Up Procedure</a></li>
  <li><a href="#example" id="toc-example" class="nav-link" data-scroll-target="#example">Example</a></li>
  <li>
<a href="#exploratory-data-analysis-eda" id="toc-exploratory-data-analysis-eda" class="nav-link" data-scroll-target="#exploratory-data-analysis-eda">Exploratory data analysis (EDA)</a>
  <ul class="collapse">
<li><a href="#pca" id="toc-pca" class="nav-link" data-scroll-target="#pca">PCA</a></li>
  <li><a href="#correlation-matrix" id="toc-correlation-matrix" class="nav-link" data-scroll-target="#correlation-matrix">Correlation matrix</a></li>
  </ul>
</li>
  <li><a href="#explore-data" id="toc-explore-data" class="nav-link" data-scroll-target="#explore-data">Explore data</a></li>
  <li><a href="#data-transformations" id="toc-data-transformations" class="nav-link" data-scroll-target="#data-transformations">Data transformations</a></li>
  <li><a href="#pseudocounts" id="toc-pseudocounts" class="nav-link" data-scroll-target="#pseudocounts">Pseudocounts</a></li>
  <li><a href="#correlation-analysis" id="toc-correlation-analysis" class="nav-link" data-scroll-target="#correlation-analysis">correlation analysis</a></li>
  <li><a href="#pca-1" id="toc-pca-1" class="nav-link" data-scroll-target="#pca-1">PCA</a></li>
  <li><a href="#pca-2" id="toc-pca-2" class="nav-link" data-scroll-target="#pca-2">PCA</a></li>
  <li><a href="#pca---prepare-visualization" id="toc-pca---prepare-visualization" class="nav-link" data-scroll-target="#pca---prepare-visualization">PCA - prepare visualization</a></li>
  <li><a href="#famous-pca-example" id="toc-famous-pca-example" class="nav-link" data-scroll-target="#famous-pca-example">Famous PCA example</a></li>
  <li><a href="#some-found-this-hard-to-believe" id="toc-some-found-this-hard-to-believe" class="nav-link" data-scroll-target="#some-found-this-hard-to-believe">Some found this hard to believe</a></li>
  <li><a href="#careful-with-batch-effects" id="toc-careful-with-batch-effects" class="nav-link" data-scroll-target="#careful-with-batch-effects">Careful with batch effects</a></li>
  <li><a href="#accounting-for-batch-effects" id="toc-accounting-for-batch-effects" class="nav-link" data-scroll-target="#accounting-for-batch-effects">Accounting for batch effects</a></li>
  <li><a href="#k-means-clustering-to-look-for-patterns" id="toc-k-means-clustering-to-look-for-patterns" class="nav-link" data-scroll-target="#k-means-clustering-to-look-for-patterns">K-means clustering to look for patterns</a></li>
  <li><a href="#k-means-data-preparation" id="toc-k-means-data-preparation" class="nav-link" data-scroll-target="#k-means-data-preparation">K-Means data preparation</a></li>
  <li><a href="#scaling-or-z-score" id="toc-scaling-or-z-score" class="nav-link" data-scroll-target="#scaling-or-z-score">Scaling or z-score</a></li>
  <li><a href="#k-means-clustering" id="toc-k-means-clustering" class="nav-link" data-scroll-target="#k-means-clustering">K-Means clustering</a></li>
  <li><a href="#how-do-we-figure-out-the-optimal-clusters" id="toc-how-do-we-figure-out-the-optimal-clusters" class="nav-link" data-scroll-target="#how-do-we-figure-out-the-optimal-clusters">How do we figure out the optimal # clusters?</a></li>
  <li><a href="#lets-cluster-once-to-see" id="toc-lets-cluster-once-to-see" class="nav-link" data-scroll-target="#lets-cluster-once-to-see">Let’s cluster once to see</a></li>
  <li><a href="#functions-in-r" id="toc-functions-in-r" class="nav-link" data-scroll-target="#functions-in-r">Functions in R</a></li>
  <li><a href="#create-function-to-calculate-wss" id="toc-create-function-to-calculate-wss" class="nav-link" data-scroll-target="#create-function-to-calculate-wss">Create function to calculate wss</a></li>
  <li><a href="#find-the-elbow" id="toc-find-the-elbow" class="nav-link" data-scroll-target="#find-the-elbow">find the elbow</a></li>
  <li><a href="#final-clustering" id="toc-final-clustering" class="nav-link" data-scroll-target="#final-clustering">Final clustering</a></li>
  <li><a href="#which-clusters-contains-dux4-targets" id="toc-which-clusters-contains-dux4-targets" class="nav-link" data-scroll-target="#which-clusters-contains-dux4-targets">Which cluster(s) contains DUX4 targets?</a></li>
  <li><a href="#lets-calculate-the-empirical-p-value-of-the-cluster-most-enriched-for-dux4-targets-by-sampling" id="toc-lets-calculate-the-empirical-p-value-of-the-cluster-most-enriched-for-dux4-targets-by-sampling" class="nav-link" data-scroll-target="#lets-calculate-the-empirical-p-value-of-the-cluster-most-enriched-for-dux4-targets-by-sampling">Let’s calculate the empirical p-value of the cluster most enriched for DUX4 targets by sampling</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/rnabioco/molb-7950/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/rnabioco/molb-7950/edit/main/exercises/ex-15.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<h1 class="title">Stats Bootcamp - class 15</h1>
<p class="subtitle lead">Dealing with big data</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Neelanjan Mukherjee </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 2, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header><section id="learning-objectives" class="level2"><h2 class="anchored" data-anchor-id="learning-objectives">Learning objectives</h2>
<ul>
<li>Types of error and multiple test corrections</li>
<li>Exploratory data analysis</li>
<li>Clustering and overlaps</li>
</ul></section><section id="types-i-and-ii-error" class="level2"><h2 class="anchored" data-anchor-id="types-i-and-ii-error">Types I and II error</h2>
<p>False positives and False negatives</p>
<p><img src="https://www.scribbr.com/wp-content/uploads/2021/01/type-i-and-ii-error-2.png" class="img-fluid"></p>
<p><span class="math inline">\(\alpha\)</span> - significance level OR evidentiary standard</p>
<p><span class="math inline">\(\beta\)</span> - type II error rate, 1 - <span class="math inline">\(\beta\)</span> is power</p>
</section><section id="different-visualization" class="level2"><h2 class="anchored" data-anchor-id="different-visualization">Different visualization</h2>
<p>Power vs Significance</p>
<p><img src="https://www.scribbr.com/wp-content/uploads/2021/01/type-i-and-type-ii-error.png" class="img-fluid"></p>
</section><section id="genomics---lots-of-data---lots-of-hypothesis-tests" class="level2"><h2 class="anchored" data-anchor-id="genomics---lots-of-data---lots-of-hypothesis-tests">Genomics -&gt; Lots of Data -&gt; Lots of Hypothesis Tests</h2>
<p>In a typical RNA-seq experiment, we test ~10K different hypotheses. For example, you have 10K genes and for each gene you test whether the mean expression changed in condition A vs condition B. Using a standard p-value cut-off of 0.05, we’d expect <strong>500 genes</strong> to be deemed “significant” by chance. Thus, we are very concerned about <strong>False Positives or Type I Errors</strong>.</p>
</section><section id="multiple-test-corrections" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="multiple-test-corrections">Multiple test corrections</h2>
<ol type="1">
<li><p>Control overall <ce><b1> (also known as family-wise error rate or <a href="https://en.wikipedia.org/wiki/Family-wise_error_rate">FWER</a>), which will affect the <ce><b1>* for each test. That is, we are controlling the overall probability of making <em>at least one</em> false discovery. Bonferroni and Sidak corrections all control FWER.</b1></ce></b1></ce></p></li>
<li><p>Control <a href="https://en.wikipedia.org/wiki/False_discovery_rate">false discovery rate</a> (FDR). These procedures allow for type 1 errors (false positives) but control the proportion of these false positives in relation to true positives. This is done by adjusting the decision made for the p-value associated with each individual test to decide rejection or not. Because this will result in a higher type 1 error rate, it has higher <a href="https://en.wikipedia.org/wiki/Power_(statistics)">power</a>. This affords a higher probability of <em>true discoveries.</em> The step procedures control for FDR.</p></li>
</ol></section><section id="bonferroni-correction" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="bonferroni-correction">Bonferroni Correction</h2>
<p>The most conservative of corrections, the Bonferroni correction is also perhaps the most straightforward in its approach. Simply divide <ce><b1> by the number of tests (<em>m</em>).</b1></ce></p>
<blockquote class="blockquote">
<p><ce><b1> = <ce><b1>/m</b1></ce></b1></ce></p>
</blockquote>
<p>However, with many tests, <ce><b1> will become very small. This reduces power, which means that we are very unlikely to make any true discoveries.</b1></ce></p>
<section id="sidak-correction" class="level3"><h3 class="anchored" data-anchor-id="sidak-correction">Sidak Correction</h3>
<blockquote class="blockquote">
<p><ce><b1> = 1-(1-<ce><b1>)^(1/<em>m</em>)</b1></ce></b1></ce></p>
</blockquote>
</section></section><section id="holms-step-down-procedure" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="holms-step-down-procedure">Holm’s Step-Down Procedure</h2>
<p>The Holm-Bonferroni method is also fairly simple to calculate, but it is more powerful than the single-step Bonferroni.</p>
<p><span class="math inline">\(HB = \displaystyle \frac {target \alpha}{n - rank + 1}\)</span></p>
<p>H1: 0.005<br>
H2: 0.01<br>
H3: 0.03<br>
H4: = 0.04<br></p>
<p>Step 1: Order the p-values from smallest to greatest (already done)</p>
<p>Step 2: Calc HB for the first rank HB = .05 / 4 – 1 + 1 = .05 / 4 = .0125 H1: 0.005 &lt; .0125, so we reject the null</p>
<p>Step 4: Repeat the HB formula for the second rank and keep going until we find <span class="math inline">\(H{_N}\)</span> &gt; <span class="math inline">\(HB{_N}\)</span>. All subsequent hypotheses are non-significant (i.e.&nbsp;not rejected).</p>
</section><section id="hochbergs-step-up-procedure" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="hochbergs-step-up-procedure">Hochberg’s Step-Up Procedure</h2>
<p>More powerful than Holm’s step-down procedure, Hochberg’s step-up procedure also seeks to control the FDR and follows a similar process, only p-values are ranked from largest to smallest.</p>
<p>For each ranked p-value, it is compared to the <ce><b1> calculated for its respective rank (same formula as Holm’s procedure). Testing continues until you reach the first non-rejected hypothesis. You would then fail to reject all following hypotheses.</b1></ce></p>
</section><section id="example" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="example">Example</h2>
<div class="cell" data-output-location="slide">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>rna <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">here</span>(<span class="st">"data/data_rna_protein.csv.gz"</span>)) <span class="sc">|&gt;</span><span class="fu">select</span>(iDUX4_pval)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>rna<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> ??, <span class="at">method =</span> <span class="st">"fdr"</span>, <span class="at">n =</span> <span class="fu">nrow</span>(rna))</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>rna<span class="sc">$</span>BH <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> rna<span class="sc">$</span>iDUX4_pval, <span class="at">method =</span> <span class="st">"BH"</span>, <span class="at">n =</span> <span class="fu">nrow</span>(rna))</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>rna<span class="sc">$</span>bon <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="at">p =</span> rna<span class="sc">$</span>iDUX4_pval, <span class="at">method =</span> <span class="st">"bonferroni"</span>, <span class="at">n =</span> <span class="fu">nrow</span>(rna))</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>rna_long <span class="ot">&lt;-</span> rna <span class="sc">|&gt;</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> iDUX4_pval<span class="sc">:</span>bon, <span class="at">names_to =</span> <span class="st">"type"</span>) </span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> rna_long, <span class="fu">aes</span>(<span class="at">x=</span>value, <span class="at">fill =</span> type)) <span class="sc">+</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>type) <span class="sc">+</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_cowplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="exploratory-data-analysis-eda" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="exploratory-data-analysis-eda">Exploratory data analysis (EDA)</h2>
<p>Our goal here is to get an top-down big picture of the similarity/differences between variables in a dataset. For example, let’s say you do RNA-seq in triplicate on 4 treatment/developmental times.</p>
<section id="pca" class="level3"><h3 class="anchored" data-anchor-id="pca">PCA</h3>
<p>We will perform PCA on all of the samples and visualize the relationship between samples.</p>
</section><section id="correlation-matrix" class="level3"><h3 class="anchored" data-anchor-id="correlation-matrix">Correlation matrix</h3>
<p>We will perform hierarchical clustering on a matrix representing the pairwise correlation between all these samples.</p>
</section></section><section id="explore-data" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="explore-data">Explore data</h2>
<p>Is it normal-ish?</p>
<div class="cell" data-output-location="column-fragment">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get dux targets</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>dux_targets <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"target_genes.csv.gz"</span>))</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># get expression data</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"data_genelevel.tsv.gz"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">target =</span> <span class="fu">case_when</span>(</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    gene_symbol <span class="sc">%in%</span> dux_targets<span class="sc">$</span>hgnc_symbol <span class="sc">~</span> <span class="st">"target"</span>,</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"not_target"</span>)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>         ) <span class="sc">|&gt;</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene_symbol<span class="sc">!=</span><span class="st">"ISCA1"</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>()</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>d <span class="sc">|&gt;</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> hour00_rep1<span class="sc">:</span>hour14_rep3) <span class="sc">|&gt;</span> </span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>value, <span class="at">color=</span>name)) <span class="sc">+</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  ??() <span class="sc">+</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_cowplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>. . .</p>
<p>Definitely not normal</p>
</section><section id="data-transformations" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="data-transformations">Data transformations</h2>
<p>We often transform data to make it closer to being normally-distributed. This allows us to use more powerful statistical tests on the same data. One approach is to log-transform the data.</p>
<div class="cell" data-output-location="column-fragment">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">d</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="va">hour00_rep1</span><span class="op">:</span><span class="va">hour14_rep3</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="op">?</span><span class="op">?</span><span class="op">(</span><span class="va">value</span><span class="op">)</span>, color<span class="op">=</span><span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_density</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>. . .</p>
<p>What is this?</p>
<blockquote class="blockquote">
<p>Warning message: Removed 1251 rows containing non-finite values (stat_density()).</p>
</blockquote>
</section><section id="pseudocounts" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="pseudocounts">Pseudocounts</h2>
<p><span class="math inline">\(log_{x}(0)\)</span> is a common problem. One solution is to add a pseudocount. Since this is read count data, the smallest unit is 1 and so we will add 1 to all the observations before perforing the log transformation. <span class="math inline">\(1\)</span> represents the pseudocount in this case.</p>
<div class="cell" data-output-location="column-fragment">
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">d</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="va">hour00_rep1</span><span class="op">:</span><span class="va">hour14_rep3</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="op">?</span><span class="op">?</span><span class="op">(</span><span class="va">value</span><span class="op">)</span>, color<span class="op">=</span><span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_density</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="correlation-analysis" class="level2"><h2 class="anchored" data-anchor-id="correlation-analysis">correlation analysis</h2>
<p>prepare the data for analysis</p>
<div class="cell" data-output-location="slide">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># pull counts</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span>  <span class="va">d</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">select_if</span><span class="op">(</span><span class="va">is.numeric</span><span class="op">)</span> <span class="op">|&gt;</span><span class="co"># keep only the numeric columns</span></span>
<span>  <span class="fu">mutate_all</span><span class="op">(</span><span class="fu">funs</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">+</span> <span class="va">.</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span><span class="co"># log2 transform</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># matrix</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">d</span><span class="op">$</span><span class="va">gene_symbol</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/scale.html">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="co"># scale</span></span>
<span></span>
<span><span class="co"># pairwise pearson correlation</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="op">?</span><span class="op">?</span><span class="op">?</span></span>
<span></span>
<span><span class="co"># heatmap</span></span>
<span><span class="fu">pheatmap</span><span class="op">(</span></span>
<span>  mat <span class="op">=</span> <span class="va">p</span>,</span>
<span>  clustering_distance_rows <span class="op">=</span> <span class="st">"??"</span>,</span>
<span>  clustering_distance_cols <span class="op">=</span> <span class="st">"??"</span>,</span>
<span>  clustering_method <span class="op">=</span> <span class="st">"??"</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="pca-1" class="level2"><h2 class="anchored" data-anchor-id="pca-1">PCA</h2>
<p>PCA is a common dimensionality reduction method that is used to visualize the similarity and differences in your data.</p>
<p><strong>Let’s watch this fantastic 5 minute video explaining PCA</strong></p>
<div class="quarto-video ratio ratio-16x9"><iframe data-external="1" src="https://www.youtube.com/embed/HMOI_lkzW08" title="" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe></div>
<p>For more detailed explanations go <a href="https://www.nature.com/articles/nbt0308-303">here</a> and <a href="https://liorpachter.wordpress.com/2014/05/26/what-is-principal-component-analysis/">here</a>.</p>
</section><section id="pca-2" class="level2"><h2 class="anchored" data-anchor-id="pca-2">PCA</h2>
<div class="cell" data-output-location="column-fragment">
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># pairwise pearson correlation</span></span>
<span><span class="va">pc</span> <span class="op">&lt;-</span> </span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">pc</span><span class="op">)</span> <span class="co"># summarize the PCs by variance</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="pca---prepare-visualization" class="level2"><h2 class="anchored" data-anchor-id="pca---prepare-visualization">PCA - prepare visualization</h2>
<div class="cell" data-output-location="slide">
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># create a dataframe with the importance/explanation of variation for each PC</span></span>
<span><span class="va">pca_data_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">pc</span><span class="op">)</span><span class="op">$</span><span class="va">importance</span> <span class="op">|&gt;</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pca_data_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">pca_data_info</span>, digits <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># we make a dataframe out of the rotations and will use this to plot</span></span>
<span><span class="va">pca_plot_data</span> <span class="op">&lt;-</span> <span class="va">pc</span><span class="op">$</span><span class="va">rotation</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">rownames_to_column</span><span class="op">(</span>var <span class="op">=</span> <span class="st">"ID"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">separate</span><span class="op">(</span></span>
<span>    <span class="va">ID</span>,</span>
<span>    into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"time"</span>,<span class="st">"rep"</span><span class="op">)</span>,</span>
<span>    sep <span class="op">=</span> <span class="st">"_"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># recode "rep"</span></span>
<span><span class="va">pca_plot_data</span><span class="op">$</span><span class="va">rep</span> <span class="op">&lt;-</span> <span class="fu">recode</span><span class="op">(</span></span>
<span>  <span class="va">pca_plot_data</span><span class="op">$</span><span class="va">rep</span>,</span>
<span>  rep1 <span class="op">=</span> <span class="st">"A"</span>,</span>
<span>  rep2 <span class="op">=</span> <span class="st">"B"</span>,</span>
<span>  rep3 <span class="op">=</span> <span class="st">"C"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># gsub hour</span></span>
<span><span class="va">pca_plot_data</span><span class="op">$</span><span class="va">time</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span></span>
<span>  pattern <span class="op">=</span> <span class="st">"hour"</span>,</span>
<span>  replacement <span class="op">=</span> <span class="st">""</span>,</span>
<span>  x <span class="op">=</span> <span class="va">pca_plot_data</span><span class="op">$</span><span class="va">time</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span></span>
<span>  <span class="va">pca_plot_data</span>, </span>
<span>  <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span>, color<span class="op">=</span><span class="va">time</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">xlab</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"PC1, %"</span>,<span class="fl">100</span> <span class="op">*</span> <span class="va">pca_data_info</span><span class="op">[</span><span class="st">"Proportion of Variance"</span>,<span class="st">"PC1"</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ylab</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"PC2, %"</span>,<span class="fl">100</span> <span class="op">*</span> <span class="va">pca_data_info</span><span class="op">[</span><span class="st">"Proportion of Variance"</span>,<span class="st">"PC2"</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"PCA for DUX4 timecourse"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="famous-pca-example" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="famous-pca-example">Famous PCA example</h2>
<p>Using gene expression as your measurement, do you think the mouse liver is more similar to a mouse heart or a human liver?</p>
<p>The Mouse ENCODE Consortium reported that comparative gene expression data from human and mouse tend to cluster more by species rather than by tissue.</p>
<p><img src="../img/bad_pca.jpg" class="img-fluid"></p>
<p><a href="https://www.nature.com/articles/nature13992">A comparative encyclopedia of DNA elements in the mouse genome</a></p>
<p><a href="https://www.pnas.org/doi/full/10.1073/pnas.1413624111">Comparison of the transcriptional landscapes between human and mouse tissues</a></p>
</section><section id="some-found-this-hard-to-believe" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="some-found-this-hard-to-believe">Some found this hard to believe</h2>
<p><a href="https://f1000research.com/articles/4-121#ref-1">Yoav Gilad’s lab recapitulated the initial result:</a></p>
<p><img src="../img/redo_encode.jpg" class="img-fluid"></p>
<p>This observation was surprising, as it contradicted much of the comparative gene regulatory data collected previously, as well as the common notion that major developmental pathways are highly conserved across a wide range of species, in particular across mammals.</p>
</section><section id="careful-with-batch-effects" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="careful-with-batch-effects">Careful with batch effects</h2>
<p>But noticed something funny about which samples were sequenced on the same lanes.</p>
<p><img src="../img/batch.jpg" class="img-fluid"></p>
</section><section id="accounting-for-batch-effects" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="accounting-for-batch-effects">Accounting for batch effects</h2>
<p><img src="../img/fixencode.jpg" class="img-fluid"></p>
<p>Here we show that the Mouse ENCODE gene expression data were collected using a flawed study design, which confounded sequencing batch (namely, the assignment of samples to sequencing flowcells and lanes) with species. When we account for the batch effect, the corrected comparative gene expression data from human and mouse tend to cluster by tissue, not by species.</p>
</section><section id="k-means-clustering-to-look-for-patterns" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="k-means-clustering-to-look-for-patterns">K-means clustering to look for patterns</h2>
<p>Goal: to partition <code>n</code> observations into <code>k</code> clusters in which each observation belongs to the cluster with the nearest mean, serving as a prototype of the cluster. –Wiki</p>
<p><strong>K-means</strong> <iframe width="300" height="150" src="https://www.youtube.com/embed/4b5d3muPQmA" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe></p>
</section><section id="k-means-data-preparation" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="k-means-data-preparation">K-Means data preparation</h2>
<ol type="1">
<li>Rows are observations (individuals) and columns are variables</li>
<li>Any missing value in the data must be removed or estimated.</li>
<li>The data must be standardized (i.e., scaled) to make variables comparable.</li>
</ol></section><section id="scaling-or-z-score" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="scaling-or-z-score">Scaling or z-score</h2>
<p><img src="http://www.z-table.com/uploads/2/1/7/9/21795380/1426878678.png" class="img-fluid">{%20}</p>
<p><span class="math inline">\(x\)</span> = observation<br><span class="math inline">\(\mu\)</span> = population mean<br><span class="math inline">\(\sigma\)</span> = population sd</p>
<p>We will be using this function on each row. This will allow comparison of relative changes across a row, for all rows.</p>
</section><section id="k-means-clustering" class="level2"><h2 class="anchored" data-anchor-id="k-means-clustering">K-Means clustering</h2>
<ol type="1">
<li>Computing k-means clustering in R (pheatmap)</li>
<li>Determine appropriate cluster number</li>
<li>Add new column with cluster number to initial data</li>
</ol></section><section id="how-do-we-figure-out-the-optimal-clusters" class="level2"><h2 class="anchored" data-anchor-id="how-do-we-figure-out-the-optimal-clusters">How do we figure out the optimal # clusters?</h2>
<p>There are many methods, but we will stick with the “elbow” method.</p>
<p>K-means is minimizing the total within cluster sum of squares (wss).</p>
<p>We pick the cluster where that drop in total reaches diminishing returns -&gt; the elbow.</p>
</section><section id="lets-cluster-once-to-see" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="lets-cluster-once-to-see">Let’s cluster once to see</h2>
<div class="cell" data-output-location="slide">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">33</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">mat =</span> x,</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_distance_rows =</span> <span class="st">"euclidean"</span>,</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_method =</span> <span class="st">"ward.D2"</span>,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">kmeans_k =</span> ??,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"none"</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>tmp<span class="sc">$</span>kmeans<span class="sc">$</span>tot.withinss</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="functions-in-r" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="functions-in-r">Functions in R</h2>
<p><img src="https://www.learnbyexample.org/wp-content/uploads/r/r-function-syntax.png" class="img-fluid"></p>
</section><section id="create-function-to-calculate-wss" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="create-function-to-calculate-wss">Create function to calculate wss</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>wss <span class="ot">&lt;-</span> <span class="cf">function</span>(knum) {</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  ph <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">mat =</span> x,</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">kmeans_k =</span> ??,</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale =</span> <span class="st">"none"</span>,</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">clustering_distance_rows =</span> <span class="st">"euclidean"</span>,</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">clustering_method =</span> <span class="st">"ward.D2"</span>,</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">silent =</span> <span class="cn">TRUE</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(ph<span class="sc">$</span>kmeans<span class="sc">$</span>tot.withinss)</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="fu">wss</span>(<span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="find-the-elbow" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="find-the-elbow">find the elbow</h2>
<div class="cell" data-output-location="fragment">
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">tibble</span><span class="op">(</span>wss<span class="op">=</span><span class="fu">map_vec</span><span class="op">(</span><span class="fl">2</span><span class="op">:</span><span class="fl">15</span>,<span class="va">wss</span><span class="op">)</span>,</span>
<span>       k<span class="op">=</span><span class="fl">2</span><span class="op">:</span><span class="fl">15</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="va">.</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">k</span>, y<span class="op">=</span><span class="va">wss</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="final-clustering" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="final-clustering">Final clustering</h2>
<div class="cell" data-output-location="fragment">
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">33</span><span class="op">)</span></span>
<span><span class="va">c</span> <span class="op">&lt;-</span> <span class="fu">pheatmap</span><span class="op">(</span></span>
<span>  mat <span class="op">=</span> <span class="va">x</span>,</span>
<span>  clustering_distance_rows <span class="op">=</span> <span class="st">"euclidean"</span>,</span>
<span>  clustering_method <span class="op">=</span> <span class="st">"ward.D2"</span>,</span>
<span>  kmeans_k <span class="op">=</span> <span class="fl">7</span>,</span>
<span>  cluster_cols <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  scale <span class="op">=</span> <span class="st">"none"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cg</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  Cluster<span class="op">=</span><span class="va">c</span><span class="op">$</span><span class="va">kmeans</span><span class="op">$</span><span class="va">cluster</span>,</span>
<span>  gene_symbol<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">c</span><span class="op">$</span><span class="va">kmeans</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">d</span>, <span class="va">cg</span>, by<span class="op">=</span><span class="st">"gene_symbol"</span><span class="op">)</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="which-clusters-contains-dux4-targets" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="which-clusters-contains-dux4-targets">Which cluster(s) contains DUX4 targets?</h2>
<p><a href="https://www.youtube.com/watch?v=udyAvvaMjfM">Fisher’s Exact Test and the Hypergeometric Distribution</a></p>
<div class="cell" data-output-location="slide">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># list of genes by dux4 targeting</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>duxList <span class="ot">&lt;-</span> <span class="fu">split</span>(cd<span class="sc">$</span>gene_symbol, cd<span class="sc">$</span>??)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co"># list of genes by clustering</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>clustList <span class="ot">&lt;-</span> <span class="fu">split</span>(cd<span class="sc">$</span>gene_symbol, <span class="fu">as.factor</span>(cd<span class="sc">$</span>??))</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate all overlaps between lists</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>gom.duxclust <span class="ot">&lt;-</span> <span class="fu">newGOM</span>(??List,</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>                       ??List,</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">genome.size =</span> ??)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="fu">getMatrix</span>(gom.duxclust, <span class="st">"pval"</span>) <span class="sc">|&gt;</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">|&gt;</span></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"clust"</span>) <span class="sc">|&gt;</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(target)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="lets-calculate-the-empirical-p-value-of-the-cluster-most-enriched-for-dux4-targets-by-sampling" class="level2 smaller"><h2 class="smaller anchored" data-anchor-id="lets-calculate-the-empirical-p-value-of-the-cluster-most-enriched-for-dux4-targets-by-sampling">Let’s calculate the empirical p-value of the cluster most enriched for DUX4 targets by sampling</h2>
<p>In order to do this, you will need to:</p>
<ol type="1">
<li>Identify which cluster is the most enriched for DUX4 targets.
<ul>
<li>Determine how many genes are in the cluster. You will need to know this to figure out how many genes to sample from the whole data set.</li>
<li>Determine how many of the genes in the cluster are DUX4 targets. This is the metric that you are interested in comparing between the null distribution and your observation.</li>
</ul>
</li>
<li>Generate 1000 random sample of the proper size from all genes and find out how many of them are DUX4 targets.</li>
<li>Visualize the distribution of DUX4 targets in these 1000 random (your null distribution) and overlay the number of DUX4 targets you observed in the cluster that was most enriched for DUX4 targets.</li>
</ol>


</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/rnabioco\.github\.io\/molb-7950\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/rnabioco/molb-7950/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/rnabioco/molb-7950/edit/main/exercises/ex-15.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This page is built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>


</body></html>