---
title: "RNA Block - Problem Set 25"
---

```{r}
#| label: load-libs

library(tidyverse)
library(cowplot)
library(here)
library(ggrepel)
library(pheatmap)
library(RColorBrewer)
```

## Problem Set

Total points: 20. Q1 - 10 pts, Q2 - 10 points each.

Exercises: Take rMATS SE output, apply filters, and make a volcano plot. Also, make a heatmap of the significant PSI for each sample using all replicates. NOTE: this analysis will be performed for the **mutually exclusive exons**.

# Exercises

We worked with an rMATS output file from an experiment in which mouse embryonic stem cells had been treated with either an shRNA against RBFOX2 or a control, nontargeting shRNA. We applied filters to remove any event from consideration in which the number of informative reads for that event (IJC + SJC) was less than 20 in **any** sample.

# Q1 make volcano plot

Read in the rMATS output for **mutually exclusive exons**. Apply a new read coverage filter such that any event with less than **10** informative junction reads in **any** sample is removed. Then make a volcano plot where each dot is an event, the x-axis is the difference in PSI between the two conditions (aka `IncLevelDifference`) and the y axis is -log10 of the `FDR`. Color dots that pass a significance threshold (FDR \< 0.05) in red. Label the 4 most significant events (i.e. 4 lowest FDR values) with the name of the gene they are in.

```{r}
#| eval: false
#| echo: true
#| label: q1

# Read in table


# Split the replicate read counts that are separated by commas into different columns


# filter events (reads >= 10)


# Separate the inclusion levels for each sample and replicte (you will need this later)



# Add another column to table that says whether or not this event is significant (FDR < 0.05)




# Volcano Plot with sig events in red
```

![explosion](/img/block-rna/psi_volcano.png)

# Exercise 2

Take your skipped exon data and make a heatmap where the rows are events, columns are samples (each replicate separately, 4 replicates per condition), and the color of the cell is a scaled PSI value. Only plot significant (FDR \< 0.05) events.

```{r}
#| eval: false
#| echo: true
#| label: q2

# Filter for those that are significant (FDR < 0.05) and only keep columns with inclusion differences


# Plot with pheatmap(), using scale = 'row' to plot scaled PSI values
```

![you don't need to have the gene symbols](/img/block-rna/psi_heatmap.png)
